--- a/programs/pluto/connections.c
+++ b/programs/pluto/connections.c
@@ -36,6 +36,7 @@
 #include <stdio.h>
 #include <stddef.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <unistd.h>
 #include <netinet/in.h>
 #include <sys/socket.h>
--- a/programs/pluto/rcv_whack.c
+++ b/programs/pluto/rcv_whack.c
@@ -25,6 +25,7 @@
  */
 
 #include <unistd.h>			/* for getsid() */
+#include <limits.h>
 
 #include "connections.h"
 #include "rcv_whack.h"
