--- a/configure.in
+++ b/configure.in
@@ -31,7 +31,6 @@ fi
 
 dnl Checks for programs.
 AC_LANG_CPLUSPLUS
-AC_PROG_CC
 AC_PROG_CXX
 AC_PROG_CXXCPP
 AC_PROG_INSTALL
@@ -60,13 +59,13 @@ if [[ -n "$snprintf" ]]; then
    snprintf="#define NO_SNPRINTF"
 fi
 
-AC_CHECK_HEADERS(algorithm algo.h algo)
+AC_CHECK_HEADERS(algorithm)
 
 AC_SUBST(linux_pthread)
 AC_TRY_COMPILE([#define _GNU_SOURCE
 #include <pthread.h>
 ] , [pthread_mutexattr_t attr;
-    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE_NP);]
+    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);]
   , linux_pthread="yes")
 if [[ -n "$linux_pthread" ]]; then
    linux_pthread="#define LINUX_PTHREAD"
@@ -83,7 +82,7 @@ void * thread_func(void * param) { return NULL; }
   , thread_ldflags="-pthread")
 
 AC_SUBST(large_file)
-AC_TRY_RUN([#ifndef _LARGEFILE64_SOURCE
+AC_TRY_COMPILE([#ifndef _LARGEFILE64_SOURCE
 #define _LARGEFILE64_SOURCE
 #endif
 #include <stdio.h>
