From: Debian PHP Maintainers <pkg-php-maint@lists.alioth.debian.org>
Date: Sat, 2 May 2015 10:26:53 +0200
Subject: dont-gitclean-in-build

---
 build/build.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build/build.mk b/build/build.mk
index 9452984..b964def 100644
--- a/build/build.mk
+++ b/build/build.mk
@@ -63,6 +63,5 @@ gitclean-work:
 	@if (test ! -f '.git/info/exclude' || grep -s "git-ls-files" .git/info/exclude); then \
 		(echo "Rebuild .git/info/exclude" && echo '*.o' > .git/info/exclude && git svn propget svn:ignore | grep -v config.nice >> .git/info/exclude); \
 	fi; \
-	git clean -X -f -d;
 
 .PHONY: $(ALWAYS) snapshot
