--- a/Zend/zend_alloc.h
+++ b/Zend/zend_alloc.h
@@ -27,12 +27,12 @@
 #include "zend.h"
 
 #ifndef ZEND_MM_ALIGNMENT
-# define ZEND_MM_ALIGNMENT ((size_t) 8)
+# define ZEND_MM_ALIGNMENT (8)
 # define ZEND_MM_ALIGNMENT_LOG2 Z_L(3)
 #elif ZEND_MM_ALIGNMENT < 4
 # undef ZEND_MM_ALIGNMENT
 # undef ZEND_MM_ALIGNMENT_LOG2
-# define ZEND_MM_ALIGNMENT ((size_t) 4)
+# define ZEND_MM_ALIGNMENT (4)
 # define ZEND_MM_ALIGNMENT_LOG2 Z_L(2)
 #endif
 
