--- a/src/main.c
+++ b/src/main.c
@@ -326,8 +326,8 @@ static void initialize_config_defaults (
         conf->errorpages = NULL;
         conf->stathost = safestrdup (TINYPROXY_STATHOST);
         conf->idletimeout = MAX_IDLE_TIME;
-        conf->logf_name = safestrdup (LOCALSTATEDIR "/log/tinyproxy/tinyproxy.log");
-        conf->pidpath = safestrdup (LOCALSTATEDIR "/run/tinyproxy/tinyproxy.pid");
+        conf->logf_name = safestrdup (LOCALSTATEDIR "/log/tinyproxy.log");
+        conf->pidpath = safestrdup (LOCALSTATEDIR "/tinyproxy.pid");
 }
 
 /**
