--- a/Makefile.am
+++ b/Makefile.am
@@ -551,12 +551,6 @@ $(srcdir)/effective_tld_names.dat:
 pubsuffix.cc: $(srcdir)/effective_tld_names.dat
 	$(srcdir)/mkpubsuffixcc $< $@
 
-## Config file
-sysconf_DATA = recursor.yml-dist
-
-recursor.yml-dist: pdns_recursor
-	dir=$$(mktemp -d) && touch "$$dir/recursor.yml" && ./pdns_recursor --config-dir="$$dir" --config=default 2> /dev/null > $@ && rm "$$dir/recursor.yml" && rmdir "$$dir"
-
 ## Manpages
 MANPAGES=pdns_recursor.1 \
 	 rec_control.1
