--- a/webserver.cc
+++ b/webserver.cc
@@ -36,6 +36,8 @@
 #include "uuid-utils.hh"
 #include <yahttp/router.hpp>
 
+using namespace boost::placeholders;
+
 json11::Json HttpRequest::json()
 {
   string err;
--- a/ws-recursor.cc
+++ b/ws-recursor.cc
@@ -46,6 +46,7 @@
 
 extern thread_local FDMultiplexer* t_fdm;
 
+using namespace boost::placeholders;
 using json11::Json;
 
 void productServerStatisticsFetch(map<string,string>& out)
