diff --git a/configure.in b/configure.in
index 9082c5b..2c04edc 100644
--- a/configure.in
+++ b/configure.in
@@ -2283,36 +2283,6 @@ if test "$with_python" = yes; then
   CPPFLAGS=$ac_save_CPPFLAGS
 fi
 
-#
-# Check for DocBook and tools
-#
-PGAC_PATH_XMLLINT
-PGAC_CHECK_DOCBOOK(4.2)
-PGAC_PATH_PROGS(DBTOEPUB, dbtoepub)
-PGAC_PATH_PROGS(XSLTPROC, xsltproc)
-PGAC_PATH_PROGS(FOP, fop)
-
-#
-# Check for test tools
-#
-if test "$enable_tap_tests" = yes; then
-  # Check for necessary modules, unless user has specified the "prove" to use;
-  # in that case it's her responsibility to have a working configuration.
-  # (prove might be part of a different Perl installation than perl, eg on
-  # MSys, so the result of AX_PROG_PERL_MODULES could be irrelevant anyway.)
-  if test -z "$PROVE"; then
-    # Test::More and Time::HiRes are supposed to be part of core Perl,
-    # but some distros omit them in a minimal installation.
-    AX_PROG_PERL_MODULES([IPC::Run Test::More=0.87 Time::HiRes], ,
-      [AC_MSG_ERROR([Additional Perl modules are required to run TAP tests])])
-  fi
-  # Now make sure we know where prove is
-  PGAC_PATH_PROGS(PROVE, prove)
-  if test -z "$PROVE"; then
-    AC_MSG_ERROR([prove not found])
-  fi
-fi
-
 # Thread testing
 
 # We have to run the thread test near the end so we have all our symbols
