--- a/configure.ac
+++ b/configure.ac
@@ -2161,13 +2161,13 @@ if test x"$USE_SLICING_BY_8_CRC32C" = x"
         USE_ARMV8_CRC32C=1
       else
         # ARM CRC Extension, with runtime check?
-        if test x"$pgac_armv8_crc32c_intrinsics" = x"yes"; then
-          USE_ARMV8_CRC32C_WITH_RUNTIME_CHECK=1
-        else
+        #if test x"$pgac_armv8_crc32c_intrinsics" = x"yes"; then
+        #  USE_ARMV8_CRC32C_WITH_RUNTIME_CHECK=1
+        #else
           # fall back to slicing-by-8 algorithm, which doesn't require any
           # special CPU support.
           USE_SLICING_BY_8_CRC32C=1
-	fi
+	#fi
       fi
     fi
   fi
