--- a/configure.in
+++ b/configure.in
@@ -2019,9 +2019,9 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [
 # flags. If not, check if adding -march=armv8-a+crc flag helps.
 # CFLAGS_ARMV8_CRC32C is set if the extra flag is required.
 PGAC_ARMV8_CRC32C_INTRINSICS([])
-if test x"$pgac_armv8_crc32c_intrinsics" != x"yes"; then
-  PGAC_ARMV8_CRC32C_INTRINSICS([-march=armv8-a+crc])
-fi
+#if test x"$pgac_armv8_crc32c_intrinsics" != x"yes"; then
+#  PGAC_ARMV8_CRC32C_INTRINSICS([-march=armv8-a+crc])
+#fi
 AC_SUBST(CFLAGS_ARMV8_CRC32C)
 
 # Select CRC-32C implementation.
