From 6945698778caf7cdaace9ce8dae82162dbe2ee9f Mon Sep 17 00:00:00 2001
From: Johannes Morgenroth <jm@m-network.de>
Date: Thu, 3 Jan 2019 07:26:51 +0100
Subject: [PATCH] Use const iterator in const function of MemoryBundleSet

---
 ibrdtn/data/MemoryBundleSet.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/ibrdtn/data/MemoryBundleSet.cpp
+++ b/ibrdtn/data/MemoryBundleSet.cpp
@@ -134,7 +134,7 @@ namespace dtn
 				// the bundles set. This happen if the MemoryBundleSet gets deserialized.
 				if (!_consistent) return true;
 
-				bundle_set::iterator iter = _bundles.find(dtn::data::MetaBundle::create(bundle));
+				bundle_set::const_iterator iter = _bundles.find(dtn::data::MetaBundle::create(bundle));
 				return (iter != _bundles.end());
 			}
 
