--- a/usr/lib/byobu/logo
+++ b/usr/lib/byobu/logo
@@ -111,6 +111,10 @@ __logo() {
 			logo=" lm "
 			$MARKUP && printf "$(color g w)$logo$(color -)$(color g w)$(color -) " || printf "$logo"
 		;;
+		*openwrt*)
+			logo="OWrt"
+			$MARKUP && printf "$(color b colour66 W)%s$(color -)" "$logo" || printf "$logo"
+		;;
 		*red*hat*|*rhel*)
 			logo=" RH "
 			$MARKUP && printf "$(color R k)%s$(color -)" "$logo" || printf "$logo"
