--- a/usr/lib/byobu/processes
+++ b/usr/lib/byobu/processes
@@ -20,15 +20,15 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 __processes_detail() {
-	ps -ej
+	ps -ej 2>/dev/null
 }
 
 __processes() {
 	local count=
 	if [ -r /proc ]; then
-		count=$(ls -d /proc/[0-9]* 2>/dev/null| wc -l)
+		count=$(ls -d /proc/[0-9]* 2>/dev/null | wc -l)
 	else
-		count=$(ps -ef | wc -l | awk '{print $1}')
+		count=$(ps -ef | wc -l)
 	fi
 	[ -n "$count" ] || return
 	color b y w; printf "%s" "$count"; color -; color y w; printf "&"; color --
