--- a/http_parser.c
+++ b/http_parser.c
@@ -1816,6 +1816,8 @@ reexecute:
             case 2:
               parser->upgrade = 1;

+             /* FALLTHROUGH */
+
             case 1:
               parser->flags |= F_SKIPBODY;
               break;
@@ -2375,7 +2377,7 @@ http_parser_parse_url(const char *buf, s
       case s_req_server_with_at:
         found_at = 1;

-      /* FALLTROUGH */
+      /* FALLTHROUGH */
       case s_req_server:
         uf = UF_HOST;
         break;
