--- a/configure.ac
+++ b/configure.ac
@@ -228,7 +228,7 @@ AC_COMPILE_IFELSE(  [AC_LANG_PROGRAM([[#
                     ]
                  )
 AC_MSG_CHECKING([for wcsrtombs])
-AC_RUN_IFELSE(  [AC_LANG_PROGRAM([[#include <wchar.h>
+AC_COMPILE_IFELSE(  [AC_LANG_PROGRAM([[#include <wchar.h>
 #include <string.h>]],
                                  [[
 mbstate_t st;
@@ -250,7 +250,7 @@ else
                     ]
                  )
 AC_MSG_CHECKING([for mbsrtowcs])
-AC_RUN_IFELSE(  [AC_LANG_PROGRAM([[#include <wchar.h>
+AC_COMPILE_IFELSE(  [AC_LANG_PROGRAM([[#include <wchar.h>
 #include <string.h>]],
                                  [[
 mbstate_t st;
