--- a/src/lib/log/logger_level_impl.cc
+++ b/src/lib/log/logger_level_impl.cc
@@ -10,6 +10,7 @@
 #include <string.h>
 #include <iostream>
 #include <boost/lexical_cast.hpp>
+#include <boost/static_assert.hpp>
 
 #include <log4cplus/logger.h>
 
--- a/src/lib/dns/rdataclass.cc
+++ b/src/lib/dns/rdataclass.cc
@@ -24,6 +24,7 @@
 #include <dns/txt_like.h>
 #include <util/buffer.h>
 #include <util/encode/encode.h>
+#include <boost/static_assert.hpp>
 
 #include <cerrno>
 #include <cstring>
