From c479fba2f0f478e828cd08dde3c3b5e8ae7908d4 Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Mon, 29 Nov 2021 18:31:14 -0800
Subject: [PATCH] fix matroska memory leaks

Signed-off-by: Rosen Penev <rosenp@gmail.com>
---
 src/metadata/matroska_handler.cc | 2 --
 1 file changed, 2 deletions(-)

--- a/src/metadata/matroska_handler.cc
+++ b/src/metadata/matroska_handler.cc
@@ -137,8 +137,6 @@ void MatroskaHandler::parseMKV(const std
 
             delete (elL1->SkipData(ebmlStream, EBML_CONTEXT(elL1)));
             delete elL1;
-            if (activeFlag == 0) // terminate search
-                break;
         } // while elementLevel1
 
         delete (elL0->SkipData(ebmlStream, LIBMATROSKA_NAMESPACE::KaxSegment_Context));
