From cafd77993ec5e16634b774b65bf6da9b34a21fc5 Mon Sep 17 00:00:00 2001
From: Jim Klimov <jimklimov+nut@gmail.com>
Date: Wed, 31 Aug 2022 11:24:19 +0200
Subject: [PATCH] clients/upsclient.h: ensure time_t is defined


--- a/clients/upsclient.h
+++ b/clients/upsclient.h
@@ -41,6 +41,18 @@
 	#include <limits.h>
 #endif
 
+/* Not including NUT timehead.h because this is part of end-user API */
+#ifdef TIME_WITH_SYS_TIME
+# include <sys/time.h>
+# include <time.h>
+#else
+# ifdef HAVE_SYS_TIME_H
+#  include <sys/time.h>
+# else
+#  include <time.h>
+# endif
+#endif
+
 #ifdef __cplusplus
 /* *INDENT-OFF* */
 extern "C" {
