diff -uprN a/internal.c b/internal.c
--- a/internal.c	2014-06-13 16:44:14.263689579 +0200
+++ b/internal.c	2014-06-13 16:46:04.229020785 +0200
@@ -337,10 +337,12 @@ int internal_init(void)
 	init_superio_ite();
 #endif
 
+#if defined(__i386__) || defined(__x86_64__) || defined (__arm__)
 	if (board_flash_enable(board_vendor, board_model, cb_vendor, cb_model)) {
 		msg_perr("Aborting to be safe.\n");
 		return 1;
 	}
+#endif
 
 #if defined(__i386__) || defined(__x86_64__) || defined (__mips)
 	register_par_programmer(&par_programmer_internal, internal_buses_supported);
