https://bugs.launchpad.net/hplip/+bug/1778626

--- a/prnt/hpcups/genPCLm.cpp
+++ b/prnt/hpcups/genPCLm.cpp
@@ -171,7 +171,7 @@ Defines
 #define rgb_2_gray(r,g,b) (ubyte)(0.299*(double)r+0.587*(double)g+0.114*(double)b)
 
 // Note: this is required for debugging
-boolean writeOutputFile(int numBytes, ubyte *ptr, char *user_name);
+bool writeOutputFile(int numBytes, ubyte *ptr, char *user_name);
 
 /* 
 ********************************************* Helper Routines **************************
@@ -343,7 +343,7 @@ bool PCLmGenerator::addKids(sint32 kidObj)
 	return(true);
 }
 
-boolean writeOutputFile(int numBytes, ubyte *ptr, char *user_name)
+bool writeOutputFile(int numBytes, ubyte *ptr, char *user_name)
 {
 	FILE *outputFile;
 	char outFileName[MAX_FILE_PATH_LEN];
@@ -1074,7 +1074,7 @@ void PCLmGenerator::writePDFGrammarPage(int imageWidth, int imageHeight, int num
 * Limitations:
 *   - 
 *****************************************************************************************/
-boolean prepImageForBacksideDuplex(ubyte *imagePtr, sint32 imageHeight, sint32 imageWidth, sint32 numComponents)
+bool prepImageForBacksideDuplex(ubyte *imagePtr, sint32 imageHeight, sint32 imageWidth, sint32 numComponents)
 {
 	sint32 numBytes=imageHeight*imageWidth*numComponents;
 	ubyte *head, *tail, t0, t1, t2; 
