--- a/source4/heimdal_build/wscript_configure	2019-12-03
+++ b/source4/heimdal_build/wscript_configure	2019-12-03
@@ -199,7 +199,7 @@ def check_system_heimdal_lib(name, funct
 def check_system_heimdal_binary(name):
     if conf.LIB_MAY_BE_BUNDLED(name):
         return False
-    if not conf.find_program(name, var=name.upper()):
+    if not conf.find_program(name + '_samba', var=name.upper()):
         return False
     conf.define('USING_SYSTEM_%s' % name.upper(), 1)
     return True
