--- a/third_party/heimdal/cf/make-proto.pl
+++ b/third_party/heimdal/cf/make-proto.pl
@@ -4,7 +4,7 @@
 use Getopt::Std;
 use File::Compare;
 
-use JSON;
+use JSON::PP;
 
 my $comment = 0;
 my $doxygen = 0;
@@ -70,7 +70,7 @@ if($opt_x) {
     my $EXP;
     local $/;
     open(EXP, '<', $opt_x) || die "open ${opt_x}";
-    my $obj = JSON->new->utf8->decode(<EXP>);
+    my $obj = JSON::PP->new->utf8->decode(<EXP>);
     close $EXP;
 
     foreach my $x (keys %$obj) {
