# Gentoo bug 855047
--- a/lib/util/access.c
+++ b/lib/util/access.c
@@ -115,7 +115,7 @@ static bool string_match(const char *tok
 			return true;
 		}
 	} else if (tok[0] == '@') { /* netgroup: look it up */
-#ifdef HAVE_NETGROUP
+#if defined(HAVE_NETGROUP) && defined(HAVE_INNETGR)
 		DATA_BLOB tmp;
 		char *mydomain = NULL;
 		char *hostname = NULL;
--- a/source3/auth/user_util.c
+++ b/source3/auth/user_util.c
@@ -135,7 +135,7 @@ static void store_map_in_gencache(TALLOC
 
 bool user_in_netgroup(TALLOC_CTX *ctx, const char *user, const char *ngname)
 {
-#ifdef HAVE_NETGROUP
+#if defined(HAVE_NETGROUP) && defined(HAVE_INNETGR)
 	char nis_domain_buf[256];
 	const char *nis_domain = NULL;
 	char *lowercase_user = NULL;
