diff --git a/ConfigureChecks.cmake b/ConfigureChecks.cmake
index c8bb2aa..344ba59 100644
--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -5,7 +5,6 @@ include(CheckFunctionExists)
 include(CheckLibraryExists)
 include(CheckTypeSize)
 include(CheckStructHasMember)
-include(TestBigEndian)
 
 set(PACKAGE ${PROJECT_NAME})
 set(VERSION ${PROJECT_VERSION})
@@ -465,6 +464,8 @@ if (WITH_GSSAPI AND NOT GSSAPI_FOUND)
 endif (WITH_GSSAPI AND NOT GSSAPI_FOUND)
 
 # ENDIAN
-if (NOT WIN32)
-    test_big_endian(WORDS_BIGENDIAN)
-endif (NOT WIN32)
+if (WITH_BIG_ENDIAN)
+    set(WORDS_BIGENDIAN 1)
+else (WITH_BIG_ENDIAN)
+    set(WORDS_BIGENDIAN 0)
+endif (WITH_BIG_ENDIAN)
