Index: btrfs-progs-v3.19.1/Makefile.in
===================================================================
--- btrfs-progs-v3.19.1.orig/Makefile.in
+++ btrfs-progs-v3.19.1/Makefile.in
@@ -21,7 +21,7 @@ CFLAGS = @CFLAGS@ \
 LDFLAGS = @LDFLAGS@ \
 	  -rdynamic
 
-LIBS = @UUID_LIBS@ @BLKID_LIBS@ @ZLIB_LIBS@ @LZO2_LIBS@ -L. -pthread
+LIBS = @ATTR_LIBS@ @UUID_LIBS@ @BLKID_LIBS@ @ZLIB_LIBS@ @LZO2_LIBS@ -L. -pthread
 LIBBTRFS_LIBS = $(LIBS)
 
 # Static compilation flags
Index: btrfs-progs-v3.19.1/configure.ac
===================================================================
--- btrfs-progs-v3.19.1.orig/configure.ac
+++ btrfs-progs-v3.19.1/configure.ac
@@ -115,6 +115,9 @@ dnl
 dnl The default PKG_CHECK_MODULES() action-if-not-found is end the
 dnl execution with error. The static libs are optional.
 
+PKG_CHECK_MODULES(ATTR, [libattr])
+PKG_STATIC(BLKID_LIBS_STATIC, [libattr])
+
 PKG_CHECK_MODULES(BLKID, [blkid])
 PKG_STATIC(BLKID_LIBS_STATIC, [blkid])
 
