--- a/loudmouth/lm-sasl.c
+++ b/loudmouth/lm-sasl.c
@@ -529,7 +529,7 @@ sasl_md5_prepare_response (LmSASL *sasl,
     }
 
     nonce = g_hash_table_lookup (challenge, "nonce");
-    if (nonce == NULL || nonce == '\0') {
+    if (nonce == NULL || *nonce == '\0') {
         g_log (LM_LOG_DOMAIN, LM_LOG_LEVEL_SASL,
                "%s: server didn't provide a nonce in the challenge",
                G_STRFUNC);
