diff --git a/setuptools/dist.py b/setuptools/dist.py
index b004f92..66b9d89 100644
--- a/setuptools/dist.py
+++ b/setuptools/dist.py
@@ -17,7 +17,6 @@ from setuptools.extern.six.moves import map
 from pkg_resources.extern import packaging
 
 from setuptools.depends import Require
-from setuptools import windows_support
 from setuptools.monkey import get_unpatched
 import pkg_resources
 
@@ -383,7 +382,6 @@ class Distribution(_Distribution):
         egg_cache_dir = os.path.join(os.curdir, '.eggs')
         if not os.path.exists(egg_cache_dir):
             os.mkdir(egg_cache_dir)
-            windows_support.hide_file(egg_cache_dir)
             readme_txt_filename = os.path.join(egg_cache_dir, 'README.txt')
             with open(readme_txt_filename, 'w') as f:
                 f.write('This directory contains eggs that were downloaded '
