diff --git a/setuptools/command/easy_install.py b/setuptools/command/easy_install.py
index e8b90c7..8598c44 100755
--- a/setuptools/command/easy_install.py
+++ b/setuptools/command/easy_install.py
@@ -1946,6 +1946,8 @@ class CommandSpec(list):
         Construct a CommandSpec from a parameter to build_scripts, which may
         be None.
         """
+        if os.environ.get('__PYVENV_LAUNCHER__'):
+            return cls.from_environment()
         if isinstance(param, cls):
             return param
         if isinstance(param, list):
