Index: shorewall-core-5.1.7.2/lib.common
===================================================================
--- shorewall-core-5.1.7.2.orig/lib.common	2017-10-11 08:01:06.191862789 +0200
+++ shorewall-core-5.1.7.2/lib.common	2017-10-11 08:01:06.187863006 +0200
@@ -345,7 +345,7 @@
 
     if [ -z "$MODULESDIR" ]; then
 	uname=$(uname -r)
-	MODULESDIR=/lib/modules/$uname/kernel/net/ipv${g_family}/netfilter:/lib/modules/$uname/kernel/net/netfilter:/lib/modules/$uname/kernel/net/sched:/lib/modules/$uname/extra:/lib/modules/$uname/extra/ipset
+	MODULESDIR=/lib/modules/$uname:/lib/modules/$uname/kernel/net/ipv${g_family}/netfilter:/lib/modules/$uname/kernel/net/netfilter:/lib/modules/$uname/kernel/net/sched:/lib/modules/$uname/extra:/lib/modules/$uname/extra/ipset
 	if [ -n "$extras" ]; then
 	    for directory in $(split "$extras"); do
 		MODULESDIR="$MODULESDIR:/lib/modules/$uname/$directory"
@@ -399,7 +399,7 @@
 
     if [ -z "$MODULESDIR" ]; then
 	uname=$(uname -r)
-	MODULESDIR=/lib/modules/$uname/kernel/net/ipv${g_family}/netfilter:/lib/modules/$uname/kernel/net/netfilter:/lib/modules/$uname/kernel/net/sched:/lib/modules/$uname/extra:/lib/modules/$uname/extra/ipset
+	MODULESDIR=/lib/modules/$uname:/lib/modules/$uname/kernel/net/ipv${g_family}/netfilter:/lib/modules/$uname/kernel/net/netfilter:/lib/modules/$uname/kernel/net/sched:/lib/modules/$uname/extra:/lib/modules/$uname/extra/ipset
 	if [ -n "$extras" ]; then
 	    for directory in $(split "$extras"); do
 		MODULESDIR="$MODULESDIR:/lib/modules/$uname/$directory"
