From 53b9bbe603b768259eaef804539f3b79d4fc3df0 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@gnutls.org>
Date: Sun, 15 Feb 2015 08:28:08 +0100
Subject: [PATCH] configure: use seccomp where it is available

---
 configure.ac | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3814aa4..e0d32a8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -283,9 +283,6 @@ AC_ARG_ENABLE(seccomp,
 
 if [ test "$seccomp_enabled" = "yes" ];then
 AC_LIB_HAVE_LINKFLAGS(seccomp,, [#include <seccomp.h>
-#if !defined(__i386__) && !defined(__arm__) && !defined(__x86_64__)
-# error 1
-#endif
 ], [seccomp_init(0);])
  if [ test -z "$LIBSECCOMP" ];then
 	seccomp_enabled="no"
-- 
2.1.3

