diff --git a/src/Makefile.am b/src/Makefile.am
index 5943a50..4b2c77c 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -23,18 +23,7 @@ noinst_LIBRARIES = libcmd-ocserv.a libipc.a
 libcmd_ocserv_a_SOURCES = ocserv-args.def ocserv-args.c ocserv-args.h
 
 ocserv-args.c: $(srcdir)/ocserv-args.def $(builddir)/version.inc
-	if test "$(AUTOGEN)" = ":";then \
-		rm -f $(builddir)/ocserv-args.c; \
-		rm -f $(builddir)/ocserv-args.h; \
-		cp $(srcdir)/autogen/ocserv-args.c $(builddir)/; \
-		cp $(srcdir)/autogen/ocserv-args.h $(builddir)/; \
-	else \
-		$(AUTOGEN) $<; \
-		if test -d $(srcdir)/autogen;then \
-			cp $(builddir)/ocserv-args.c $(srcdir)/autogen; \
-			cp $(builddir)/ocserv-args.h $(srcdir)/autogen; \
-		fi; \
-	fi
+	$(AUTOGEN) $<
 ocserv-args.h: ocserv-args.c
 
 # Authentication module sources
diff --git a/src/ocpasswd/Makefile.am b/src/ocpasswd/Makefile.am
index 97e2491..6304607 100644
--- a/src/ocpasswd/Makefile.am
+++ b/src/ocpasswd/Makefile.am
@@ -21,17 +21,6 @@ ocpasswd_LDADD += $(LIBGNUTLS_LIBS) $(LIBCRYPT) $(CODE_COVERAGE_LDFLAGS) \
 	$(LIBNETTLE_LIBS)
 
 args.c: $(srcdir)/args.def $(builddir)/../version.inc
-	if test "$(AUTOGEN)" = ":";then \
-		rm -f $(builddir)/args.c; \
-		rm -f $(builddir)/args.h; \
-		cp $(srcdir)/../autogen/ocpasswd-args.c $(builddir)/args.c; \
-		cp $(srcdir)/../autogen/ocpasswd-args.h $(builddir)/args.h; \
-	else \
-		$(AUTOGEN) $<; \
-		if test -d $(srcdir)/autogen;then \
-			cp $(builddir)/args.c $(srcdir)/../autogen/ocpasswd-args.c; \
-			cp $(builddir)/args.h $(srcdir)/../autogen/ocpasswd-args.h; \
-		fi; \
-	fi
+	$(AUTOGEN) $<
 args.h: args.c
 
