diff -u --recursive gst-plugins-good-1.12.4-vanilla/configure.ac gst-plugins-good-1.12.4/configure.ac
--- gst-plugins-good-1.12.4-vanilla/configure.ac	2018-02-11 10:34:37.773144289 -0500
+++ gst-plugins-good-1.12.4/configure.ac	2018-02-11 10:35:17.103223474 -0500
@@ -1061,7 +1061,6 @@
 sys/v4l2/Makefile
 sys/waveform/Makefile
 sys/ximage/Makefile
-po/Makefile.in
 tests/Makefile
 tests/check/Makefile
 tests/examples/Makefile
diff -u --recursive gst-plugins-good-1.12.4-vanilla/Makefile.am gst-plugins-good-1.12.4/Makefile.am
--- gst-plugins-good-1.12.4-vanilla/Makefile.am	2018-02-11 10:34:37.786144315 -0500
+++ gst-plugins-good-1.12.4/Makefile.am	2018-02-11 10:35:23.529236411 -0500
@@ -4,7 +4,6 @@
 	gst sys ext 		\
 	tests			\
 	docs			\
-	po			\
 	common			\
 	m4			\
 	pkgconfig
