diff -u --recursive gst-plugins-good-1.6.2-vanilla/configure.ac gst-plugins-good-1.6.2/configure.ac
--- gst-plugins-good-1.6.2-vanilla/configure.ac	2016-01-01 10:40:17.994764882 -0500
+++ gst-plugins-good-1.6.2/configure.ac	2016-01-01 10:45:25.668599823 -0500
@@ -1014,7 +1014,6 @@
 sys/v4l2/Makefile
 sys/waveform/Makefile
 sys/ximage/Makefile
-po/Makefile.in
 tests/Makefile
 tests/check/Makefile
 tests/examples/Makefile
diff -u --recursive gst-plugins-good-1.6.2-vanilla/Makefile.am gst-plugins-good-1.6.2/Makefile.am
--- gst-plugins-good-1.6.2-vanilla/Makefile.am	2016-01-01 10:40:17.994764882 -0500
+++ gst-plugins-good-1.6.2/Makefile.am	2016-01-01 10:45:19.923089598 -0500
@@ -4,7 +4,6 @@
 	gst sys ext 		\
 	tests			\
 	docs			\
-	po			\
 	common			\
 	m4			\
 	pkgconfig
