--- a/configure.ac
+++ b/configure.ac
@@ -1049,21 +1049,6 @@ sys/sunaudio/Makefile
 sys/v4l2/Makefile
 sys/waveform/Makefile
 sys/ximage/Makefile
-tests/Makefile
-tests/check/Makefile
-tests/examples/Makefile
-tests/examples/audiofx/Makefile
-tests/examples/cairo/Makefile
-tests/examples/equalizer/Makefile
-tests/examples/jack/Makefile
-tests/examples/level/Makefile
-tests/examples/pulse/Makefile
-tests/examples/rtp/Makefile
-tests/examples/shapewipe/Makefile
-tests/examples/spectrum/Makefile
-tests/examples/v4l2/Makefile
-tests/files/Makefile
-tests/icles/Makefile
 common/Makefile
 common/m4/Makefile
 m4/Makefile
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,7 +2,6 @@ DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-d
 
 ALWAYS_SUBDIRS =		\
 	gst sys ext 		\
-	tests			\
 	docs			\
 	common			\
 	m4			\
