--- a/configure.ac
+++ b/configure.ac
@@ -1083,7 +1083,6 @@ sys/sunaudio/Makefile
 sys/v4l2/Makefile
 sys/waveform/Makefile
 sys/ximage/Makefile
-po/Makefile.in
 tests/Makefile
 tests/check/Makefile
 tests/examples/Makefile
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,7 +2,6 @@ DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-d
 
 ALWAYS_SUBDIRS =		\
 	gst sys ext 		\
-	po			\
 	common			\
 	m4			\
 	pkgconfig
--- a/Makefile.in
+++ b/Makefile.in
@@ -570,7 +570,6 @@ top_srcdir = @top_srcdir@
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 ALWAYS_SUBDIRS = \
 	gst sys ext 		\
-	po			\
 	common			\
 	m4			\
 	pkgconfig
