diff -u --recursive gst-plugins-good-1.0.7-vanilla/configure.ac gst-plugins-good-1.0.7/configure.ac
--- gst-plugins-good-1.0.7-vanilla/configure.ac	2013-06-01 19:56:23.004218467 -0500
+++ gst-plugins-good-1.0.7/configure.ac	2013-06-01 19:56:52.919033496 -0500
@@ -1157,7 +1157,6 @@
 sys/v4l2/Makefile
 sys/waveform/Makefile
 sys/ximage/Makefile
-po/Makefile.in
 tests/Makefile
 tests/check/Makefile
 tests/examples/Makefile
diff -u --recursive gst-plugins-good-1.0.7-vanilla/Makefile.am gst-plugins-good-1.0.7/Makefile.am
--- gst-plugins-good-1.0.7-vanilla/Makefile.am	2013-06-01 19:56:23.043216922 -0500
+++ gst-plugins-good-1.0.7/Makefile.am	2013-06-01 19:56:43.671399777 -0500
@@ -2,7 +2,6 @@
 
 ALWAYS_SUBDIRS =		\
 	gst sys ext 		\
-	po			\
 	common			\
 	m4			\
 	pkgconfig
diff -u --recursive gst-plugins-good-1.0.7-vanilla/Makefile.in gst-plugins-good-1.0.7/Makefile.in
--- gst-plugins-good-1.0.7-vanilla/Makefile.in	2013-06-01 19:56:23.043216922 -0500
+++ gst-plugins-good-1.0.7/Makefile.in	2013-06-01 19:56:47.703240080 -0500
@@ -522,7 +522,6 @@
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 ALWAYS_SUBDIRS = \
 	gst sys ext 		\
-	po			\
 	common			\
 	m4			\
 	pkgconfig
