--- a/Makefile.am
+++ b/Makefile.am
@@ -2,7 +2,6 @@ DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
 ALWAYS_SUBDIRS =		\
 	gst sys ext 		\
-	docs			\
 	common			\
 	m4			\
 	pkgconfig
--- a/configure.ac
+++ b/configure.ac
@@ -1295,9 +1295,6 @@ sys/ximage/Makefile
 common/Makefile
 common/m4/Makefile
 m4/Makefile
-docs/Makefile
-docs/plugins/Makefile
-docs/version.entities
 pkgconfig/Makefile
 pkgconfig/gstreamer-plugins-good-uninstalled.pc
 )
