diff --git a/Makefile.am b/Makefile.am
index 14bf91d..dfa6d07 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,5 +1,3 @@
-DISTCHECK_CONFIGURE_FLAGS = --with-systemdsystemunitdir=$$dc_install_base/$(systemdsystemunitdir)
-
 AM_CFLAGS = @TANG_CFLAGS@ @jose_CFLAGS@
 LDADD = @jose_LIBS@ @http_parser_LIBS@
 
@@ -7,21 +5,11 @@ nagiosdir = $(libdir)/nagios/plugins
 cachedir = $(localstatedir)/cache/$(PACKAGE_NAME)
 jwkdir = $(localstatedir)/db/$(PACKAGE_NAME)
 
-nodist_systemdsystemunit_DATA = \
-    units/tangd@.service \
-    units/tangd.socket \
-    units/tangd-update.path \
-    units/tangd-update.service \
-    units/tangd-keygen.service
-
 dist_libexec_SCRIPTS = src/tangd-update src/tangd-keygen
 libexec_PROGRAMS = src/tangd
-nagios_PROGRAMS = src/tang
-man1_MANS = doc/tang-nagios.1
 man8_MANS = doc/tang.8
 
 src_tangd_SOURCES = src/http.c src/http.h src/tangd.c
-src_tang_SOURCES = src/nagios.c
 
 %: %.in
 	$(AM_V_GEN)mkdir -p "`dirname "$@"`"
@@ -32,11 +20,9 @@ src_tang_SOURCES = src/nagios.c
 		$(srcdir)/$@.in > $@
 
 AM_TESTS_ENVIRONMENT = SD_ACTIVATE="@SD_ACTIVATE@" PATH=$(srcdir)/src:$(builddir)/src:$(PATH)
-TESTS = tests/adv tests/rec tests/nagios
+TESTS = tests/adv tests/rec 
 
-CLEANFILES = $(nodist_systemdsystemunit_DATA)
 EXTRA_DIST = \
-    $(foreach unit,$(nodist_systemdsystemunit_DATA),$(unit).in) \
     COPYING \
     $(TESTS) \
     $(man1_MANS) \
