--- a/config/gdlib-config.in
+++ b/config/gdlib-config.in
@@ -71,7 +71,7 @@ while test $# -gt 0; do
 	echo @LDFLAGS@
 	;;
     --libs)
-	echo -lgd @LIBS@ @LIBICONV@
+	echo -lgd @LIBS@
 	;;
     --cflags|--includes)
 	echo -I@includedir@
@@ -84,7 +84,7 @@ while test $# -gt 0; do
 	echo "includedir: $includedir"
 	echo "cflags:     -I@includedir@"
 	echo "ldflags:    @LDFLAGS@"
-	echo "libs:       @LIBS@ @LIBICONV@"
+	echo "libs:       @LIBS@"
 	echo "libdir:     $libdir"
 	echo "features:   @FEATURES@"
 	;;
