diff -urN bmxd-r1075/Makefile bmxd-r1075.new/Makefile
--- bmxd-r1075/Makefile	2008-08-04 20:10:56.000000000 +0200
+++ bmxd-r1075.new/Makefile	2009-07-14 02:28:16.000000000 +0200
@@ -19,11 +19,15 @@
 
 # CC =		gcc
 
-CFLAGS =	-Wall -O1 -g -pg
+# JoW: "-pg" results in "gcrt1.o: No such file" ?!
+#CFLAGS =	-Wall -O1 -g -pg
+CFLAGS =	-Wall -O1 -g
 
 #-DDEBUG_MALLOC -DMEMORY_USAGE -DPROFILE_DATA
 
-LDFLAGS =	-lpthread -static -g -pg
+# JoW: "-pg" results in "gcrt1.o: No such file" ?!
+#LDFLAGS =	-lpthread -static -g -pg
+LDFLAGS =	-lpthread -g
 
 
 #STRIP=			strip
@@ -80,4 +84,4 @@
 		rm -f $(BINARY_NAME) *.o
 
 clean-all:
-		rm -rf $(PACKAGE_NAME)_* dl/*
\ No newline at end of file
+		rm -rf $(PACKAGE_NAME)_* dl/*
