--- a/isisd/isis_lsp.c
+++ b/isisd/isis_lsp.c
@@ -2413,7 +2413,7 @@ top_lsp_refresh (struct thread *thread)
   isis_dynhn_insert (lsp->lsp_header->lsp_id, lsp->tlv_data.hostname,
 		     IS_LEVEL_1);
 
-  lsp->lsp_header->lsp_bits = lsp_bits_generate (level,
+  lsp->lsp_header->lsp_bits = lsp_bits_generate (lsp->level,
                                                  lsp->area->overload_bit);
   rem_lifetime = lsp_rem_lifetime (lsp->area, IS_LEVEL_1);
   lsp->lsp_header->rem_lifetime = htons (rem_lifetime);
