--- a/src/linux/net.c
+++ b/src/linux/net.c
@@ -107,6 +107,7 @@ bind_socket_to_device(int sock, char *de
 int
 enable_ip_forwarding(int version)
 {
+#if 0
   FILE *proc_fwd;
   const char *const procfile = version == AF_INET ? "/proc/sys/net/ipv4/ip_forward" : "/proc/sys/net/ipv6/conf/all/forwarding";
 
@@ -142,6 +143,7 @@ enable_ip_forwarding(int version)
     }
     fclose(proc_fwd);
   }
+#endif
   return 1;
 }
 
@@ -268,6 +270,7 @@ restore_settings(int version)
   OLSR_PRINTF(1, "Restoring network state\n");
 
   /* Restore IP forwarding to "off" */
+#if 0
   if (orig_fwd_state == '0') {
     const char *const procfile = version == AF_INET ? "/proc/sys/net/ipv4/ip_forward" : "/proc/sys/net/ipv6/conf/all/forwarding";
     FILE *proc_fd;
@@ -280,6 +283,7 @@ restore_settings(int version)
       fclose(proc_fd);
     }
   }
+#endif
 
   /* Restore global ICMP redirect setting */
   if (orig_global_redirect_state != '0') {
