--- a/src/linux/net.c
+++ b/src/linux/net.c
@@ -174,11 +174,13 @@ static bool is_at_least_linuxkernel_2_6_
  */
 void
 net_os_set_global_ifoptions(void) {
+#if 0
   if (writeToProc(olsr_cnf->ip_version == AF_INET ? PROC_IPFORWARD_V4 : PROC_IPFORWARD_V6, &orig_fwd_state, '1')) {
     OLSR_PRINTF(1, "Warning, could not enable IP forwarding!\n"
         "you should manually ensure that IP forwarding is enabled!\n\n");
     olsr_startup_sleep(3);
   }
+#endif
 
   if (olsr_cnf->smart_gw_active) {
     char procfile[FILENAME_MAX];
@@ -273,10 +275,12 @@ net_os_restore_ifoptions(void)
 
   OLSR_PRINTF(1, "Restoring network state\n");
 
+#if 0
   /* Restore IP forwarding to "off" */
   if (writeToProc(olsr_cnf->ip_version == AF_INET ? PROC_IPFORWARD_V4 : PROC_IPFORWARD_V6, NULL, orig_fwd_state)) {
     OLSR_PRINTF(1, "Error, could not restore ip_forward settings\n");
   }
+#endif
 
   if (olsr_cnf->smart_gw_active && (olsr_cnf->ip_version == AF_INET || olsr_cnf->use_niit)) {
     /* Generate the procfile name */
