--- a/src/linux/net.c
+++ b/src/linux/net.c
@@ -115,6 +115,7 @@
 int
 enable_ip_forwarding(int version)
 {
+#if 0
   FILE *proc_fwd;
   const char *const procfile = version == AF_INET ? "/proc/sys/net/ipv4/ip_forward" : "/proc/sys/net/ipv6/conf/all/forwarding";
 
@@ -150,6 +151,7 @@
     }
     fclose(proc_fwd);
   }
+#endif
   return 1;
 }
 
@@ -276,6 +278,7 @@
   OLSR_PRINTF(1, "Restoring network state\n");
 
   /* Restore IP forwarding to "off" */
+#if 0
   if (orig_fwd_state == '0') {
     const char *const procfile = version == AF_INET ? "/proc/sys/net/ipv4/ip_forward" : "/proc/sys/net/ipv6/conf/all/forwarding";
     FILE *proc_fd;
@@ -288,6 +291,7 @@
       fclose(proc_fd);
     }
   }
+#endif
 
   /* Restore global ICMP redirect setting */
   if (orig_global_redirect_state != '0') {
