Index: batmand_0.2-rv478_sources/Makefile
===================================================================
--- batmand_0.2-rv478_sources.orig/Makefile	2007-07-31 14:26:18.000000000 -0500
+++ batmand_0.2-rv478_sources/Makefile	2007-07-31 14:28:43.000000000 -0500
@@ -21,6 +21,7 @@
 CFLAGS =		-Wall -O1 -g3 -DDEBUG_MALLOC -DMEMORY_USAGE -DPROFILE_DATA
 STRIP=			strip
 LDFLAGS =		-lpthread -static
+SBINDIR =		$(INSTALL_PREFIX)/usr/sbin
 
 CFLAGS_MIPS =	-Wall -O1 -g3
 LDFLAGS_MIPS =	-lpthread
@@ -264,3 +265,7 @@
 
 clean-long:
 		rm -rf batmand_*
+
+install:
+		mkdir -p $(SBINDIR)
+		install -m 0755 batmand $(SBINDIR)
