diff -ruN batman-III-0.2.0a.old/linux-specific.c batman-III-0.2.0a/linux-specific.c
--- batman-III-0.2.0a.old/linux-specific.c	2007-02-20 21:07:29.000000000 +0100
+++ batman-III-0.2.0a/linux-specific.c	2007-03-19 15:34:36.000000000 +0100
@@ -111,9 +111,9 @@
 
 	int32_t fd;
 
-	if ( ( fd = open( "/dev/net/tun", O_RDWR ) ) < 0 ) {
+	if ( ( fd = open( "/dev/tun", O_RDWR ) ) < 0 ) {
 
-		debug_output( 0, "Error - could not open '/dev/net/tun' ! Is the tun kernel module loaded ?\n" );
+		debug_output( 0, "Error - could not open '/dev/tun' ! Is the tun kernel module loaded ?\n" );
 		return 0;
 
 	}
@@ -151,9 +151,9 @@
 	memset( &ifr_if, 0, sizeof(ifr_if) );
 	ifr_tun.ifr_flags = IFF_TUN | IFF_NO_PI;
 
-	if ( ( *fd = open( "/dev/net/tun", O_RDWR ) ) < 0 ) {
+	if ( ( *fd = open( "/dev/tun", O_RDWR ) ) < 0 ) {
 
-		debug_output( 0, "Error - can't create tun device (/dev/net/tun): %s\n", strerror(errno) );
+		debug_output( 0, "Error - can't create tun device (/dev/tun): %s\n", strerror(errno) );
 		return -1;
 
 	}
