Index: olsrd-0.5.6-rc7/Makefile
===================================================================
--- olsrd-0.5.6-rc7/Makefile.orig	2008-07-10 01:42:40.000000000 +0200
+++ olsrd-0.5.6-rc7/Makefile	2008-07-10 01:43:24.000000000 +0200
@@ -72,7 +72,7 @@
 	$(MAKECMD) -C $(SWITCHDIR)
 
 # generate it always
-.PHONY: src/builddata.c
+#.PHONY: src/builddata.c
 src/builddata.c:
 	@$(RM) "$@"
 	@echo "#include \"defs.h\"" >> "$@" 
