--- olsrd-0.5.4/lib/dyn_gw/src/olsrd_dyn_gw-orig.c      2007-10-31 15:14:47.000000000 +0100
+++ olsrd-0.5.4/lib/dyn_gw/src/olsrd_dyn_gw.c   2007-10-31 15:15:17.000000000 +0100
@@ -92,7 +92,7 @@


 /* set default interval, in case none is given in the config file */
-static int check_interval = 5;
+static double check_interval = 5;

 /* list to store the Ping IP addresses given in the config file */
 struct ping_list {
@@ -307,7 +307,7 @@
     struct hna_list *li;
     struct timespec remainder_spec;
     /* the time to wait in "Interval" sec (see connfig), default=5sec */
-    struct timespec sleeptime_spec  = { check_interval, 0L };
+    struct timespec sleeptime_spec  = { (int) check_interval, 0L };

     for(li = the_hna_list; li; li = li->next){
       /* check for gw in table entry and if Ping IPs are given also do pings */
