--- a/lib/txtinfo/src/olsrd_txtinfo.c
+++ b/lib/txtinfo/src/olsrd_txtinfo.c
@@ -562,7 +562,7 @@ ipc_print_gateway(struct autobuf *abuf)
   struct lqtextbuffer lqbuf;
 
   // Status IP ETX Hopcount Uplink-Speed Downlink-Speed ipv4/ipv4-nat/- ipv6/- ipv6-prefix/-
-  abuf_puts(abuf, "Table: Gateways\n   Gateway\tETX\tHopcnt\tUplink\tDownlnk\tIPv4\tIPv6\tPrefix\n");
+  abuf_puts(abuf, "Table: Gateways\nGateway\tETX\tHopcnt\tUplink\tDownlnk\tIPv4\tIPv6\tPrefix\n");
   OLSR_FOR_ALL_GATEWAY_ENTRIES(gw) {
     char v4 = '-', v6 = '-';
     bool autoV4 = false, autoV6 = false;
@@ -595,7 +595,7 @@ ipc_print_gateway(struct autobuf *abuf)
       v6type = IPV6;
     }
 
-    abuf_appendf(abuf, "%c%c %s\t%s\t%d\t%u\t%u\t%s\t%s\t%s\n",
+    abuf_appendf(abuf, "%c%c\t%s\t%s\t%d\t%u\t%u\t%s\t%s\t%s\n",
         v4, v6, olsr_ip_to_string(&buf, &gw->originator),
         get_linkcost_text(tc->path_cost, true, &lqbuf), tc->hops,
         gw->uplink, gw->downlink, v4type, v6type,
