Index: dosfstools-3.0.28/src/mkfs.fat.c
===================================================================
--- dosfstools-3.0.28.orig/src/mkfs.fat.c
+++ dosfstools-3.0.28/src/mkfs.fat.c
@@ -1412,7 +1412,7 @@ int main(int argc, char **argv)
 
     gettimeofday(&create_timeval, NULL);
     create_time = create_timeval.tv_sec;
-    volume_id = (u_int32_t) ((create_timeval.tv_sec << 20) | create_timeval.tv_usec);	/* Default volume ID = creation time, fudged for more uniqueness */
+    volume_id = (uint32_t) ((create_timeval.tv_sec << 20) | create_timeval.tv_usec);	/* Default volume ID = creation time, fudged for more uniqueness */
     check_atari();
 
     printf("mkfs.fat " VERSION " (" VERSION_DATE ")\n");
