--- a/src/system.h
+++ b/src/system.h
@@ -83,10 +83,6 @@
 #include <linux/sysinfo.h>
 #endif
 
-#ifdef HAVE_SYS_SYSINFO_H
-#include <sys/sysinfo.h>
-#endif
-
 #ifdef HAVE_TIME_H
 #include <time.h>
 #endif
@@ -139,6 +135,10 @@
 #include <linux/un.h>
 #endif
 
+#ifdef HAVE_SYS_SYSINFO_H
+#include <sys/sysinfo.h>
+#endif
+
 #elif defined (__FreeBSD__)  || defined (__APPLE__) || defined (__OpenBSD__) || defined (__NetBSD__) 
 #include <net/if.h>
 #include <net/bpf.h>
