--- a/configure.in
+++ b/configure.in
@@ -363,7 +363,7 @@ AC_ARG_WITH([cyassl],
  [AS_HELP_STRING([--with-cyassl], [enable support for cyassl])],[],[with_cyassl=no])
 
 AS_IF([test x"$with_cyassl" != xno],
-  [AC_CHECK_LIB([cyassl], [CyaSSL_Init],
+  [AC_CHECK_LIB([cyassl], [wolfSSL_Init],
               [AC_SUBST([LIBSSL], ["-lcyassl"])
                AC_DEFINE([HAVE_CYASSL], [1],
                          [Define if you have cyassl])
--- a/src/ippool.c
+++ b/src/ippool.c
@@ -20,6 +20,8 @@
 
 #include "chilli.h"
 
+#undef USED
+
 /*#define _DEBUG_PRINT_ 1*/
 
 const unsigned int IPPOOL_STATSIZE = 0x10000;
--- a/src/md5.c
+++ b/src/md5.c
@@ -18,7 +18,7 @@
 #include <string.h>		/* for memcpy() */
 #include "md5.h"
 
-#ifndef HAVE_OPENSSL
+#if !defined(HAVE_OPENSSL) && !defined(HAVE_CYASSL)
 
 void byteReverse(unsigned char *buf, size_t longs);
 
--- a/src/md5.h
+++ b/src/md5.h
@@ -28,6 +28,16 @@
 #define MD5Update MD5_Update
 #define MD5Final MD5_Final
 
+#elif HAVE_CYASSL
+
+#include <cyassl/openssl/md5.h>
+
+#define MD5Init MD5_Init
+#define MD5Update MD5_Update
+#define MD5Final MD5_Final
+
+#define MD5Context MD5_CTX
+
 #else
 
 struct MD5Context {
--- a/src/ssl.h
+++ b/src/ssl.h
@@ -47,6 +47,8 @@ typedef struct {
 #include <time.h>
 #include <string.h>
 
+#define OPENSSL_EXTRA
+
 #include <cyassl/ssl.h>
 #include <cyassl/openssl/bio.h>
 #include <cyassl/openssl/crypto.h>
