--- a/Makefile
+++ b/Makefile
@@ -113,10 +113,10 @@ BINDIR?=$(DESTDIR)${PREFIX}/bin
 MANDIR?=$(DESTDIR)${PREFIX}/share/man
 
 
-all: $(EXEC) pfc $(CONTROL_EXEC)
+all: $(EXEC) $(CONTROL_EXEC)
 
 clean:
-	rm -f $(OBJS) $(EXEC) pfc.o pfc $(CONTROL_EXEC)
+	rm -f $(OBJS) $(EXEC) $(CONTROL_EXEC)
 
 $(EXEC): $(OBJS) $(HDRS)
 	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LDLIBS)
@@ -124,14 +124,10 @@ $(EXEC): $(OBJS) $(HDRS)
 $(CONTROL_EXEC): $(CONTROL_SRCS)
 	$(CC) $(CFLAGS) $(LDFLAGS) $(CONTROL_SRCS) -o $@
 
-pfc:
-	$(CC) $(CFLAGS) -c contrib/pfc.c
-	$(CC) $(LDFLAGS) -o pfc pfc.o -lpcap $(LDLIBS)
-
 romfs:
 	$(ROMFSINST) /bin/$(EXEC)
 
-install: ${EXEC} pfc ${CONTROL_EXEC}
+install: ${EXEC} ${CONTROL_EXEC}
 	install -d -m 0755 ${SBINDIR}
 	install -m 0755 $(EXEC) ${SBINDIR}/$(EXEC)
 	install -d -m 0755 ${MANDIR}/man5
@@ -139,11 +135,6 @@ install: ${EXEC} pfc ${CONTROL_EXEC}
 	install -m 0644 doc/xl2tpd.8 ${MANDIR}/man8/
 	install -m 0644 doc/xl2tpd.conf.5 doc/l2tp-secrets.5 \
 		 ${MANDIR}/man5/
-	# pfc
-	install -d -m 0755 ${BINDIR}
-	install -m 0755 pfc ${BINDIR}/pfc
-	install -d -m 0755 ${MANDIR}/man1
-	install -m 0644 contrib/pfc.1 ${MANDIR}/man1/
 	# control exec
 	install -d -m 0755 ${SBINDIR}
 	install -m 0755 $(CONTROL_EXEC) ${SBINDIR}/$(CONTROL_EXEC)
