From efa0a18b5dfd827792b07acdcb35101229ccf612 Mon Sep 17 00:00:00 2001
From: Yousong Zhou <yszhou4tech@gmail.com>
Date: Wed, 29 Apr 2015 10:32:37 +0800
Subject: [PATCH 201/210] xl2tpd-control: define _GNU_SOURCE to use fmemopen()
 and friends.

---
 xl2tpd-control.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/xl2tpd-control.c b/xl2tpd-control.c
index e5904d8..9b6235d 100644
--- a/xl2tpd-control.c
+++ b/xl2tpd-control.c
@@ -10,6 +10,8 @@
  *
  */
  
+#define _GNU_SOURCE
+
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
-- 
1.7.10.4

