From 1c148f6645f43bf5abd8a9b8f0708a598b357d97 Mon Sep 17 00:00:00 2001
From: Yousong Zhou <yszhou4tech@gmail.com>
Date: Thu, 14 May 2015 19:56:39 +0800
Subject: [PATCH 205/210] xl2tpd: fix typo in reporting available lns count.

---
 xl2tpd.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xl2tpd.c b/xl2tpd.c
index c9b7579..6c945fc 100644
--- a/xl2tpd.c
+++ b/xl2tpd.c
@@ -1039,7 +1039,7 @@ int control_handle_available(FILE* resf, char* bufp){
         lns_count++;
     }                                               
 
-    write_res (resf, "%02i AVAILABLE lns.cout=%d\n", 0, lns_count);
+    write_res (resf, "%02i AVAILABLE lns.count=%d\n", 0, lns_count);
 
     lac  = laclist;
     int lac_count = 0;
@@ -1053,7 +1053,7 @@ int control_handle_available(FILE* resf, char* bufp){
     if(deflac){
         write_res (resf, "%02i AVAILABLE lac.%d.name=%s\n", 0, lac_count, deflac->entname);
         lac_count++;
-    }                                               
+    }
 
     write_res (resf, "%02i AVAILABLE lac.count=%d\n", 0, lac_count);
     return 1;
-- 
1.7.10.4

