From 44ced2bbf1d6b39bb36c3cb8be6e40c8764e2e8a Mon Sep 17 00:00:00 2001
From: Yousong Zhou <yszhou4tech@gmail.com>
Date: Thu, 30 Apr 2015 13:57:36 +0800
Subject: [PATCH 7/7] l2tp_log: remove log prefix that will duplicate with
 procd.

---
 misc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/misc.c
+++ b/misc.c
@@ -61,7 +61,7 @@ void l2tp_log (int level, const char *fm
 	init_log();
 	SYSLOG_CALL( syslog (level, "%s", buf) );
     } else {
-	fprintf(stderr, "xl2tpd[%d]: %s", getpid(), buf);
+	fprintf(stderr, "%s", buf);
     }
 }
 
