--- a/folly/CachelinePadded.h
+++ b/folly/CachelinePadded.h
@@ -35,10 +35,6 @@ namespace folly {
  */
 template <typename T>
 class CachelinePadded {
-  static_assert(
-      alignof(T) <= max_align_v,
-      "CachelinePadded does not support over-aligned types.");
-
  public:
   template <typename... Args>
   explicit CachelinePadded(Args&&... args)
--- a/folly/external/farmhash/farmhash.cpp
+++ b/folly/external/farmhash/farmhash.cpp
@@ -181,6 +181,7 @@
 
 #undef bswap_32
 #undef bswap_64
+#undef _BYTESWAP_H
 #include <byteswap.h>
 
 #endif
