--- a/zip.c
+++ b/zip.c
@@ -1028,8 +1028,7 @@ local void help_extended()
 
   for (i = 0; i < sizeof(text)/sizeof(char *); i++)
   {
-    printf(text[i]);
-    putchar('\n');
+    puts(text[i]);
   }
 #ifdef DOS
   check_for_windows("Zip");
@@ -1225,8 +1224,7 @@ local void version_info()
             CR_MAJORVER, CR_MINORVER, CR_BETA_VER, CR_VERSION_DATE);
   for (i = 0; i < sizeof(cryptnote)/sizeof(char *); i++)
   {
-    printf(cryptnote[i]);
-    putchar('\n');
+    puts(cryptnote[i]);
   }
   ++i;  /* crypt support means there IS at least one compilation option */
 #endif /* CRYPT */
