diff -ur portmap_6.0.org/Makefile portmap_6.0/Makefile
--- portmap_6.0.org/Makefile	2008-04-19 22:44:35.000000000 +0200
+++ portmap_6.0/Makefile	2008-04-19 00:02:01.000000000 +0200
@@ -127,7 +127,7 @@
 CPPFLAGS += $(HOSTS_ACCESS)
 portmap: CFLAGS   += -fpie
 portmap: LDLIBS   += $(WRAP_LIB)
-portmap: LDFLAGS  += -pie
+#portmap: LDFLAGS  += -pie
 portmap: portmap.o pmap_check.o from_local.o
 
 from_local: CPPFLAGS += -DTEST
