--- a/src/ptunnel.h
+++ b/src/ptunnel.h
@@ -45,9 +45,6 @@
 #define PING_TUNNEL_H 1
 
 #ifndef WIN32
-#ifdef HAVE_ICMPFILTER
-#include <linux/icmp.h>
-#endif
 #ifdef HAVE_SYS_UNISTD_H
 #include <sys/unistd.h>
 #endif
@@ -56,6 +53,9 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <netdb.h>
+#ifdef HAVE_ICMPFILTER
+#include <linux/icmp.h>
+#endif
 #include <pthread.h>
 #include <errno.h>
 #include <net/ethernet.h>
