diff -u --recursive krb5-1.12.1-vanilla/src/lib/krad/packet.c krb5-1.12.1/src/lib/krad/packet.c
--- krb5-1.12.1-vanilla/src/lib/krad/packet.c	2014-07-18 00:10:11.916603406 -0400
+++ krb5-1.12.1/src/lib/krad/packet.c	2014-07-18 00:20:23.471574387 -0400
@@ -253,7 +253,7 @@
 {
     krb5_error_code retval;
     krad_packet *pkt;
-    uchar id;
+    uchar id = 0;
     size_t attrset_len;
 
     pkt = packet_new();
