diff -ur krb5-1.11-vanilla/src/lib/krb5/krb/preauth2.c krb5-1.11/src/lib/krb5/krb/preauth2.c
--- krb5-1.11-vanilla/src/lib/krb5/krb/preauth2.c	2012-12-18 03:47:05.000000000 +0100
+++ krb5-1.11/src/lib/krb5/krb/preauth2.c	2013-02-18 03:53:20.580840173 +0100
@@ -956,7 +956,7 @@
     size_t i, h;
     int out_pa_list_size = 0;
     krb5_pa_data **out_pa_list = NULL;
-    krb5_error_code ret, module_ret;
+    krb5_error_code ret, module_ret = 0;
     krb5_responder_fn responder = opte->opt_private->responder;
     static const int paorder[] = { PA_INFO, PA_REAL };
 
