--- a/src/tini.c
+++ b/src/tini.c
@@ -18,6 +18,8 @@
 #include "tiniConfig.h"
 #include "tiniLicense.h"
 
+#define basename(name) (strrchr((name),'/') ? strrchr((name),'/')+1 : (name))
+
 #if TINI_MINIMAL
 #define PRINT_FATAL(...)                         fprintf(stderr, __VA_ARGS__); fprintf(stderr, "\n");
 #define PRINT_WARNING(...)  if (verbosity > 0) { fprintf(stderr, __VA_ARGS__); fprintf(stderr, "\n"); }
