--- a/configure.ac
+++ b/configure.ac
@@ -3438,10 +3438,6 @@ AS_VAR_IF([have_uuid], [missing], [
             AC_DEFINE([HAVE_UUID_GENERATE_TIME_SAFE], [1])
           ])
         ])
-        AS_VAR_IF([have_uuid], [yes], [
-          LIBUUID_CFLAGS=${LIBUUID_CFLAGS-""}
-          LIBUUID_LIBS=${LIBUUID_LIBS-"-luuid"}
-        ])
       ])
     ]
   )
@@ -3452,8 +3448,8 @@ AS_VAR_IF([have_uuid], [missing], [
   AC_CHECK_HEADERS([uuid/uuid.h], [
     AC_CHECK_FUNC([uuid_generate_time], [
       have_uuid=yes
-      LIBUUID_CFLAGS=${LIBUUID_CFLAGS-""}
-      LIBUUID_LIBS=${LIBUUID_LIBS-""}
+      LIBUUID_CFLAGS=${LIBUUID_CFLAGS:""}
+      LIBUUID_LIBS=${LIBUUID_LIBS:""}
     ])
   ])
 ])
--- a/setup.py
+++ b/setup.py
@@ -1118,8 +1118,6 @@ class PyBuildExt(build_ext):
         panel_library = 'panel'
         if curses_library == 'ncursesw':
             curses_defines.append(('HAVE_NCURSESW', '1'))
-            if not CROSS_COMPILING:
-                curses_includes.append('/usr/include/ncursesw')
             # Bug 1464056: If _curses.so links with ncursesw,
             # _curses_panel.so must link with panelw.
             panel_library = 'panelw'
