From 0343f6378994b9c45f851847062f1a1fef1bee92 Mon Sep 17 00:00:00 2001
From: Vladimir Ermakov <vooon341@gmail.com>
Date: Sun, 1 Jun 2025 18:24:41 +0200
Subject: [PATCH 4/6] v4l2: eanble risc-v 64 support

Signed-off-by: Vladimir Ermakov <vooon341@gmail.com>
---
 internal/v4l2/v4l2.go       | 2 +-
 internal/v4l2/v4l2_linux.go | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/internal/v4l2/v4l2.go
+++ b/internal/v4l2/v4l2.go
@@ -1,4 +1,4 @@
-//go:build !(linux && (386 || arm || mipsle || amd64 || arm64))
+//go:build !(linux && (386 || arm || mipsle || amd64 || arm64 || riscv64))
 
 package v4l2
 
--- a/internal/v4l2/v4l2_linux.go
+++ b/internal/v4l2/v4l2_linux.go
@@ -1,4 +1,4 @@
-//go:build linux && (386 || arm || mipsle || amd64 || arm64)
+//go:build linux && (386 || arm || mipsle || amd64 || arm64 || riscv64)
 
 package v4l2
 
