References:

https://github.com/rdoc/rdoc/pull/340

--- ruby-2.1.2.orig/lib/rdoc.rb	2014-09-02 17:14:28.719224215 -0300
+++ ruby-2.1.2/lib/rdoc.rb	2014-09-02 17:14:28.762223911 -0300
@@ -109,6 +109,8 @@
   def self.load_yaml
     begin
       gem 'psych'
+    rescue NameError => e # --disable-gems
+       raise unless e.name == :gem
     rescue Gem::LoadError
     end
 
--- ruby-2.1.2.orig/lib/rdoc/markdown.rb	2014-09-02 17:14:28.761223918 -0300
+++ ruby-2.1.2/lib/rdoc/markdown.rb	2014-09-02 17:14:28.805223607 -0300
@@ -525,7 +525,6 @@
 
 
 
-  require 'rubygems'
   require 'rdoc'
   require 'rdoc/markup/to_joined_paragraph'
   require 'rdoc/markdown/entities'
--- ruby-2.1.2.orig/lib/rdoc/text.rb	2014-09-02 17:14:28.721224201 -0300
+++ ruby-2.1.2/lib/rdoc/text.rb	2014-09-02 17:14:28.764223897 -0300
@@ -10,6 +10,8 @@
 
 begin
   gem 'json'
+rescue NameError => e # --disable-gems
+  raise unless e.name == :gem
 rescue Gem::LoadError
 end
 
