--- a/Makefile
+++ b/Makefile
@@ -113,9 +113,6 @@ install: $(TARGET) $(TRANSLATIONS)
 	$(INSTALLMAN) $(MAN_NL) $(DESTDIR)/$(MANDIR)/nl/man1
 	$(INSTALLDIR) $(DESTDIR)/$(DOCDIR)
 	$(INSTALLDOC) $(DOCS) $(DESTDIR)/$(DOCDIR)
-ifneq ($(DEBUG),yes)
-	$(STRIP) $(DESTDIR)/$(BINDIR)/$(TARGET)
-endif
 	mkdir -p $(DESTDIR)/$(PREFIX)/share/locale/nl/LC_MESSAGES
 	cp nl.mo $(DESTDIR)/$(PREFIX)/share/locale/nl/LC_MESSAGES/httping.mo
 
