diff -uprN a/unrar/makefile b/unrar/makefile
--- a/makefile	2014-06-12 13:36:14.723816266 +0200
+++ b/makefile	2014-06-12 13:36:36.174531961 +0200
@@ -2,12 +2,12 @@
 # Makefile for UNIX - unrar
 
 # Linux using GCC
-CXX=g++
-CXXFLAGS=-O2
+#CXX=g++
+#CXXFLAGS=-O2
 LIBFLAGS=-fPIC
 DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
 STRIP=strip
-LDFLAGS=-pthread
+LDFLAGS=-lpthread
 DESTDIR=/usr
 
 # Linux using LCC
