--- a/server/Makefile.in
+++ b/server/Makefile.in
@@ -26,7 +26,10 @@ gen_test_char: $(gen_test_char_OBJECTS)
 	$(LINK) $(EXTRA_LDFLAGS) $(gen_test_char_OBJECTS) $(EXTRA_LIBS)
 
 test_char.h: gen_test_char
-	./gen_test_char > test_char.h
+	true
+#	./gen_test_char > test_char.h
+#	doesn't matter if you run it on the buildhost or on an wl500gd,
+#	same output on both, so i just patched in the test_char.h :).
 
 util.lo: test_char.h
 
