Index: httpd-2.4.25/build/mkconfNW.awk
===================================================================
--- httpd-2.4.25.orig/build/mkconfNW.awk
+++ httpd-2.4.25/build/mkconfNW.awk
@@ -23,7 +23,7 @@ BEGIN {
     A["sysconfdir"] = "conf"
     A["iconsdir"] = "icons"
     A["manualdir"] = "manual"
-    A["runtimedir"] = "logs"
+    A["runtimedir"] = "log"
     A["errordir"] = "error"
     A["proxycachedir"] = "proxy"
 
Index: httpd-2.4.25/config.layout
===================================================================
--- httpd-2.4.25.orig/config.layout
+++ httpd-2.4.25/config.layout
@@ -28,8 +28,8 @@
     cgidir:        ${datadir}/cgi-bin
     includedir:    ${prefix}/include
     localstatedir: ${prefix}
-    runtimedir:    ${localstatedir}/logs
-    logfiledir:    ${localstatedir}/logs
+    runtimedir:    ${localstatedir}/log
+    logfiledir:    ${localstatedir}/log
     proxycachedir: ${localstatedir}/proxy
 </Layout>
 
@@ -150,7 +150,7 @@
     includedir:    ${prefix}/include
     localstatedir: /var${prefix}
     runtimedir:    ${localstatedir}/run
-    logfiledir:    ${localstatedir}/logs
+    logfiledir:    ${localstatedir}/log
     proxycachedir: ${localstatedir}/proxy
 </Layout>
 
