Index: httpd-2.4.25/include/scoreboard.h
===================================================================
--- httpd-2.4.25.orig/include/scoreboard.h
+++ httpd-2.4.25/include/scoreboard.h
@@ -40,7 +40,7 @@ extern "C" {
 
 /* Scoreboard file, if there is one */
 #ifndef DEFAULT_SCOREBOARD
-#define DEFAULT_SCOREBOARD "logs/apache_runtime_status"
+#define DEFAULT_SCOREBOARD "log/apache_runtime_status"
 #endif
 
 /* Scoreboard info on a process is, for now, kept very brief ---
