--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 
 
-PLUGINS=df cpu if_ if_err_ load memory processes swap netstat uptime interrupts irqstats ntpdate plugindir_
-#PLUGINS=cpu if_ if_err_ load memory processes netstat uptime interrupts irqstats
+#PLUGINS=df cpu if_ if_err_ load memory processes swap netstat uptime interrupts irqstats ntpdate plugindir_
+PLUGINS=cpu if_ if_err_ load memory processes netstat uptime interrupts irqstats
 
 munin-node: plugins/* munin-node.conf
 	@VERSION=$$(cat VERSION); \
