--- a/muninlite.in
+++ b/muninlite.in
@@ -113,7 +113,7 @@ PLUGINS=$RES
 
 # ===== MAIN LOOP =====
 FUNCTIONS="list nodes config fetch version quit"
-HOSTNAME=$( { hostname -f || hostname || cat /proc/sys/kernel/hostname || echo "unknown"; } 2>/dev/null )
+HOSTNAME=$(/sbin/uci get "system.@system[0].hostname" 2>/dev/null || cat /proc/sys/kernel/hostname)
 echo "# munin node at $HOSTNAME"
 while read -r arg0 arg1
 do
