--- a/munin-node.in
+++ b/munin-node.in
@@ -72,7 +72,7 @@ RES=""
 for PLUG in $PLUGINS
 do 
   if [ "$PLUG" = "if_" ]; then  
-    for INTER in $(grep '^ *\(ppp\|eth\|wlan\|ath\|ra\|ipsec\)\([^:]\)\{1,\}:' /proc/net/dev | cut -f1 -d: | sed 's/ //g');
+    for INTER in $(grep '^ *\(ppp\|eth\|wlan\|ath\|ra\|ipsec\|tap\)\([^:]\)\{1,\}:' /proc/net/dev | cut -f1 -d: | sed 's/ //g');
     do
       INTERRES=$(echo $INTER | sed 's/\./VLAN/')
       RES="$RES if_$INTERRES"
@@ -80,7 +80,7 @@ do
       eval "config_if_${INTERRES}() { config_if $INTER $@; };"
     done
   elif [ "$PLUG" = "if_err_" ]; then
-    for INTER in $(grep '^ *\(ppp\|eth\|wlan\|ath\|ra\|ipsec\)\([^:]\)\{1,\}:' /proc/net/dev | cut -f1 -d: | sed 's/ //g');
+    for INTER in $(grep '^ *\(ppp\|eth\|wlan\|ath\|ra\|ipsec\|tap\)\([^:]\)\{1,\}:' /proc/net/dev | cut -f1 -d: | sed 's/ //g');
     do
       INTERRES=$(echo $INTER | sed 's/\./VLAN/')
       RES="$RES if_err_$INTERRES"
