From 99e590f2177148420f3556bbb0dc2e493e94e45f Mon Sep 17 00:00:00 2001
From: Julien Malik <julien.malik@paraiso.me>
Date: Sat, 26 Nov 2022 13:05:20 +0100
Subject: [PATCH] add setup.py to avoid depending on flit for building

---
 setup.py | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)
 create mode 100644 setup.py

--- /dev/null
+++ b/setup.py
@@ -0,0 +1,18 @@
+#!/usr/bin/env python
+
+import setuptools
+
+setuptools.setup(
+      name='exceptiongroup',
+      version='1.0.4',
+      description='Backport of PEP 654 (exception groups)',
+      author='Alex Grönholm',
+      author_email='Alex Grönholm <alex.gronholm@nextday.fi>',
+      url='https://github.com/agronholm/exceptiongroup',
+      packages=['exceptiongroup'],
+      package_data={'': ['*']},
+      package_dir={'': 'src'},
+      extras_require={'test': ['pytest >= 6']},
+      python_requires='>=3.7',
+)
+
