--- a/configure.in
+++ b/configure.in
@@ -41,7 +41,7 @@ AC_CHECK_PROG(AR, ar, ar, gar)
 # fail
 AC_LANG_COMPILER_REQUIRE()
 
-if test "$CC" = "gcc"; then
+if test "$GCC" = yes; then
    CFLAGS="$CFLAGS -D_GNU_SOURCE -Wall -Wno-parentheses"
    ERRONWARN="-Werror -O0"
 elif test "$CC" = "clang"; then
--- a/configure
+++ b/configure
@@ -3348,7 +3348,7 @@ fi
 
 
 
-if test "$CC" = "gcc"; then
+if test "$GCC" = yes; then
    CFLAGS="$CFLAGS -D_GNU_SOURCE -Wall -Wno-parentheses"
    ERRONWARN="-Werror -O0"
 elif test "$CC" = "clang"; then
