diff -rupN seafile-server-4.1.2.orig/server/Makefile.am seafile-server-4.1.2/server/Makefile.am
--- seafile-server-4.1.2.orig/server/Makefile.am	2015-05-02 11:04:11.000000000 +0200
+++ seafile-server-4.1.2/server/Makefile.am	2015-06-14 01:28:55.924834806 +0200
@@ -4,6 +4,7 @@ AM_CFLAGS = -DPKGDATADIR=\"$(pkgdatadir)
 	-DPACKAGE_DATA_DIR=\""$(pkgdatadir)"\" \
 	-DSEAFILE_SERVER \
 	-DFULL_FEATURE \
+	-I$(STAGING_DIR)/usr/include/libevhtp-1.1 \
 	-I$(top_srcdir)/include \
 	-I$(top_srcdir)/lib \
 	-I$(top_builddir)/lib \
@@ -122,4 +123,4 @@ seaf_server_LDADD = @CCNET_LIBS@ \
 	@SEARPC_LIBS@ @JANSSON_LIBS@ @ZDB_LIBS@ @CURL_LIBS@ ${LIB_WS32} @ZLIB_LIBS@ \
 	@LIBARCHIVE_LIBS@
 
-seaf_server_LDFLAGS = @STATIC_COMPILE@ @SERVER_PKG_RPATH@
+seaf_server_LDFLAGS = @STATIC_COMPILE@ @SERVER_PKG_RPATH@ -L$(STAGING_DIR)/usr/lib/libevhtp-1.1
