diff -rupN seafile-server-4.1.2.orig/lib/net.c seafile-server-4.1.2/lib/net.c
--- seafile-server-4.1.2.orig/lib/net.c	2015-09-09 19:31:56.000000000 +0200
+++ seafile-server-4.1.2/lib/net.c	2015-09-14 20:20:30.906021296 +0200
@@ -31,11 +31,7 @@
 
 #include <fcntl.h>
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <event2/util.h>
-#else
-#include <evutil.h>
-#endif
 
 #include "net.h"
 
diff -rupN seafile-server-4.1.2.orig/lib/net.h seafile-server-4.1.2/lib/net.h
--- seafile-server-4.1.2.orig/lib/net.h	2015-09-09 19:31:56.000000000 +0200
+++ seafile-server-4.1.2/lib/net.h	2015-09-14 20:20:30.906021296 +0200
@@ -19,11 +19,7 @@
     #include <netinet/tcp.h>
 #endif
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <event2/util.h>
-#else
-#include <evutil.h>
-#endif
 
 #ifdef WIN32
     #define ECONNREFUSED WSAECONNREFUSED
diff -rupN seafile-server-4.1.2.orig/lib/utils.h seafile-server-4.1.2/lib/utils.h
--- seafile-server-4.1.2.orig/lib/utils.h	2015-09-09 19:31:56.000000000 +0200
+++ seafile-server-4.1.2/lib/utils.h	2015-09-14 20:20:30.907021326 +0200
@@ -13,11 +13,7 @@
 #include <stdlib.h>
 #include <sys/stat.h>
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <event2/util.h>
-#else
-#include <evutil.h>
-#endif
 
 #ifdef __linux__
 #include <endian.h>
diff -rupN seafile-server-4.1.2.orig/server/access-file.c seafile-server-4.1.2/server/access-file.c
--- seafile-server-4.1.2.orig/server/access-file.c	2015-09-09 19:31:56.000000000 +0200
+++ seafile-server-4.1.2/server/access-file.c	2015-09-14 20:20:30.907021326 +0200
@@ -3,13 +3,9 @@
 #define DEBUG_FLAG SEAFILE_DEBUG_HTTP
 #include "log.h"
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <event2/event.h>
 #include <event2/bufferevent.h>
 #include <event2/bufferevent_struct.h>
-#else
-#include <event.h>
-#endif
 
 #include <evhtp.h>
 
diff -rupN seafile-server-4.1.2.orig/server/http-server.c seafile-server-4.1.2/server/http-server.c
--- seafile-server-4.1.2.orig/server/http-server.c	2015-09-09 19:31:56.000000000 +0200
+++ seafile-server-4.1.2/server/http-server.c	2015-09-14 20:20:30.908021355 +0200
@@ -6,11 +6,7 @@
 #include <locale.h>
 #include <sys/types.h>
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <event2/event.h>
-#else
-#include <event.h>
-#endif
 
 #include <evhtp.h>
 
diff -rupN seafile-server-4.1.2.orig/server/listen-mgr.c seafile-server-4.1.2/server/listen-mgr.c
--- seafile-server-4.1.2.orig/server/listen-mgr.c	2015-09-09 19:31:56.000000000 +0200
+++ seafile-server-4.1.2/server/listen-mgr.c	2015-09-14 20:23:15.614452334 +0200
@@ -3,11 +3,9 @@
 #include <event2/event.h>
 #include <event2/listener.h>
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <event2/bufferevent.h>
 #include <event2/buffer_compat.h>
 #include <event2/bufferevent_struct.h>
-#endif
 
 
 #include "seafile-session.h"
diff -rupN seafile-server-4.1.2.orig/server/upload-file.c seafile-server-4.1.2/server/upload-file.c
--- seafile-server-4.1.2.orig/server/upload-file.c	2015-09-09 19:31:56.000000000 +0200
+++ seafile-server-4.1.2/server/upload-file.c	2015-09-14 20:20:30.909021385 +0200
@@ -6,11 +6,7 @@
 #include <getopt.h>
 #include <fcntl.h>
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <event2/event.h>
-#else
-#include <event.h>
-#endif
 
 #include <evhtp.h>
 
