diff --git a/tools/seafile-admin b/tools/seafile-admin
index 495ceab..72b6a07 100755
--- a/tools/seafile-admin
+++ b/tools/seafile-admin
@@ -502,7 +502,7 @@ def init_seahub():
     # create seahub_settings.py
     create_seahub_settings_py()
 
-    argv = [PYTHON, 'manage.py', 'syncdb']
+    argv = [PYTHON, 'manage.py', 'migrate']
     # Set proper PYTHONPATH before run django syncdb command
     env = get_seahub_env()
 
@@ -512,7 +512,7 @@ def init_seahub():
     print
 
     if run_argv(argv, cwd=seahub_dir, env=env) != 0:
-        error('Seahub syncdb failed')
+        error('Seahub migrate failed')
 
     info('done')
 
