--- a/configure
+++ b/configure
@@ -254,6 +254,13 @@ do
     --host-machine=* )
       MACHINE=`echo $option | sed -e 's/^.*=//;'`
     ;;
+    --target=* )
+      TARGET=`echo $option | sed -e 's/[^=]*=//;'`
+      OPERATINGSYSTEM=`echo $TARGET | sed -e 's/.*-//;'`
+      MACHINE=`echo $TARGET | sed -e 's/-.*//;'`
+      VERSION=""
+      SYSTEM=${OPERATINGSYSTEM}-${MACHINE}
+    ;;
     --help | -h )
       usage
       exit 0
@@ -287,7 +294,7 @@ case $SYSTEM in
         ;;
     esac
     ;;
-    Linux* )
+    Linux* | linux* )
         EXTRA_OBJECTS="sys_linux.o wrap_adjtimex.o"
         try_linuxcaps=1
         try_rtc=1
