From d7fcff4a8995dab26ae87d9a3048282cbe639014 Mon Sep 17 00:00:00 2001
From: "Schanzenbach, Martin" <mschanzenbach@posteo.de>
Date: Mon, 16 Dec 2019 09:33:36 +0100
Subject: [PATCH 03/12] fix mysql build

---
 src/my/my.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/my/my.c b/src/my/my.c
index a4099d088..256f71742 100644
--- a/src/my/my.c
+++ b/src/my/my.c
@@ -239,7 +239,7 @@ GNUNET_MY_extract_result (struct GNUNET_MYSQL_StatementHandle *sh,
           mysql_stmt_free_result (stmt);
           for (unsigned int j = 0; j < i; j++)
             if (NULL != rs[j].cleaner)
-              rs[j].cleaner (rs[j].cls,
+              rs[j].cleaner (rs[j].conv_cls,
                              rs[j].dst);
           return GNUNET_SYSERR;
         }
-- 
2.24.1

