From b3e6c45fcc78c75ea97bff528f34bd96089abdbf Mon Sep 17 00:00:00 2001
From: Christian Grothoff <christian@grothoff.org>
Date: Wed, 18 Dec 2019 10:01:46 +0100
Subject: [PATCH 10/12] set expriation to 1 year for URLs added via QR code

---
 src/namestore/namestore.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/namestore/namestore.conf.in b/src/namestore/namestore.conf.in
index e6cc74aec..39de06c6d 100644
--- a/src/namestore/namestore.conf.in
+++ b/src/namestore/namestore.conf.in
@@ -36,7 +36,7 @@ TEMPORARY_TABLE = NO
 ASYNC_COMMIT = NO
 
 [uri]
-gns = gnunet-namestore -u
+gns = gnunet-namestore -u -e 1a
 
 
 [fcfsd]
-- 
2.24.1

