--- a/src/include/gnunet_common.h
+++ b/src/include/gnunet_common.h
@@ -142,7 +142,6 @@ enum GNUNET_GenericReturnValue
  * Endian operations
  */
 
-#if __BYTE_ORDER == __LITTLE_ENDIAN
 #if defined(__linux__)
 #define BYTE_SWAP_16(x) __bswap_16 (x)
 #define BYTE_SWAP_32(x) __bswap_32 (x)
@@ -162,6 +161,7 @@ enum GNUNET_GenericReturnValue
                                              56))
 #endif
 
+#if __BYTE_ORDER == __LITTLE_ENDIAN
 #define GNUNET_htobe16(x) BYTE_SWAP_16 (x)
 #define GNUNET_htole16(x) (x)
 #define GNUNET_be16toh(x) BYTE_SWAP_16 (x)
