Index: gnunet-0.10.1/src/dht/gnunet-service-dht_neighbours.c
===================================================================
--- gnunet-0.10.1.orig/src/dht/gnunet-service-dht_neighbours.c
+++ gnunet-0.10.1/src/dht/gnunet-service-dht_neighbours.c
@@ -45,7 +45,6 @@
 #include "gnunet-service-dht_neighbours.h"
 #include "gnunet-service-dht_nse.h"
 #include "gnunet-service-dht_routing.h"
-#include <fenv.h>
 #include "dht.h"
 
 #define LOG_TRAFFIC(kind,...) GNUNET_log_from (kind, "dht-traffic",__VA_ARGS__)
Index: gnunet-0.10.1/src/dht/gnunet-service-wdht_neighbours.c
===================================================================
--- gnunet-0.10.1.orig/src/dht/gnunet-service-wdht_neighbours.c
+++ gnunet-0.10.1/src/dht/gnunet-service-wdht_neighbours.c
@@ -44,7 +44,6 @@
 #include "gnunet-service-wdht_datacache.h"
 #include "gnunet-service-wdht_neighbours.h"
 #include "gnunet-service-wdht_nse.h"
-#include <fenv.h>
 #include <stdlib.h>
 #include <string.h>
 #include "dht.h"
Index: gnunet-0.10.1/src/dht/gnunet-service-xdht_neighbours.c
===================================================================
--- gnunet-0.10.1.orig/src/dht/gnunet-service-xdht_neighbours.c
+++ gnunet-0.10.1/src/dht/gnunet-service-xdht_neighbours.c
@@ -41,7 +41,6 @@
 #include "gnunet-service-xdht_datacache.h"
 #include "gnunet-service-xdht_neighbours.h"
 #include "gnunet-service-xdht_routing.h"
-#include <fenv.h>
 #include "dht.h"
 
 /**
