Index: gnunet-0.10.1-svn35844/m4/libgnurl.m4
===================================================================
--- gnunet-0.10.1-svn35844.orig/m4/libgnurl.m4
+++ gnunet-0.10.1-svn35844/m4/libgnurl.m4
@@ -146,7 +146,7 @@ AC_DEFUN([LIBGNURL_CHECK_CONFIG],
            _libgnurl_save_libs=$LIBS
            LIBS="$LIBGNURL $LIBS"
 
-           AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <curl/curl.h>],[
+           AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <gnurl/curl.h>],[
 /* Try and use a few common options to force a failure if we are
    missing symbols or can't link. */
 int x;
Index: gnunet-0.10.1-svn35844/src/gns/gnunet-gns-proxy.c
===================================================================
--- gnunet-0.10.1-svn35844.orig/src/gns/gnunet-gns-proxy.c
+++ gnunet-0.10.1-svn35844/src/gns/gnunet-gns-proxy.c
@@ -29,7 +29,7 @@
  */
 #include "platform.h"
 #include <microhttpd.h>
-#include <curl/curl.h>
+#include <gnurl/curl.h>
 #include <gnutls/gnutls.h>
 #include <gnutls/x509.h>
 #include <gnutls/abstract.h>
Index: gnunet-0.10.1-svn35844/src/gns/test_gns_proxy.c
===================================================================
--- gnunet-0.10.1-svn35844.orig/src/gns/test_gns_proxy.c
+++ gnunet-0.10.1-svn35844/src/gns/test_gns_proxy.c
@@ -24,7 +24,7 @@
  * @author Martin Schanzenbach
  */
 #include "platform.h"
-#include <curl/curl.h>
+#include <gnurl/curl.h>
 #include <microhttpd.h>
 #include "gnunet_namestore_service.h"
 #include "gnunet_gns_service.h"
Index: gnunet-0.10.1-svn35844/src/hostlist/gnunet-daemon-hostlist_client.c
===================================================================
--- gnunet-0.10.1-svn35844.orig/src/hostlist/gnunet-daemon-hostlist_client.c
+++ gnunet-0.10.1-svn35844/src/hostlist/gnunet-daemon-hostlist_client.c
@@ -29,7 +29,7 @@
 #include "gnunet_statistics_service.h"
 #include "gnunet_transport_service.h"
 #include "gnunet-daemon-hostlist.h"
-#include <curl/curl.h>
+#include <gnurl/curl.h>
 
 
 /**
Index: gnunet-0.10.1-svn35844/src/pt/test_gns_vpn.c
===================================================================
--- gnunet-0.10.1-svn35844.orig/src/pt/test_gns_vpn.c
+++ gnunet-0.10.1-svn35844/src/pt/test_gns_vpn.c
@@ -24,7 +24,7 @@
  * @author Martin Schanzenbach
  */
 #include "platform.h"
-#include <curl/curl.h>
+#include <gnurl/curl.h>
 #include <microhttpd.h>
 #include "gnunet_identity_service.h"
 #include "gnunet_namestore_service.h"
Index: gnunet-0.10.1-svn35844/src/pt/test_gnunet_vpn.c
===================================================================
--- gnunet-0.10.1-svn35844.orig/src/pt/test_gnunet_vpn.c
+++ gnunet-0.10.1-svn35844/src/pt/test_gnunet_vpn.c
@@ -24,7 +24,7 @@
  * @author Christian Grothoff
  */
 #include "platform.h"
-#include <curl/curl.h>
+#include <gnurl/curl.h>
 #include <microhttpd.h>
 #include "gnunet_vpn_service.h"
 #include "gnunet_testing_lib.h"
Index: gnunet-0.10.1-svn35844/src/transport/plugin_transport_http.h
===================================================================
--- gnunet-0.10.1-svn35844.orig/src/transport/plugin_transport_http.h
+++ gnunet-0.10.1-svn35844/src/transport/plugin_transport_http.h
@@ -41,7 +41,7 @@
 #include "gnunet_os_lib.h"
 #include "gnunet_nat_lib.h"
 #include "microhttpd.h"
-#include <curl/curl.h>
+#include <gnurl/curl.h>
 
 
 #define DEBUG_HTTP GNUNET_EXTRA_LOGGING
Index: gnunet-0.10.1-svn35844/src/transport/plugin_transport_http_client.c
===================================================================
--- gnunet-0.10.1-svn35844.orig/src/transport/plugin_transport_http_client.c
+++ gnunet-0.10.1-svn35844/src/transport/plugin_transport_http_client.c
@@ -49,7 +49,7 @@
 #include "gnunet_protocols.h"
 #include "gnunet_transport_plugin.h"
 #include "plugin_transport_http_common.h"
-#include <curl/curl.h>
+#include <gnurl/curl.h>
 
 
 #define LOG(kind,...) GNUNET_log_from(kind, PLUGIN_NAME, __VA_ARGS__)
