Index: gnunet-0.10.1-svn35844/configure.ac
===================================================================
--- gnunet-0.10.1-svn35844.orig/configure.ac
+++ gnunet-0.10.1-svn35844/configure.ac
@@ -854,14 +854,7 @@ AC_ARG_WITH(microhttpd,
         AC_CHECK_HEADERS([microhttpd.h],
 	  AC_CHECK_DECL(MHD_OPTION_PER_IP_CONNECTION_LIMIT,
             AC_CHECK_LIB([microhttpd], [MHD_get_fdset2],
-              [AC_MSG_CHECKING([for libmicrohttpd >= 0.9.32])
-              AC_RUN_IFELSE([AC_LANG_SOURCE([
-                #include "$native_srcdir/src/include/platform.h"
-                #include <microhttpd.h>
-                int main () { return MHD_VERSION >= 0x0093200 ? 0 : 1; }
-               ])], [
-               AC_MSG_RESULT(ok)
-               lmhd=1],[AC_MSG_RESULT(failed)],lmhd=0)]),
+              ),
 	    [],[#include "$native_srcdir/src/include/platform.h"
                 #include <microhttpd.h>]),,
             [#include "$native_srcdir/src/include/platform.h"])
@@ -873,14 +866,7 @@ AC_ARG_WITH(microhttpd,
 	  AC_CHECK_DECL(MHD_OPTION_PER_IP_CONNECTION_LIMIT,
             AC_CHECK_LIB([microhttpd], [MHD_get_fdset2],
               EXT_LIB_PATH="-L$with_microhttpd/lib $EXT_LIB_PATH"
-              [AC_MSG_CHECKING([for libmicrohttpd >= 0.9.32])
-               AC_RUN_IFELSE([AC_LANG_SOURCE([
-                #include "$native_srcdir/src/include/platform.h"
-                #include <microhttpd.h>
-                int main () { return MHD_VERSION >= 0x0093200 ? 0 : 1; }
-               ])], [
-               AC_MSG_RESULT(ok)
-               lmhd=1],[AC_MSG_RESULT(failed)],lmhd=0)]),
+              ),
 	    [],[#include "$native_srcdir/src/include/platform.h"
                 #include <microhttpd.h>]),,
             [#include "$native_srcdir/src/include/platform.h"])
@@ -891,19 +877,12 @@ AC_ARG_WITH(microhttpd,
     AC_CHECK_HEADERS([microhttpd.h],
       AC_CHECK_DECL(MHD_OPTION_PER_IP_CONNECTION_LIMIT,
         AC_CHECK_LIB([microhttpd], [MHD_get_fdset2],
-          [AC_MSG_CHECKING([for libmicrohttpd >= 0.9.32])
-              AC_RUN_IFELSE([AC_LANG_SOURCE([
-                #include "$native_srcdir/src/include/platform.h"
-		#include <microhttpd.h>
-                int main () { return MHD_VERSION >= 0x0093200 ? 0 : 1; }
-               ])], [
-               AC_MSG_RESULT(ok)
-               lmhd=1],[AC_MSG_RESULT(failed)],lmhd=0)]),
+	),
 	[],[#include "$native_srcdir/src/include/platform.h"
             #include <microhttpd.h>]),,
        [#include "$native_srcdir/src/include/platform.h"])])
-AM_CONDITIONAL(HAVE_MHD, test x$lmhd = x1)
-AC_DEFINE_UNQUOTED([HAVE_MHD], $lmhd, [We have libmicrohttpd])
+AM_CONDITIONAL(HAVE_MHD, test x1 = x1)
+AC_DEFINE_UNQUOTED([HAVE_MHD], 1, [We have libmicrohttpd])
 
 
 # restore LIBS
@@ -1655,7 +1634,7 @@ then
 fi
 
 # MHD
-if test "x$lmhd" != "x1"
+if test "x1" != "x1"
 then
  AC_MSG_NOTICE([NOTICE: libmicrohttpd not found, http transport will not be installed.])
 fi
