--- a/configure.ac
+++ b/configure.ac
@@ -727,12 +727,7 @@ LIBCURL_CHECK_CONFIG([], [7.85.0], [],
 
 OLD_LIBS=$LIBS
 LIBS="$LIBS $LIBCURL"
-AC_RUN_IFELSE(
-     [AC_LANG_PROGRAM([[#include <curl/curl.h>]],
-                      [[return (CURLSSLSET_OK != curl_global_sslset(CURLSSLBACKEND_GNUTLS, NULL, NULL));]])],
-                    [curl_gnutls=1],
-                    [AC_MSG_WARN([cURL does not have GnuTLS backend])
-                     curl_gnutls=0])
+curl_gnutls=1
 LIBS=$OLD_LIBS
 AM_CONDITIONAL([HAVE_GNUTLS_CURL], [test "x$curl_gnutls" = "x1"])
 AC_SUBST([curl_gnutls])
