diff -rupN libevhtp-1.1-1.1.7-orig/CMakeLists.txt libevhtp-1.1-1.1.7/CMakeLists.txt
--- libevhtp-1.1-1.1.7-orig/CMakeLists.txt	2015-06-14 00:15:28.000000000 +0200
+++ libevhtp-1.1-1.1.7/CMakeLists.txt	2015-06-14 00:21:17.441299942 +0200
@@ -54,7 +54,7 @@ if (NOT HAVE_SYS_UN)
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DNO_SYS_UN")
 endif(NOT HAVE_SYS_UN)
 
-OPTION(EVHTP_DISABLE_SSL       "Disable ssl support"      OFF)
+OPTION(EVHTP_DISABLE_SSL       "Disable ssl support"      ON)
 OPTION(EVHTP_DISABLE_EVTHR     "Disable evthread support" OFF)
 OPTION(EVHTP_DIABLE_REGEX      "Disable regex support"    OFF)
 
@@ -182,11 +182,11 @@ endif()
 add_library(libevhtp STATIC ${LIBEVHTP_SOURCES} ${ONIG_SOURCES})
 set_target_properties(libevhtp PROPERTIES OUTPUT_NAME "evhtp")
 
-install (TARGETS libevhtp DESTINATION lib)
-install (FILES evhtp.h DESTINATION include)
-install (FILES htparse/htparse.h DESTINATION include)
-install (FILES evthr/evthr.h DESTINATION include)
-install (FILES oniguruma/onigposix.h DESTINATION include)
+install (TARGETS libevhtp DESTINATION lib/libevhtp-1.1)
+install (FILES evhtp.h DESTINATION include/libevhtp-1.1)
+install (FILES htparse/htparse.h DESTINATION include/libevhtp-1.1)
+install (FILES evthr/evthr.h DESTINATION include/libevhtp-1.1)
+install (FILES oniguruma/onigposix.h DESTINATION include/libevhtp-1.1)
 
 find_library (LIB_DL dl)
 set (SYS_LIBS ${LIB_DL})
