--- a/libsoup/soup-xmlrpc-old.c
+++ b/libsoup/soup-xmlrpc-old.c
@@ -11,7 +11,7 @@
 
 #include <string.h>
 
-#include <libxml/tree.h>
+#include <libxml/parser.h>
 
 #include "soup-xmlrpc-old.h"
 #include "soup.h"
--- a/libsoup/soup-xmlrpc.c
+++ b/libsoup/soup-xmlrpc.c
@@ -17,7 +17,7 @@
 
 #include <string.h>
 #include <errno.h>
-#include <libxml/tree.h>
+#include <libxml/parser.h>
 #include "soup-xmlrpc.h"
 #include "soup.h"
 
