--- a/sepolicy/sepolicy.py
+++ b/sepolicy/sepolicy.py
@@ -686,7 +686,6 @@ if __name__ == '__main__':
     gen_booleans_args(subparsers)
     gen_communicate_args(subparsers)
     gen_generate_args(subparsers)
-    gen_gui_args(subparsers)
     gen_interface_args(subparsers)
     gen_manpage_args(subparsers)
     gen_network_args(subparsers)
--- a/sepolicy/setup.py
+++ b/sepolicy/setup.py
@@ -13,10 +13,5 @@ setup(
     packages=[
         "sepolicy",
         "sepolicy.templates",
-        "sepolicy.help"
     ],
-    package_data={
-        'sepolicy': ['*.glade'],
-        'sepolicy.help': ['*.txt', '*.png']
-    }
 )
