--- a/src/webui/static/lovcombo/lovcombo-all.js
+++ b/src/webui/static/lovcombo/lovcombo-all.js
@@ -297,7 +297,7 @@ Ext.ux.form.LovCombo = Ext.extend(Ext.ux
 				this.store.clearFilter();
 				this.store.each(function(r) {
 					var checked = !(!v.match(
-						 '(^|' + this.separator + ')' + RegExp.escape(r.get(this.valueField))
+						 '(^|' + this.separator + ')' + RegExp.escape(String(r.get(this.valueField)))
 						+'(' + this.separator + '|$)'))
 					;
 
