From d048a1e98363197c0d2609f6adb6919bde473df4 Mon Sep 17 00:00:00 2001
From: Yousong Zhou <yszhou4tech@gmail.com>
Date: Mon, 23 Mar 2020 14:18:30 +0800
Subject: [PATCH] build: skip building tests

Signed-off-by: Yousong Zhou <yszhou4tech@gmail.com>
---
 Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 04400e184..b2c42d84c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -492,7 +492,6 @@ include Documentation/automake.mk
 include m4/automake.mk
 include lib/automake.mk
 include utilities/automake.mk
-include tests/automake.mk
 include include/automake.mk
 include third-party/automake.mk
 include debian/automake.mk
