--- a/utils/keytable/keytable.c
+++ b/utils/keytable/keytable.c
@@ -1686,7 +1686,7 @@ static void test_event(struct rc_device *rc_dev, int fd)
 
 		for (i = 0; i < rd / sizeof(struct input_event); i++) {
 			printf(_("%ld.%06ld: event type %s(0x%02x)"),
-				ev[i].time.tv_sec, ev[i].time.tv_usec,
+				ev[i].input_event_sec, ev[i].input_event_usec,
 				get_event_name(events_type, ev[i].type), ev[i].type);
 
 			switch (ev[i].type) {
