--- a/src/util/dict_db.c	2017-07-16 16:56:00.819659962 +0300
+++ b/src/util/dict_db.c	2017-07-16 16:58:31.011401358 +0300
@@ -733,8 +733,8 @@
 	msg_fatal("create DB database: %m");
     if (db == 0)
 	msg_panic("db_create null result");
-    if (type == DB_HASH && db->set_h_nelem(db, DICT_DB_NELM) != 0)
-	msg_fatal("set DB hash element count %d: %m", DICT_DB_NELM);
+//    if (type == DB_HASH && db->set_h_nelem(db, DICT_DB_NELM) != 0)
+//	msg_fatal("set DB hash element count %d: %m", DICT_DB_NELM);
 #if DB_VERSION_MAJOR == 6 || DB_VERSION_MAJOR == 5 || \
 	(DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR > 0)
     if ((errno = db->open(db, 0, db_path, 0, type, db_flags, 0644)) != 0)
