From 20f5118f7505e2e82be3504624ac934b8837d25b Mon Sep 17 00:00:00 2001
From: Scott Shambarger <devel@shambarger.net>
Date: Tue, 11 Jun 2019 20:10:04 -0700
Subject: [PATCH] [alsa] asoundlib.h should be alsa/asoundlib.h

---
 configure.ac       | 2 +-
 src/outputs/alsa.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9ec5eb101..98068bf11 100644
--- a/configure.ac
+++ b/configure.ac
@@ -238,7 +238,7 @@ AC_CHECK_SIZEOF([void *])
 dnl --- Begin configuring the options ---
 dnl ALSA
 FORK_ARG_WITH_CHECK([FORKED_OPTS], [ALSA support], [alsa], [ALSA],
-	[alsa], [snd_mixer_open], [asoundlib.h])
+	[alsa], [snd_mixer_open], [alsa/asoundlib.h])
 AM_CONDITIONAL([COND_ALSA], [[test "x$with_alsa" = "xyes"]])
 
 dnl PULSEAUDIO
diff --git a/src/outputs/alsa.c b/src/outputs/alsa.c
index b696f0c3d..1a32aad19 100644
--- a/src/outputs/alsa.c
+++ b/src/outputs/alsa.c
@@ -30,7 +30,7 @@
 #include <inttypes.h>
 
 #include <event2/event.h>
-#include <asoundlib.h>
+#include <alsa/asoundlib.h>
 
 #include "misc.h"
 #include "conffile.h"
