--- a/src/outputs.h
+++ b/src/outputs.h
@@ -185,7 +185,7 @@ struct output_data
   int samples;
 };
 
-struct output_buffer
+static struct output_buffer
 {
   struct timespec pts;
   // The array is two larger than max quality subscriptions because element 0
@@ -262,7 +262,7 @@ struct output_definition
 };
 
 // Our main list of devices, not for use by backend modules
-struct output_device *output_device_list;
+static struct output_device *output_device_list;
 
 /* ------------------------------- General use ------------------------------ */
 
