--- a/setup.py
+++ b/setup.py
@@ -12,7 +12,7 @@ setup(
     Self-service finite-state machines for the programmer on the go.
     """.strip(),
     readme='README.md',
-    packages=find_packages(exclude=[]),
+    packages=find_packages(exclude=["*._test", "*._test.*"]),
     package_dir={'automat': 'automat'},
     setup_requires=[
         'wheel',
@@ -25,6 +25,7 @@ setup(
     author='Glyph',
     author_email='glyph@twistedmatrix.com',
     include_package_data=True,
+    exclude_package_data={'':['_test/*']},
     license="MIT",
     keywords='fsm finite state machine automata',
     classifiers=[
