From: Florian Fainelli <florian@openwrt.org>
Date: Tue, 29 Jul 2008 10:22:38 +0000
Subject: [PATCH] Do not compile and generate curl tests and docs
--- a/Makefile.am
+++ b/Makefile.am
@@ -71,7 +71,7 @@ DISTCLEANFILES = buildinfo.txt
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests projects include docs
+DIST_SUBDIRS = $(SUBDIRS) projects include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -141,12 +141,9 @@ check-docs:
 if BUILD_DOCS
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
 else
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
 endif
 
 # We extend the standard uninstall with a custom hook:
