--- a/Makefile.am
+++ b/Makefile.am
@@ -96,7 +96,7 @@ DISTCLEANFILES = buildinfo.txt
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests packages include docs
+DIST_SUBDIRS = $(SUBDIRS) packages include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -217,12 +217,9 @@ cygwinbin:
 if BUILD_DOCS
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
 else
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
 endif
 
 # We extend the standard uninstall with a custom hook:
