--- a/Makefile.am
+++ b/Makefile.am
@@ -134,7 +134,7 @@ CLEANFILES = $(VC14_LIBVCXPROJ) \
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+DIST_SUBDIRS = $(SUBDIRS) packages include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -248,8 +248,6 @@ cygwinbin:
 # We extend the standard install with a custom hook:
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
