From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Fainelli <florian@openwrt.org>
Date: Tue, 29 Jul 2008 10:22:38 +0000
Subject: [PATCH] Do not compile and generate curl tests and docs
--- a/Makefile.am
+++ b/Makefile.am
@@ -96,7 +96,7 @@ ETAGS_ARGS = $(CURL_ETAGS_FILES)
 TAGS_DEPENDENCIES = $(CURL_ETAGS_FILES)
 
 SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests projects include docs
+DIST_SUBDIRS = $(SUBDIRS) projects include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -169,12 +169,9 @@ check-docs:
 if BUILD_DOCS
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
 else
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
 endif
 
 # We extend the standard uninstall with a custom hook:
