--- a/setup.py
+++ b/setup.py
@@ -109,7 +109,6 @@ ext_py_mods = [Extension('setools.policy
                          extra_compile_args=['-Werror', '-Wextra',
                                              '-Waggregate-return',
                                              '-Wfloat-equal',
-                                             '-Wformat', '-Wformat=2',
                                              '-Winit-self',
                                              '-Wmissing-format-attribute',
                                              '-Wmissing-include-dirs',
