diff -u --recursive setools-vanilla/setup.py setools/setup.py
--- setools-vanilla/setup.py	2020-04-01 09:57:49.000000000 -0500
+++ setools/setup.py	2020-08-12 21:44:41.265149504 -0500
@@ -109,7 +109,6 @@
                          extra_compile_args=['-Werror', '-Wextra',
                                              '-Waggregate-return',
                                              '-Wfloat-equal',
-                                             '-Wformat', '-Wformat=2',
                                              '-Winit-self',
                                              '-Wmissing-format-attribute',
                                              '-Wmissing-include-dirs',
