--- a/setup.py
+++ b/setup.py
@@ -125,6 +125,7 @@ ext_py_mods = [Extension('setools.policy
                                              '-Wno-unreachable-code',
                                              '-Wno-implicit-fallthrough',
                                              '-Wno-cast-function-type',
+                                             '-Wno-deprecated-declarations',
                                              '-fno-exceptions'])]
 
 installed_data = [('share/man/man1', glob.glob("man/*.1"))]
