From 89bdc6148cd8cffb1483f4fc0aa14d636f8f5b4f Mon Sep 17 00:00:00 2001
From: Bruno Silvestre <bruno.silvestre@gmail.com>
Date: Fri, 29 Jun 2018 14:06:51 -0300
Subject: [PATCH 2/3] Removing SSLv3 support

---
 src/config.c  | 5 -----
 src/context.c | 3 ---
 2 files changed, 8 deletions(-)

diff --git a/src/config.c b/src/config.c
index ce74997..6939fca 100644
--- a/src/config.c
+++ b/src/config.c
@@ -32,11 +32,6 @@ LSEC_API int luaopen_ssl_config(lua_State *L)
   lua_pushstring(L, "protocols");
   lua_newtable(L);
 
-#ifndef OPENSSL_NO_SSL3
-  lua_pushstring(L, "sslv3");
-  lua_pushboolean(L, 1);
-  lua_rawset(L, -3);
-#endif
   lua_pushstring(L, "tlsv1");
   lua_pushboolean(L, 1);
   lua_rawset(L, -3);
diff --git a/src/context.c b/src/context.c
index b9e8cda..d8fc8b6 100644
--- a/src/context.c
+++ b/src/context.c
@@ -66,9 +66,6 @@ static const SSL_METHOD* str2method(const char *method)
 {
   if (!strcmp(method, "any"))     return SSLv23_method();
   if (!strcmp(method, "sslv23"))  return SSLv23_method();  // deprecated
-#ifndef OPENSSL_NO_SSL3
-  if (!strcmp(method, "sslv3"))   return SSLv3_method();
-#endif
   if (!strcmp(method, "tlsv1"))   return TLSv1_method();
 #if (OPENSSL_VERSION_NUMBER >= 0x1000100fL)
   if (!strcmp(method, "tlsv1_1")) return TLSv1_1_method();
-- 
2.19.1

