--- a/meson.build
+++ b/meson.build
@@ -681,7 +681,7 @@ if get_option('daemon')
     cdata.set('HAVE_ALSA_UCM', 1)
   endif
 
-  gio_dep = dependency('gio-2.0', version : '>= 2.26.0')
+  gio_dep = dependency('gio-2.0', version : '>= 2.26.0', required : false)
   if get_option('gsettings').enabled()
     assert(gio_dep.found(), 'GSettings support needs glib I/O library (GIO)')
     cdata.set('HAVE_GSETTINGS', 1)
