From 485217d0ff8d0d17ea3815244b2bc2b747451e15 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sat, 10 Dec 2022 16:43:38 +0100
Subject: [PATCH] * configure.ac: Allow disabling NTLM if nettle present (Savannah #63431)

---
 configure.ac | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -648,8 +648,11 @@ else
 
   if test x"$HAVE_NETTLE" = xyes; then
     AC_DEFINE([HAVE_NETTLE], [1], [Use libnettle])
-    ENABLE_NTLM=yes
-    AC_DEFINE([ENABLE_NTLM], 1, [Define if you want the NTLM authorization support compiled in.])
+    if test x"$ENABLE_NTLM" != xno
+    then
+      ENABLE_NTLM=yes
+      AC_DEFINE([ENABLE_NTLM], 1, [Define if you want the NTLM authorization support compiled in.])
+    fi
   fi
 fi
 
