From c69030a904f8ab25b9ca2704c8a6dd03554e9503 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sun, 11 Dec 2022 13:31:38 +0100
Subject: [PATCH] * configure.ac: Disable nettle if NTLM is explicitly disabled

---
 configure.ac | 41 +++++++++++++++++++----------------------
 1 file changed, 19 insertions(+), 22 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -622,34 +622,31 @@ AS_IF([test x"$with_ssl" = xopenssl], [
 ]) # endif: --with-ssl == openssl?
 
 dnl Enable NTLM if requested and if SSL is available.
-if test x"$LIBSSL" != x || test "$ac_cv_lib_ssl32_SSL_connect" = yes
+if test x"$ENABLE_NTLM" != xno
 then
-  if test x"$ENABLE_NTLM" != xno
+  if test x"$LIBSSL" != x || test "$ac_cv_lib_ssl32_SSL_connect" = yes
   then
     ENABLE_NTLM=yes
     AC_DEFINE([ENABLE_NTLM], 1, [Define if you want the NTLM authorization support compiled in.])
-  fi
-else
-  PKG_CHECK_MODULES([NETTLE], nettle, [
-    HAVE_NETTLE=yes
-    LIBS="$NETTLE_LIBS $LIBS"
-    CFLAGS="$NETTLE_CFLAGS $CFLAGS"
-  ], [
-    AC_CHECK_LIB(nettle, nettle_md4_init, [HAVE_NETTLE=yes], [HAVE_NETTLE=no; AC_MSG_WARN(*** libnettle was not found. You will not be able to use NTLM)])
-    if test x"$HAVE_NETTLE" != xyes; then
-      if  test x"$ENABLE_NTLM" = xyes; then
-        AC_MSG_ERROR([NTLM authorization requested and SSL not enabled; aborting])
-      fi
-    else
-      AC_SUBST(NETTLE_LIBS, "-lnettle")
+  else
+    PKG_CHECK_MODULES([NETTLE], nettle, [
+      HAVE_NETTLE=yes
       LIBS="$NETTLE_LIBS $LIBS"
-    fi
-  ])
+      CFLAGS="$NETTLE_CFLAGS $CFLAGS"
+    ], [
+      AC_CHECK_LIB(nettle, nettle_md4_init, [HAVE_NETTLE=yes], [HAVE_NETTLE=no; AC_MSG_WARN(*** libnettle was not found. You will not be able to use NTLM)])
+      if test x"$HAVE_NETTLE" != xyes; then
+        if  test x"$ENABLE_NTLM" = xyes; then
+          AC_MSG_ERROR([NTLM authorization requested and SSL not enabled; aborting])
+        fi
+      else
+        AC_SUBST(NETTLE_LIBS, "-lnettle")
+        LIBS="$NETTLE_LIBS $LIBS"
+      fi
+    ])
 
-  if test x"$HAVE_NETTLE" = xyes; then
-    AC_DEFINE([HAVE_NETTLE], [1], [Use libnettle])
-    if test x"$ENABLE_NTLM" != xno
-    then
+    if test x"$HAVE_NETTLE" = xyes; then
+      AC_DEFINE([HAVE_NETTLE], [1], [Use libnettle])
       ENABLE_NTLM=yes
       AC_DEFINE([ENABLE_NTLM], 1, [Define if you want the NTLM authorization support compiled in.])
     fi
