From 94218682b15832fd6f8ed09a767941974075a1b4 Mon Sep 17 00:00:00 2001
From: "R.E. Wolff" <R.E.Wolff@BitWizard.nl>
Date: Tue, 7 Nov 2017 17:24:14 +0100
Subject: [PATCH] Sami Kerola: prevent MTR reporting unknown revision

---
 Makefile.am | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index c0709ca..23ac1fc 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -156,3 +156,7 @@ endif  # if CYGWIN
 if BUILD_BASH_COMPLETION
 dist_bashcompletion_DATA = bash-completion/mtr
 endif
+
+dist-hook:
+	$(AM_V_GEN)echo $(VERSION) > $(distdir)/.tarball-version
+
-- 
2.17.1

