--- a/dns.c
+++ b/dns.c
@@ -1308,28 +1308,6 @@ res_nmkquery(res_state statp,
                 return (-1);
         memset(buf, 0, HFIXEDSZ);
         hp = (HEADER *) buf;
-        /* We randomize the IDs every time.  The old code just
-           incremented by one after the initial randomization which
-           still predictable if the application does multiple
-           requests.  */
-#if 0
-        hp->id = htons(++statp->id);
-#else
-        hp->id = htons(statp->id);
-        int randombits;
-        do
-          {
-#ifdef RANDOM_BITS
-            RANDOM_BITS (randombits);
-#else
-            struct timeval tv;
-            gettimeofday (&tv, NULL);
-            randombits = (tv.tv_sec << 8) ^ tv.tv_usec;
-#endif
-          }
-        while ((randombits & 0xffff) == 0);
-        statp->id = (statp->id + randombits) & 0xffff;
-#endif
         hp->opcode = op;
         hp->rd = (statp->options & RES_RECURSE) != 0;
         hp->rcode = NOERROR;
