--- a/main.c
+++ b/main.c
@@ -100,7 +100,7 @@ struct l2tp_data {
 };
 
 /* netlink socket */
-static struct nl_handle *nl_sock;
+static struct nl_sock *nl_sock;
 static int nl_family;
 
 /*****************************************************************************
@@ -788,9 +788,9 @@ static int do_show(int argc, char **argv
 
 int do_ipl2tp(int argc, char **argv)
 {
-	nl_sock = nl_handle_alloc();
+	nl_sock = nl_socket_alloc();
 	if (!nl_sock) {
-		perror("nl_handle_alloc");
+		perror("nl_socket_alloc");
 		return 1;
 	}
 
