From 47422cf01f0a352c3f5f015585a28c99dba79862 Mon Sep 17 00:00:00 2001
From: Michael Heimpold <mhei@heimpold.de>
Date: Wed, 13 Jul 2016 00:40:08 +0200
Subject: [PATCH 08/16] RETURN_STRINGL takes only 2 arguments

Signed-off-by: Michael Heimpold <mhei@heimpold.de>
---
 dio.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dio.c b/dio.c
index 6f2d58c..e8660f8 100644
--- a/dio.c
+++ b/dio.c
@@ -24,6 +24,7 @@
 #include "php.h"
 #include "php_ini.h"
 #include "ext/standard/info.h"
+#include "ext/standard/php_string.h"
 
 #include "php_dio.h"
 #include "php_dio_stream_wrappers.h"
@@ -213,7 +214,7 @@ PHP_FUNCTION(dio_read)
 	data = erealloc(data, res + 1);
 	data[res] = 0;
 
-	RETURN_STRINGL(data, res, 0);
+	RETURN_STRINGL(data, res);
 }
 /* }}} */
 
-- 
2.5.0

