--- a/configure.ac
+++ b/configure.ac
@@ -122,7 +122,6 @@ OVS_CHECK_SOCKET_LIBS
 OVS_CHECK_XENSERVER_VERSION
 OVS_CHECK_GROFF
 OVS_CHECK_TLS
-OVS_CHECK_ATOMIC_LIBS
 OVS_CHECK_GCC4_ATOMICS
 OVS_CHECK_ATOMIC_ALWAYS_LOCK_FREE(1)
 OVS_CHECK_ATOMIC_ALWAYS_LOCK_FREE(2)
--- a/lib/netdev-linux.c
+++ b/lib/netdev-linux.c
@@ -39,7 +39,9 @@
 #include <netpacket/packet.h>
 #include <net/if.h>
 #include <net/if_arp.h>
+#if defined(__UCLIBC__) || defined(__GLIBC__)
 #include <net/if_packet.h>
+#endif
 #include <net/route.h>
 #include <poll.h>
 #include <stdlib.h>
--- a/lib/ovs-atomic.h
+++ b/lib/ovs-atomic.h
@@ -320,7 +320,7 @@
 #include "util.h"
 
 #define IN_OVS_ATOMIC_H
-    #if __CHECKER__
+    #if 1
         /* sparse doesn't understand some GCC extensions we use. */
         #include "ovs-atomic-pthreads.h"
     #elif __has_extension(c_atomic)
