--- a/miniupnpd.c
+++ b/miniupnpd.c
@@ -39,7 +39,7 @@
 #include <sys/param.h>
 #if defined(sun)
 #include <kstat.h>
-#else
+#elif !defined(__linux__)
 /* for BSD's sysctl */
 #include <sys/sysctl.h>
 #endif
--- a/netfilter/iptpinhole.h
+++ b/netfilter/iptpinhole.h
@@ -8,6 +8,7 @@
 #define IPTPINHOLE_H_INCLUDED
 
 #ifdef ENABLE_UPNPPINHOLE
+#include <sys/types.h>
 
 int add_pinhole(const char * ifname,
                 const char * rem_host, unsigned short rem_port,
