--- a/src/configure
+++ b/src/configure
@@ -3691,7 +3691,7 @@ done
 
 for ac_header in linux/if.h
 do :
-  ac_fn_c_check_header_compile "$LINENO" "linux/if.h" "ac_cv_header_linux_if_h" "#include<sys/socket.h>
+  ac_fn_c_check_header_compile "$LINENO" "linux/if.h" "ac_cv_header_linux_if_h" "#include <sys/socket.h>
 "
 if test "x$ac_cv_header_linux_if_h" = xyes; then :
   cat >>confdefs.h <<_ACEOF
@@ -3705,10 +3705,11 @@ done
 for ac_header in linux/if_pppox.h
 do :
   ac_fn_c_check_header_compile "$LINENO" "linux/if_pppox.h" "ac_cv_header_linux_if_pppox_h" "
-#include<sys/socket.h>
-#include<net/ethernet.h>
-#include<linux/if.h>
-#include<linux/in.h>
+#include <sys/socket.h>
+#include <net/ethernet.h>
+#include <linux/if.h>
+#include <linux/in.h>
+#include <linux/in6.h>
 
 "
 if test "x$ac_cv_header_linux_if_pppox_h" = xyes; then :
@@ -4644,7 +4645,7 @@ esac
 $as_echo_n "checking packing order of bit fields... " >&6; }
 if test "${rpppoe_cv_pack_bitfields+set}" != set ; then
 if test "$cross_compiling" = yes; then :
-  $ECHO "no defaults for cross-compiling"; exit 0
+  rpppoe_cv_pack_bitfields="$ac_cv_pack_bitfields"
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
