--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -71,7 +71,7 @@ pppoe-sniff: pppoe-sniff.o if.o common.o
 	@CC@ -o $@ $^ $(LDFLAGS)
 
 pppoe-server: pppoe-server.o if.o debug.o common.o md5.o libevent/libevent.a @PPPOE_SERVER_DEPS@
-	@CC@ -o $@ @RDYNAMIC@ $^ $(LDFLAGS) $(PPPOE_SERVER_LIBS) -Llibevent -levent
+	@CC@ -o $@ @RDYNAMIC@ $^ $(LDFLAGS) $(PPPOE_SERVER_LIBS)
 
 pppoe: pppoe.o if.o debug.o common.o ppp.o discovery.o
 	@CC@ -o $@ $^ $(LDFLAGS)
