diff -urN libtheora-1.1.1/Makefile.am libtheora-1.1.1.new/Makefile.am
--- libtheora-1.1.1/Makefile.am	2009-07-14 01:55:02.000000000 +0200
+++ libtheora-1.1.1.new/Makefile.am	2012-12-05 14:21:06.103521554 +0100
@@ -8,7 +8,7 @@
 EXAMPLES_DIR =
 endif
 
-SUBDIRS = lib include doc tests m4 $(EXAMPLES_DIR)
+SUBDIRS = lib include m4 $(EXAMPLES_DIR)
 
 
 # we include the whole debian/ dir in EXTRA_DIST because there's a problem
diff -urN libtheora-1.1.1/Makefile.in libtheora-1.1.1.new/Makefile.in
--- libtheora-1.1.1/Makefile.in	2009-10-01 20:04:07.000000000 +0200
+++ libtheora-1.1.1.new/Makefile.in	2012-12-05 14:21:18.239521864 +0100
@@ -164,7 +164,7 @@
 @THEORA_ENABLE_EXAMPLES_TRUE@EXAMPLES_DIR = examples
 @THEORA_ENABLE_EXAMPLES_FALSE@EXAMPLES_DIR = 
 
-SUBDIRS = lib include doc tests m4 $(EXAMPLES_DIR)
+SUBDIRS = lib include m4 $(EXAMPLES_DIR)
 
 # we include the whole debian/ dir in EXTRA_DIST because there's a problem
 # with autotools and HFS+ MacOSX file systems that caused debian/Makefile.am
@@ -200,7 +200,7 @@
 	missing mkinstalldirs theora-uninstalled.pc.in theora.pc.in \
 	theoradec-uninstalled.pc.in theoradec.pc.in \
 	theoraenc-uninstalled.pc.in theoraenc.pc.in
-DIST_SUBDIRS = lib include doc tests m4 examples
+DIST_SUBDIRS = lib include m4 examples
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
