--- libtheora-1.1.1/configure.ac	2009-10-01 20:00:33.000000000 +0200
+++ libtheora-1.1.1.new/configure.ac	2012-12-05 14:22:26.319523608 +0100
@@ -313,11 +313,6 @@
 dnl check for SDL
 HAVE_SDL=no
 
-AM_PATH_SDL(,[
-  HAVE_SDL=yes
-  SDL_LIBS=`$SDL_CONFIG --libs`
-],AC_MSG_WARN([*** Unable to find SDL -- Not compiling example players ***]))
-
 dnl check for OSS
 HAVE_OSS=no
 AC_CHECK_HEADERS([sys/soundcard.h soundcard.h machine/soundcard.h],[
