From 070cee8d1efba1ca260777a204fba7cdfe676ca5 Mon Sep 17 00:00:00 2001
From: Nicholas Marriott <nicholas.marriott@gmail.com>
Date: Tue, 17 Jan 2017 23:14:03 +0000
Subject: [PATCH] Look for OPENSSL_init_ssl, from Tomasz Miasko.

---
 configure.ac | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 5793c2d..161040c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -93,11 +93,16 @@ if test "x$found_libcrypto" = xno; then
 	AC_MSG_ERROR("libcrypto not found")
 fi
 AC_SEARCH_LIBS(
-	SSL_library_init,
+	OPENSSL_init_ssl,
 	[ssl],
 	found_libssl=yes,
 	found_libssl=no
 )
+AC_SEARCH_LIBS(
+	SSL_library_init,
+	[ssl],
+	found_libssl=yes
+)
 if test "x$found_libssl" = xno; then
 	AC_MSG_ERROR("libssl not found")
 fi
