From 028f59bef0ea9435fb8fbe095b2939652ce63479 Mon Sep 17 00:00:00 2001
From: Nicholas Marriott <nicholas.marriott@gmail.com>
Date: Mon, 3 Apr 2023 08:54:28 +0100
Subject: [PATCH] Fix use-after-free, GitHub issue 126.

---
 connect.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/connect.c
+++ b/connect.c
@@ -550,8 +550,8 @@ httpproxy(struct server *srv,
 			if (strlen(line) < 12 ||
 			    strncmp(line, "HTTP/", 5) != 0 ||
 			    strncmp(line + 8, " 200", 4) != 0) {
-				xfree(line);
 				xasprintf(cause, "unexpected data: %s", line);
+				xfree(line);
 				return (-1);
 			}
 			header = 1;
