--- a/libavcodec/mips/cabac.h
+++ b/libavcodec/mips/cabac.h
@@ -30,6 +30,7 @@
 #include "libavutil/mips/mmiutils.h"
 #include "config.h"
 
+#ifndef __mips16
 #define get_cabac_inline get_cabac_inline_mips
 static av_always_inline int get_cabac_inline_mips(CABACContext *c,
                                                   uint8_t * const state){
@@ -225,4 +226,6 @@ static av_always_inline int get_cabac_by
 
     return res;
 }
+
+#endif
 #endif /* AVCODEC_MIPS_CABAC_H */
