From bb3c50b46d50b8bf4f45d7ae8f24607aaf23acea Mon Sep 17 00:00:00 2001
From: Andreas Rheinhardt <andreas.rheinhardt@outlook.com>
Date: Sun, 2 Jun 2024 06:11:46 +0200
Subject: [PATCH] avcodec/tiff: Suppress unused variable warnings

Signed-off-by: Andreas Rheinhardt <andreas.rheinhardt@outlook.com>
---
 libavcodec/tiff.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/libavcodec/tiff.c
+++ b/libavcodec/tiff.c
@@ -427,7 +427,7 @@ static void av_always_inline horizontal_
             uint8_t shift = is_dng ? 0 : 16 - bpp;
             GetBitContext gb;
 
-            int ret = init_get_bits8(&gb, src, width);
+            av_unused int ret = init_get_bits8(&gb, src, width);
             av_assert1(ret >= 0);
             for (int i = 0; i < s->width; i++) {
                 dst16[i] = get_bits(&gb, bpp) << shift;
@@ -462,7 +462,7 @@ static void unpack_gray(TiffContext *s,
     GetBitContext gb;
     uint16_t *dst = (uint16_t *)(p->data[0] + lnum * p->linesize[0]);
 
-    int ret = init_get_bits8(&gb, src, width);
+    av_unused int ret = init_get_bits8(&gb, src, width);
     av_assert1(ret >= 0);
 
     for (int i = 0; i < s->width; i++) {
