From b8edcb3683570867864a3d26253c4e48be51baab Mon Sep 17 00:00:00 2001
From: chewitt <github@chrishewitt.net>
Date: Sun, 11 Aug 2019 07:08:19 +0000
Subject: [PATCH 2/2] add long-term yuv2rgb logging patch

---
 libswscale/yuv2rgb.c | 4 ----
 1 file changed, 4 deletions(-)

--- a/libswscale/yuv2rgb.c
+++ b/libswscale/yuv2rgb.c
@@ -575,10 +575,6 @@ SwsFunc ff_yuv2rgb_get_func_ptr(SwsInter
     if (t)
         return t;
 
-    av_log(c, AV_LOG_WARNING,
-           "No accelerated colorspace conversion found from %s to %s.\n",
-           av_get_pix_fmt_name(c->opts.src_format), av_get_pix_fmt_name(c->opts.dst_format));
-
     if (c->opts.src_format == AV_PIX_FMT_YUV422P) {
         switch (c->opts.dst_format) {
         case AV_PIX_FMT_BGR48BE:
