diff --git a/src/node_crypto.cc b/src/node_crypto.cc
index 972b1e4..7c0f65a 100644
--- a/src/node_crypto.cc
+++ b/src/node_crypto.cc
@@ -5623,7 +5623,7 @@ void PBKDF2(const FunctionCallbackInfo<Value>& args) {
   }
 
   raw_keylen = args[3]->NumberValue();
-  if (raw_keylen < 0.0 || std::isnan(raw_keylen) || std::isinf(raw_keylen) ||
+  if (raw_keylen < 0.0 || isnan(raw_keylen) || isinf(raw_keylen) ||
       raw_keylen > INT_MAX) {
     type_error = "Bad key length";
     goto err;
