--- a/configure
+++ b/configure
@@ -60,7 +60,7 @@
 
 valid_os = ('win', 'mac', 'solaris', 'freebsd', 'openbsd', 'linux',
             'android', 'aix')
-valid_arch = ('arm', 'arm64', 'ia32', 'mips', 'mipsel', 'mips64el', 'ppc',
+valid_arch = ('arm', 'arm64', 'ia32', 'mips', 'mipsel', 'mips64', 'mips64el', 'ppc',
               'ppc64', 'x32','x64', 'x86', 'x86_64', 's390', 's390x')
 valid_arm_float_abi = ('soft', 'softfp', 'hard')
 valid_arm_fpu = ('vfp', 'vfpv3', 'vfpv3-d16', 'neon')
@@ -795,6 +795,9 @@
   if rtn == 'mipsel' and '_LP64' in k:
     rtn = 'mips64el'
 
+  if rtn == 'mips' and '_LP64' in k:
+    rtn = 'mips64'
+
   return rtn
 
 
@@ -877,7 +880,7 @@
 
   if target_arch == 'arm':
     configure_arm(o)
-  elif target_arch in ('mips', 'mipsel', 'mips64el'):
+  elif target_arch in ('mips', 'mipsel', 'mips64', 'mips64el'):
     configure_mips(o)
 
   if flavor == 'aix':
