--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -242,6 +242,7 @@
       'sources': [
         '<(V8_ROOT)/src/init/setup-isolate-full.cc',
       ],
+      'include_dirs': [ '<!@(echo "$STAGING_DIR"/usr/../usr/include)' ],
     },  # v8_init
     {
       'target_name': 'v8_initializers',
@@ -714,6 +715,7 @@
         'v8_shared_internal_headers',
       ],
       'sources': ['<@(v8_compiler_sources)'],
+      'include_dirs': [ '<!@(echo "$STAGING_DIR"/usr/../usr/include)' ],
       'conditions': [
         ['OS=="win"', {
           'msvs_precompiled_header': '<(V8_ROOT)/../../tools/msvs/pch/v8_pch.h',
@@ -1435,6 +1437,7 @@
       'type': 'executable',
       'libraries!':[ '-lcrypto', '-lssl', '-lz', '-lhttp_parser', '-luv', '-lnghttp2', '-lcares' ],
       'library_dirs':[ '../../../../staging_dir/hostpkg/share/icu/current/lib' ],
+      'include_dirs': [ '<!@(echo "$STAGING_DIR"/usr/../usr/include)' ],
       'dependencies': [
         'v8_base_without_compiler',
         'v8_compiler_for_mksnapshot',
