--- a/Makefile.am
+++ b/Makefile.am
@@ -23,7 +23,7 @@ ACLOCAL_AMFLAGS = -I m4
 # VERSION = @VERSION@@GITVERSION@ # for development version
 VERSION = @VERSION@
 
-SUBDIRS = src tests doc tools
+SUBDIRS = src
 
 EXTRA_HEADERS = src/mpc-log.h
 include_HEADERS = src/mpc.h @MPC_LOG_H@
--- a/Makefile.in
+++ b/Makefile.in
@@ -376,7 +376,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = src tests doc tools
+SUBDIRS = src
 EXTRA_HEADERS = src/mpc-log.h
 include_HEADERS = src/mpc.h @MPC_LOG_H@
 EXTRA_DIST = doc/fdl-1.3.texi src/mpc-log.h
