--- a/configure.ac
+++ b/configure.ac
@@ -141,7 +141,7 @@ AC_ARG_ENABLE([openssl],
         else
                 openssl="true"
                 if test "x$enableval" = "xyes"; then
-                        AC_CHECK_LIB([ssl], [SSL_library_init], [], [AC_MSG_ERROR([libssl not found])])
+                        AC_CHECK_LIB([ssl], [SSL_CTX_new], [], [AC_MSG_ERROR([libssl not found])])
                         AC_CHECK_LIB([crypto], [SHA1_Init], [], [AC_MSG_ERROR([libcrypto not found])])
                 else
                         AC_MSG_CHECKING([for openssl in $enableval])
