diff -u --recursive vips-8.10.2-vanilla/configure.ac vips-8.10.2/configure.ac
--- vips-8.10.2-vanilla/configure.ac	2020-10-12 11:43:59.000000000 -0500
+++ vips-8.10.2/configure.ac	2020-11-19 12:16:45.187129100 -0600
@@ -26,7 +26,11 @@
 VIPS_MINOR_VERSION=vips_minor_version()
 VIPS_MICRO_VERSION=vips_micro_version()
 VIPS_VERSION=vips_version()
-VIPS_VERSION_STRING=$VIPS_VERSION-`date -u -r $srcdir/ChangeLog`
+if test "x$SOURCE_DATE_EPOCH" = "x"; then
+  VIPS_VERSION_STRING=$VIPS_VERSION-`date -u -r ChangeLog`
+else
+  VIPS_VERSION_STRING=$VIPS_VERSION-`LC_ALL=C date --utc --date="@$SOURCE_DATE_EPOCH"`
+fi
 
 # libtool library versioning ... not user-visible (except as part of the
 # library file name) and does not correspond to major/minor/micro above
