diff --git a/src/socket.c b/src/socket.c
index 82c6252..e2feda9 100644
--- a/src/socket.c
+++ b/src/socket.c
@@ -241,10 +241,6 @@ socket_open(socket_st * hd, const char *hostname, const char *service,
 	/* get server name */
 	memset(&hints, 0, sizeof(hints));
 
-#ifdef AI_IDN
-	hints.ai_flags = AI_IDN|AI_IDN_ALLOW_UNASSIGNED;
-#endif
-
 	hints.ai_socktype = udp ? SOCK_DGRAM : SOCK_STREAM;
 	if ((err = getaddrinfo(hostname, service, &hints, &res))) {
 		fprintf(stderr, "Cannot resolve %s:%s: %s\n", hostname,
