diff --git a/src/cli-debug.c b/src/cli-debug.c
index 5f9eb00..a8dde6c 100644
--- a/src/cli-debug.c
+++ b/src/cli-debug.c
@@ -165,8 +165,10 @@ static const TLS_TEST tls_tests[] = {
 #endif
 	{"for max record size (RFC6066) support", test_max_record_size, "yes",
 	 "no", "dunno"},
+#ifdef ENABLE_OCSP
 	{"for OCSP status response (RFC6066) support", test_ocsp_status, "yes",
 	 "no", "dunno"},
+#endif
 	{"for OpenPGP authentication (RFC6091) support", test_openpgp1,
 	 "yes", "no", "dunno"},
 	{NULL, NULL, NULL, NULL, NULL}
diff --git a/src/common.c b/src/common.c
index fff9449..ffa1896 100644
--- a/src/common.c
+++ b/src/common.c
@@ -572,9 +572,11 @@ int print_info(gnutls_session_t session, int verbose, int print_cert)
 	printf("- Options:");
 	if (gnutls_safe_renegotiation_status(session)!=0)
 		printf(" safe renegotiation,");
+#ifdef ENABLE_OCSP
 	if (gnutls_ocsp_status_request_get(session, &resp)==0) {
 		printf(" OCSP status request%s,", gnutls_ocsp_status_request_is_checked(session,0)!=0?"":"[ignored]");
 	}
+#endif
 	printf("\n");
 
 #ifdef ENABLE_DTLS_SRTP
diff --git a/src/tests.c b/src/tests.c
index 6cb251d..5c45159 100644
--- a/src/tests.c
+++ b/src/tests.c
@@ -246,6 +246,7 @@ test_code_t test_safe_renegotiation(gnutls_session_t session)
 	return ret;
 }
 
+#ifdef ENABLE_OCSP
 test_code_t test_ocsp_status(gnutls_session_t session)
 {
 	int ret;
@@ -275,6 +276,7 @@ test_code_t test_ocsp_status(gnutls_session_t session)
 
 	return TEST_FAILED;
 }
+#endif
 
 test_code_t test_safe_renegotiation_scsv(gnutls_session_t session)
 {
