--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,5 @@
 from setuptools import setup
+import os

 # TODO: if py gets upgrade to >=1.6,
 #       remove _width_of_current_line in terminal.py
@@ -18,8 +19,8 @@ INSTALL_REQUIRES = [

 def main():
     setup(
-        use_scm_version={"write_to": "src/_pytest/_version.py"},
-        setup_requires=["setuptools-scm", "setuptools>=40.0"],
+        version=os.getenv('PKG_VERSION'),
+        setup_requires=["setuptools>=40.0"],
         package_dir={"": "src"},
         # fmt: off
         extras_require={
