diff --git a/setup.py b/setup.py
index 1d1ae72..511aed5 100644
--- a/setup.py
+++ b/setup.py
@@ -444,7 +444,8 @@ class PyBuildExt(build_ext):
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
         if cross_compiling:
             self.add_gcc_paths()
-        self.add_multiarch_paths()
+        else:
+            self.add_multiarch_paths()
 
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
