--- a/setup.py
+++ b/setup.py
@@ -859,7 +859,7 @@ class PyBuildExt(build_ext):
 
         min_openssl_ver = 0x00907000
         have_any_openssl = ssl_incs is not None and ssl_libs is not None
-        have_usable_openssl = (have_any_openssl and
+        have_usable_openssl = cross_compiling and (have_any_openssl and
                                openssl_ver >= min_openssl_ver)
 
         if have_any_openssl:
