---
 Modules/Setup.dist |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Modules/Setup.dist
+++ b/Modules/Setup.dist
@@ -460,7 +460,7 @@ GLHACK=-Dclear=__GLclear
 # Andrew Kuchling's zlib module.
 # This require zlib 1.1.3 (or later).
 # See http://www.gzip.org/zlib/
-#zlib zlibmodule.c -I$(prefix)/include -L$(exec_prefix)/lib -lz
+zlib zlibmodule.c -lz
 
 # Interface to the Expat XML parser
 #
