--- a/libbridge/libbridge_devif.c
+++ b/libbridge/libbridge_devif.c
@@ -24,6 +24,7 @@
 #include <string.h>
 #include <dirent.h>
 #include <fcntl.h>
+#include <limits.h>
 
 #include "libbridge.h"
 #include "libbridge_private.h"
--- a/libbridge/libbridge_init.c
+++ b/libbridge/libbridge_init.c
@@ -24,6 +24,7 @@
 #include <dirent.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <limits.h>
 
 #include "libbridge.h"
 #include "libbridge_private.h"
