--- a/libhttpd/protocol.c
+++ b/libhttpd/protocol.c
@@ -28,6 +28,7 @@
 
 #if defined(_WIN32)
 #else
+#include <fcntl.h>
 #include <unistd.h>
 #include <sys/file.h>
 #endif
--- a/src/firewall.c
+++ b/src/firewall.c
@@ -35,7 +35,6 @@
 #include <pthread.h>
 #include <sys/wait.h>
 #include <sys/types.h>
-#include <sys/unistd.h>
 
 #include <string.h>
 
--- a/src/client_list.c
+++ b/src/client_list.c
@@ -31,9 +31,9 @@
 #include <syslog.h>
 #include <errno.h>
 #include <pthread.h>
+#include <unistd.h>
 #include <sys/wait.h>
 #include <sys/types.h>
-#include <sys/unistd.h>
 
 #include <string.h>
 
--- a/src/util.c
+++ b/src/util.c
@@ -33,10 +33,10 @@
 #include <syslog.h>
 #include <errno.h>
 #include <pthread.h>
+#include <unistd.h>
 #include <sys/wait.h>
 #include <sys/types.h>
 #include <sys/time.h>
-#include <sys/unistd.h>
 #include <netinet/in.h>
 #include <sys/ioctl.h>
 #include <arpa/inet.h>
