Index: boost_1_57_0/tools/build/src/tools/gcc.jam
===================================================================
--- boost_1_57_0.orig/tools/build/src/tools/gcc.jam
+++ boost_1_57_0/tools/build/src/tools/gcc.jam
@@ -1037,7 +1037,7 @@ rule setup-threading ( targets * : sourc
             case *bsd    : option = -pthread ;  # There is no -lrt on BSD.
             case sgi     : # gcc on IRIX does not support multi-threading.
             case darwin  : # No threading options.
-            case *       : option = -pthread ; libs = rt ;
+            case *       : # pass appropriate options via OpenWrt
         }
 
         if $(option)
Index: boost_1_57_0/tools/build/src/tools/gcc.py
===================================================================
--- boost_1_57_0.orig/tools/build/src/tools/gcc.py
+++ boost_1_57_0/tools/build/src/tools/gcc.py
@@ -700,8 +700,8 @@ elif bjam.variable('UNIX'):
         # Darwin has no threading options, don't set anything here.
         pass
     else:
-        flags('gcc', 'OPTIONS', ['<threading>multi'], ['-pthread'])
-        flags('gcc', 'FINDLIBS-SA', [], ['rt'])
+        # pass appropriate options via OpenWrt
+        pass
 
 def cpu_flags(toolset, variable, architecture, instruction_set, values, default=None):
     #FIXME: for some reason this fails.  Probably out of date feature code
