--- a/meson.build
+++ b/meson.build
@@ -51,7 +51,7 @@ sshfs_deps = [ dependency('fuse3', version: '>= 3.1.0'),
 executable('sshfs', sshfs_sources,
            include_directories: include_dirs,
            dependencies: sshfs_deps,
-           c_args: ['-DFUSE_USE_VERSION=31'],
+           c_args: ['-DFUSE_USE_VERSION=30'],
            install: true,
            install_dir: get_option('bindir'))
 
