--- a/fizz/CMakeLists.txt
+++ b/fizz/CMakeLists.txt
@@ -52,7 +52,7 @@ if (NOT folly_FOUND)
   find_package(Folly MODULE REQUIRED)
 endif()
 
-find_package(Boost REQUIRED COMPONENTS system thread filesystem regex context
+find_package(Boost REQUIRED COMPONENTS system filesystem regex context
                                        program_options)
 find_package(OpenSSL REQUIRED)
 find_package(Glog REQUIRED)
