--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,5 @@
+cmake_minimum_required(VERSION 2.6)
+
+add_subdirectory(server)
+add_subdirectory(cli)
+add_subdirectory(keystored)
--- a/cli/CMakeLists.txt
+++ b/cli/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 
 # include custom Modules
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/../CMakeModules/")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/CMakeModules/")
 
 project(netopeer2-cli C)
 include(GNUInstallDirs)
@@ -84,7 +84,7 @@ endif()
 install(FILES ${PROJECT_SOURCE_DIR}/doc/${PROJECT_NAME}.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
 
 # clean cmake cache
-add_custom_target(cleancache
+add_custom_target(cleancache_cli
                   COMMAND make clean
                   COMMAND find . -iname '*cmake*' -not -name CMakeLists.txt -exec rm -rf {} +
                   COMMAND rm -rf Makefile Doxyfile
--- a/keystored/CMakeLists.txt
+++ b/keystored/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 
 # include custom Modules
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/../CMakeModules/")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/CMakeModules/")
 
 project(keystored C)
 include(GNUInstallDirs)
@@ -147,7 +147,7 @@ if (SSH_KEY_INSTALL)
         execute_process(COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/scripts/ssh-key-import.sh)")
 endif()
 
-add_custom_target(install-scripts-ide
+add_custom_target(install-scripts-ide-keystored
     scripts/model-install.sh
     scripts/ssh-key-import.sh
 )
--- a/server/CMakeLists.txt
+++ b/server/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.8.9)
 
 # include custom Modules
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/../CMakeModules/")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/CMakeModules/")
 
 project(netopeer2-server C)
 include(GNUInstallDirs)
@@ -208,7 +208,7 @@ if (ENABLE_CONFIGURATION)
         execute_process(COMMAND ${PROJECT_BINARY_DIR}/model-install.sh)")
 endif()
 
-add_custom_target(install-scripts-ide
+add_custom_target(install-scripts-ide-server
     ${PROJECT_BINARY_DIR}/model-install.sh
 )
 
