From 4d7955f069922c482886e03e5cd352281dbce146 Mon Sep 17 00:00:00 2001
From: Yousong Zhou <yszhou4tech@gmail.com>
Date: Tue, 2 Apr 2019 06:31:31 +0000
Subject: [PATCH 2/4] configure: allow enabling/disabling libudev from command
 line

---
 configure | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure b/configure
index d274df816c..a138faeb72 100755
--- a/configure
+++ b/configure
@@ -1491,6 +1491,10 @@ for opt do
   ;;
   --disable-libpmem) libpmem=no
   ;;
+  --enable-libudev) libudev=yes
+  ;;
+  --disable-libudev) libudev=no
+  ;;
   --disable-fortify-source) fortify_source="no"
   ;;
   *)
