--- a/meson.build
+++ b/meson.build
@@ -53,7 +53,7 @@ endforeach
 have_tools = 'CONFIG_TOOLS' in config_host
 have_block = have_system or have_tools
 
-python = import('python').find_installation()
+python = 'python3'
 
 supported_oses = ['windows', 'freebsd', 'netbsd', 'openbsd', 'darwin', 'sunos', 'linux']
 supported_cpus = ['ppc', 'ppc64', 's390x', 'riscv32', 'riscv64', 'x86', 'x86_64',
@@ -2442,7 +2442,7 @@ summary(summary_info, bool_yn: true, sec
 summary_info = {}
 summary_info += {'git':               config_host['GIT']}
 summary_info += {'make':              config_host['MAKE']}
-summary_info += {'python':            '@0@ (version: @1@)'.format(python.full_path(), python.language_version())}
+#summary_info += {'python':            '@0@ (version: @1@)'.format(python.full_path(), python.language_version())}
 summary_info += {'sphinx-build':      sphinx_build.found()}
 if config_host.has_key('HAVE_GDB_BIN')
   summary_info += {'gdb':             config_host['HAVE_GDB_BIN']}
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -1,4 +1,4 @@
-py3 = import('python').find_installation()
+py3 = 'python3'
 
 subdir('bench')
 
--- a/ui/meson.build
+++ b/ui/meson.build
@@ -128,7 +128,7 @@ if have_system or xkbcommon.found()
                   output: output,
                   capture: true,
                   input: files('keycodemapdb/data/keymaps.csv'),
-                  command: [python.full_path(), files('keycodemapdb/tools/keymap-gen'),
+                  command: ['python3', files('keycodemapdb/tools/keymap-gen'),
                             'code-map',
                             '--lang', 'glib2',
                             '--varname', 'qemu_input_map_@0@_to_@1@'.format(e[0], e[1]),
--- a/tests/qapi-schema/meson.build
+++ b/tests/qapi-schema/meson.build
@@ -201,7 +201,8 @@ schemas = [
 # Because people may want to use test-qapi.py from the command line, we
 # are not using the "#! /usr/bin/env python3" trick here.  See
 # docs/devel/build-system.txt
-test('QAPI schema regression tests', python, args: files('test-qapi.py', schemas),
+pyexec = find_program('python3')
+test('QAPI schema regression tests', pyexec, args: files('test-qapi.py', schemas),
      env: test_env, suite: ['qapi-schema', 'qapi-frontend'])
 
 diff = find_program('diff')
