--- a/meson.build
+++ b/meson.build
@@ -3451,10 +3451,6 @@ subdir('common-user')
 subdir('bsd-user')
 subdir('linux-user')
 
-# needed for fuzzing binaries
-subdir('tests/qtest/libqos')
-subdir('tests/qtest/fuzz')
-
 # accel modules
 tcg_real_module_ss = ss.source_set()
 tcg_real_module_ss.add_all(when: 'CONFIG_TCG_MODULAR', if_true: tcg_module_ss)
@@ -3945,10 +3941,6 @@ subdir('scripts')
 subdir('tools')
 subdir('pc-bios')
 subdir('docs')
-subdir('tests')
-if gtk.found()
-  subdir('po')
-endif
 
 if host_machine.system() == 'windows'
   nsis_cmd = [
