--- a/meson.build
+++ b/meson.build
@@ -3828,10 +3828,6 @@ subdir('common-user')
 subdir('bsd-user')
 subdir('linux-user')
 
-# needed for fuzzing binaries
-subdir('tests/qtest/libqos')
-subdir('tests/qtest/fuzz')
-
 # accel modules
 target_modules += { 'accel' : { 'qtest': qtest_module_ss }}
 
@@ -4456,10 +4452,6 @@ subdir('scripts')
 subdir('tools')
 subdir('pc-bios')
 subdir('docs')
-subdir('tests')
-if gtk.found()
-  subdir('po')
-endif
 
 if host_machine.system() == 'windows'
   nsis_cmd = [
