From 19fe46ecb796c0d30d66dd7e7038fd7f2d6f9bf4 Mon Sep 17 00:00:00 2001
From: Florian Lindner <florian.lindner@student.tuwien.ac.at>
Date: Thu, 8 Jun 2023 16:55:34 +0200
Subject: [PATCH] bindings: include <cstdint> in key.hpp for uint8_t

---
 src/bindings/cpp/include/key.hpp | 1 +
 1 file changed, 1 insertion(+)

--- a/src/bindings/cpp/include/key.hpp
+++ b/src/bindings/cpp/include/key.hpp
@@ -10,6 +10,7 @@
 #define ELEKTRA_KEY_HPP
 
 #include <cstdarg>
+#include <cstdint>
 #include <cstring>
 #include <functional>
 #include <locale>
