--- a/ssh_config
+++ b/ssh_config
@@ -46,3 +46,6 @@
 #   VisualHostKey no
 #   ProxyCommand ssh -q -W %h:%p gateway.example.com
 #   RekeyLimit 1G 1h
+
+# enable DSCP QoS values (per RFC-4594)
+#IPQoS AF21 AF11
--- a/sshd_config
+++ b/sshd_config
@@ -122,6 +122,9 @@ UsePrivilegeSeparation sandbox		# Defaul
 # no default banner path
 #Banner none
 
+# enable DSCP QoS values (per RFC-4594)
+#IPQoS AF21 AF11
+
 # override default of no subsystems
 Subsystem	sftp	/usr/libexec/sftp-server
 
