--- a/runtime/doc/Makefile
+++ b/runtime/doc/Makefile
@@ -317,7 +317,6 @@ all: tags vim.man evim.man vimdiff.man v
 # Use Vim to generate the tags file.  Can only be used when Vim has been
 # compiled and installed.  Supports multiple languages.
 vimtags: $(DOCS)
-	$(VIMEXE) -u NONE -esX -c "helptags ++t ." -c quit
 
 # Use "doctags" to generate the tags file.  Only works for English!
 tags: doctags $(DOCS)
