--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,12 +41,6 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
   target_compile_definitions(miniupnpc-private INTERFACE _DARWIN_C_SOURCE)
 endif ()
 
-# Set compiler specific build flags
-if (CMAKE_COMPILER_IS_GNUCC AND NOT CMAKE_SYSTEM_NAME STREQUAL "AmigaOS")
-  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-  target_compile_options(miniupnpc-private INTERFACE -Wall)
-endif ()
-
 # Suppress noise warnings
 if (MSVC)
   target_compile_definitions(miniupnpc-private INTERFACE _CRT_SECURE_NO_WARNINGS)
@@ -221,16 +215,16 @@ endif ()
 
 if (NOT UPNPC_NO_INSTALL)
   install (FILES
-    miniupnpc.h
-    miniwget.h
-    upnpcommands.h
-    igd_desc_parse.h
-    upnpreplyparse.h
-    upnperrors.h
-    upnpdev.h
-    miniupnpctypes.h
-    portlistingparse.h
-    miniupnpc_declspec.h
+    include/miniupnpc.h
+    include/miniwget.h
+    include/upnpcommands.h
+    include/igd_desc_parse.h
+    include/upnpreplyparse.h
+    include/upnperrors.h
+    include/upnpdev.h
+    include/miniupnpctypes.h
+    include/portlistingparse.h
+    include/miniupnpc_declspec.h
     DESTINATION include/miniupnpc
   )
 
