--- a/argz.h
+++ b/argz.h
@@ -39,6 +39,7 @@
 #define _ARGZ_H	1
 
 #include <features.h>
+#include <sys/cdefs.h>
 
 #define __need_error_t
 #include <errno.h>
@@ -76,9 +77,9 @@ extern error_t argz_create_sep (__const char *__restrict __string,
 
 /* Returns the number of strings in ARGZ.  */
 extern size_t __argz_count (__const char *__argz, size_t __len)
-     __THROW __attribute_pure__;
+     __THROW;
 extern size_t argz_count (__const char *__argz, size_t __len)
-     __THROW __attribute_pure__;
+     __THROW;
 
 /* Puts pointers to each string in ARGZ into ARGV, which must be large enough
    to hold them all.  */
