--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,8 +2,7 @@
 requires = [
     "setuptools; python_version != '3.3'",
     "setuptools<40.0; python_version == '3.3'",
-    "wheel",
-    "setuptools_scm"
+    "wheel"
 ]
 build-backend = "setuptools.build_meta"
 
--- a/setup.py
+++ b/setup.py
@@ -49,9 +49,6 @@ README = README()  # NOQA
 
 
 setup(name="python-dateutil",
-      use_scm_version={
-          'write_to': 'dateutil/_version.py',
-      },
       description="Extensions to the standard Python datetime module",
       author="Gustavo Niemeyer",
       author_email="gustavo@niemeyer.net",
@@ -65,7 +62,6 @@ setup(name="python-dateutil",
       python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*",
       package_data={"dateutil.zoneinfo": ["dateutil-zoneinfo.tar.gz"]},
       zip_safe=True,
-      setup_requires=['setuptools_scm'],
       install_requires=["six >=1.5"],
       classifiers=[
           'Development Status :: 5 - Production/Stable',
