From d6feb6f574933753371687ee42fa19d0b0d8d777 Mon Sep 17 00:00:00 2001
From: Thomas Kriechbaumer <kriechbaumer@gmail.com>
Date: Tue, 17 Feb 2015 10:08:30 +0000
Subject: [PATCH] fix savebuff timer initialization

closes #868
---
 modules/savebuff.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/modules/savebuff.cpp
+++ b/modules/savebuff.cpp
@@ -93,13 +93,13 @@ public:
 		else
 			m_sPassword = CBlowfish::MD5(sArgs);
 
+		AddTimer(new CSaveBuffJob(this, 60, 0, "SaveBuff", "Saves the current buffer to disk every 1 minute"));
+
 		return( !m_bBootError );
 	}
 
 	virtual bool OnBoot() override
 	{
-		AddTimer(new CSaveBuffJob(this, 60, 0, "SaveBuff", "Saves the current buffer to disk every 1 minute"));
-
 		CDir saveDir(GetSavePath());
 		for (CFile* pFile : saveDir) {
 			CString sName;
