From fa14938321eda39f16bee6068296e6abc9df7b85 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski+openwrt@gmail.com>
Date: Wed, 6 Apr 2011 04:11:48 +0200
Subject: [PATCH] Add a uClibc++ build workaround

---
 modules/webadmin.cpp |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

--- a/modules/webadmin.cpp
+++ b/modules/webadmin.cpp
@@ -20,6 +20,7 @@
 #include <znc/IRCNetwork.h>
 #include <znc/IRCSock.h>
 
+using std::string;
 using std::stringstream;
 using std::make_pair;
 using std::set;
@@ -75,7 +76,7 @@ class CWebAdminMod : public CModule {
 public:
 	MODCONSTRUCTOR(CWebAdminMod) {
 		VPair vParams;
-		vParams.push_back(make_pair("user", ""));
+		vParams.push_back(make_pair((string)"user", (string)""));
 		AddSubPage(new CWebSubPage("settings", "Global Settings", CWebSubPage::F_ADMIN));
 		AddSubPage(new CWebSubPage("edituser", "Your Settings", vParams));
 		AddSubPage(new CWebSubPage("traffic", "Traffic Info", CWebSubPage::F_ADMIN));
