 src/su.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/src/su.c
+++ b/src/su.c
@@ -1170,8 +1170,12 @@ int main (int argc, char **argv)
 
 		if (fd >= 0) {
 			err = ioctl (fd, TIOCNOTTY, (char *) NULL);
+			if ((err == -1) && (errno == ENOTTY)) {
+				/* There are no controlling terminal already */
+				err = 0;
+			}
 			(void) close (fd);
-		} else if (ENXIO == errno) {
+		} else if ((errno == ENXIO) || (errno == EACCES)) {
 			/* There are no controlling terminal already */
 			err = 0;
 		}
