--- a/configure.ac
+++ b/configure.ac
@@ -31,30 +31,6 @@ AC_CANONICAL_HOST
 AM_PROG_CC_C_O
 AC_GNU_SOURCE
 
-# Detect the distribution. This is used for the default configuration and
-# for some distro-specific build options.
-AC_MSG_CHECKING([host distribution])
-AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO], [Specify the Linux distribution to target: One of redhat, oracle, centos, fedora, suse, gentoo, debian, arch, slackware, paldo, openmandriva or pardus.]))
-if type lsb_release >/dev/null 2>&1 && test "z$with_distro" = "z"; then
-	with_distro=`lsb_release -is`
-fi
-if test "z$with_distro" = "z"; then
-	AC_CHECK_FILE(/etc/redhat-release,with_distro="redhat")
-	AC_CHECK_FILE(/etc/oracle-release,with_distro="oracle")
-	AC_CHECK_FILE(/etc/centos-release,with_distro="centos")
-	AC_CHECK_FILE(/etc/fedora-release,with_distro="fedora")
-	AC_CHECK_FILE(/etc/SuSE-release,with_distro="suse")
-	AC_CHECK_FILE(/etc/gentoo-release,with_distro="gentoo")
-	AC_CHECK_FILE(/etc/debian_version,with_distro="debian")
-	AC_CHECK_FILE(/etc/arch-release,with_distro="arch")
-	AC_CHECK_FILE(/etc/slackware-version,with_distro="slackware")
-	AC_CHECK_FILE(/etc/frugalware-release,with_distro="frugalware")
-	AC_CHECK_FILE(/etc/mandrakelinux-release, with_distro="openmandriva")
-	AC_CHECK_FILE(/etc/mandriva-release,with_distro="openmandriva")
-	AC_CHECK_FILE(/etc/pardus-release,with_distro="pardus")
-fi
-with_distro=`echo ${with_distro} | tr '[[:upper:]]' '[[:lower:]]'`
-
 if test "z$with_distro" = "z"; then
 	with_distro="unknown"
 fi
