From 3efa0cf3455cbe330b4e79a647a57ad8e9cf3476 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Sun, 27 Jun 2021 23:42:52 -0400
Subject: [PATCH] lxc-download: Switch GPG server
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 templates/lxc-download.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/templates/lxc-download.in
+++ b/templates/lxc-download.in
@@ -55,11 +55,11 @@ LXC_PATH=
 LXC_ROOTFS=
 
 if [ -z "${DOWNLOAD_KEYSERVER:-}" ]; then
-  DOWNLOAD_KEYSERVER="hkp://pool.sks-keyservers.net"
+  DOWNLOAD_KEYSERVER="hkp://keyserver.ubuntu.com"
 
   # Deal with GPG over http proxy
   if [ -n "${http_proxy:-}" ]; then
-    DOWNLOAD_KEYSERVER="hkp://p80.pool.sks-keyservers.net:80"
+    DOWNLOAD_KEYSERVER="hkp://keyserver.ubuntu.com:80"
     DOWNLOAD_GPG_PROXY="--keyserver-options http-proxy=\"${http_proxy}\""
   fi
 fi
