--- a/configure.ac
+++ b/configure.ac
@@ -555,9 +555,6 @@ dnl it should be safe to re-edit 0.40 ba
 use_nls=no
 if test "x$enable_nls" = "xyes" ; then
     use_nls=yes
-    m4_ifdef([IT_PROG_INTLTOOL],
-             [IT_PROG_INTLTOOL([0.35.0],[no-xml])],
-             [AC_MSG_ERROR("--enable-nls requires intltool to be installed.")])
     AC_CHECK_HEADERS([libintl.h])
     GETTEXT_PACKAGE=transmission-gtk
     AC_SUBST(GETTEXT_PACKAGE)
