From 6de0a3d7a3cd0b2ca6d1b33b5d4d7f44908c1ac3 Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Thu, 1 Mar 2018 11:21:43 -0800
Subject: [PATCH] portcheck: Switch to HTTPS.

Useful for OpenWrt.
---
 libtransmission/rpcimpl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libtransmission/rpcimpl.c b/libtransmission/rpcimpl.c
index 46fd3192c..dcd96df59 100644
--- a/libtransmission/rpcimpl.c
+++ b/libtransmission/rpcimpl.c
@@ -1455,7 +1455,7 @@ portTest (tr_session               * session,
           struct tr_rpc_idle_data  * idle_data)
 {
   const int port = tr_sessionGetPeerPort (session);
-  char * url = tr_strdup_printf ("http://portcheck.transmissionbt.com/%d", port);
+  char * url = tr_strdup_printf ("https://portcheck.transmissionbt.com/%d", port);
   tr_webRun (session, url, portTested, idle_data);
   tr_free (url);
   return NULL;
-- 
2.14.3

