--- a/libstdc++-v3/config/os/gnu-linux/os_defines.h	2015-01-05 06:33:28.000000000 -0600
+++ b/libstdc++-v3/config/os/gnu-linux/os_defines.h	2016-04-15 12:32:49.374356001 -0500
@@ -41,8 +41,8 @@
 // Provide a declaration for the possibly deprecated gets function, as
 // glibc 2.15 and later does not declare gets for ISO C11 when
 // __GNU_SOURCE is defined.
-#if __GLIBC_PREREQ(2,15) && defined(_GNU_SOURCE)
+//#if __GLIBC_PREREQ(2,15) && defined(_GNU_SOURCE)
 # undef _GLIBCXX_HAVE_GETS
-#endif
+//#endif
 
 #endif
