--- -	2015-05-19 16:27:29.770936016 +0200
+++ protobuf-2.6.1/src/google/protobuf/stubs/platform_macros.h	2015-05-19 13:49:52.115444643 +0200
@@ -55,7 +55,7 @@
 #elif defined(__aarch64__)
 #define GOOGLE_PROTOBUF_ARCH_AARCH64 1
 #define GOOGLE_PROTOBUF_ARCH_64_BIT 1
-#elif defined(__MIPSEL__)
+#elif defined(__MIPSEL__) || defined(__MIPSEB__)
 #if defined(__LP64__)
 #define GOOGLE_PROTOBUF_ARCH_MIPS64 1
 #define GOOGLE_PROTOBUF_ARCH_64_BIT 1
