--- a/src/output.c
+++ b/src/output.c
@@ -234,7 +234,7 @@ int print_header(int option) {
 			if (output_method==PLAIN_OUT && ansi_output) printf("\033[1;2H");
 	        printf("bwm-ng v" VERSION " (delay %2.3fs); ",(float)delay/1000);
 			if (output_method==PLAIN_OUT) printf("press 'ctrl-c' to end this%s",(ansi_output ? "\033[2;2H" : "")); else printf("input: ");
-            printf(input2str());
+            printf("%s\n",input2str());
             printf("%s\n",show_all_if2str());
 			if (output_method==PLAIN_OUT) {
 				if (ansi_output)
