From e76b4f915c80d0fdfdd6602593b1f7bd7e078c3b Mon Sep 17 00:00:00 2001
From: Stepan Henek <stepan.henek@nic.cz>
Date: Thu, 5 Mar 2026 12:38:21 +0100
Subject: [PATCH] make project compatible with python3.13

---
 ubus_python.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/ubus_python.c
+++ b/ubus_python.c
@@ -1078,7 +1078,8 @@ static PyObject *ubus_python_add(PyObjec
 	if (PyList_Append(python_alloc_list, methods)) {
 		ubus_remove_object(ctx, &object->object);
 		free_ubus_object(object);
-		PyEval_CallMethod(python_alloc_list, "pop", "");
+		PyObject *_pop_result = PyObject_CallMethod(python_alloc_list, "pop", NULL);
+		Py_XDECREF(_pop_result);
 		return NULL;
 	}
 
