diff --git a/Makefile b/Makefile
index 4a62f44..fee93e8 100644
--- a/Makefile
+++ b/Makefile
@@ -85,12 +85,12 @@ supportedsites:
 	$(PYTHON) devscripts/make_supportedsites.py docs/supportedsites.md
 
 README.txt: README.md
-	pandoc -f $(MARKDOWN) -t plain README.md -o README.txt
+#	pandoc -f $(MARKDOWN) -t plain README.md -o README.txt
 
 youtube-dl.1: README.md
-	$(PYTHON) devscripts/prepare_manpage.py youtube-dl.1.temp.md
-	pandoc -s -f $(MARKDOWN) -t man youtube-dl.1.temp.md -o youtube-dl.1
-	rm -f youtube-dl.1.temp.md
+#	$(PYTHON) devscripts/prepare_manpage.py youtube-dl.1.temp.md
+#	pandoc -s -f $(MARKDOWN) -t man youtube-dl.1.temp.md -o youtube-dl.1
+#	rm -f youtube-dl.1.temp.md
 
 youtube-dl.bash-completion: youtube_dl/*.py youtube_dl/*/*.py devscripts/bash-completion.in
 	$(PYTHON) devscripts/bash-completion.py
