Index: picocom-1.7/term.c
===================================================================
--- picocom-1.7.orig/term.c
+++ picocom-1.7/term.c
@@ -33,10 +33,11 @@
 #include <string.h>
 #include <errno.h>
 #include <unistd.h>
-#ifdef __linux__
+#if defined(__linux__) && (defined(__GLIBC__) || defined(__UCLIBC__))
 #include <termio.h>
 #else
 #include <termios.h>
+#include <sys/ioctl.h>
 #endif /* of __linux__ */
 
 #include "term.h"
