--- a/driver_ais.c
+++ b/driver_ais.c
@@ -24,6 +24,7 @@
 
 /* strlcpy() needs _DARWIN_C_SOURCE */
 #define _DARWIN_C_SOURCE
+#define _BSD_SOURCE
 
 #include <stdlib.h>
 #include <string.h>
--- a/gpsctl.c
+++ b/gpsctl.c
@@ -15,6 +15,7 @@
 
 /* strlcpy() needs _DARWIN_C_SOURCE */
 #define _DARWIN_C_SOURCE
+#define _BSD_SOURCE
 
 #include <stdio.h>
 #include <stdlib.h>
--- a/gpsd_json.c
+++ b/gpsd_json.c
@@ -24,6 +24,7 @@ PERMISSIONS
 #define __DARWIN_C_LEVEL 200112L
 /* strlcpy() needs _DARWIN_C_SOURCE */
 #define _DARWIN_C_SOURCE
+#define _BSD_SOURCE
 
 #include <stdio.h>
 #include <math.h>
--- a/libgpsd_core.c
+++ b/libgpsd_core.c
@@ -20,6 +20,7 @@
 
 /* strlcpy() needs _DARWIN_C_SOURCE */
 #define _DARWIN_C_SOURCE
+#define _BSD_SOURCE
 
 #include <time.h>
 #include <stdio.h>
