--- a/setup.py
+++ b/setup.py
@@ -71,15 +71,6 @@ data = dict(
         version='1.1.10',
         url='https://bitbucket.org/stoneleaf/enum34',
         packages=packages,
-        package_data={
-            'enum' : [
-                'LICENSE',
-                'README',
-                'doc/enum.rst',
-                'doc/enum.pdf',
-                'test.py',
-                ]
-            },
         license='BSD License',
         description='Python 3.4 Enum backported to 3.3, 3.2, 3.1, 2.7, 2.6, 2.5, and 2.4',
         long_description=long_desc,
