--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
 DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
 SUBDIRS = \
-	gst-libs gst ext docs m4 tests common pkgconfig po
+	gst-libs gst ext docs m4 tests common pkgconfig
 	
 EXTRA_DIST = \
 	depcomp \
--- a/configure.ac
+++ b/configure.ac
@@ -427,7 +427,6 @@ tests/Makefile
 tests/check/Makefile
 tests/files/Makefile
 m4/Makefile
-po/Makefile.in
 pkgconfig/Makefile
 pkgconfig/gstreamer-plugins-ugly-uninstalled.pc
 )
