diff -u --recursive gst-plugins-ugly-1.12.4-vanilla/configure.ac gst-plugins-ugly-1.12.4/configure.ac
--- gst-plugins-ugly-1.12.4-vanilla/configure.ac	2018-02-11 20:19:01.875710150 -0500
+++ gst-plugins-ugly-1.12.4/configure.ac	2018-02-11 20:19:27.117658780 -0500
@@ -470,9 +470,6 @@
 docs/Makefile
 docs/plugins/Makefile
 docs/version.entities
-tests/Makefile
-tests/check/Makefile
-tests/files/Makefile
 m4/Makefile
 pkgconfig/Makefile
 pkgconfig/gstreamer-plugins-ugly-uninstalled.pc
diff -u --recursive gst-plugins-ugly-1.12.4-vanilla/Makefile.am gst-plugins-ugly-1.12.4/Makefile.am
--- gst-plugins-ugly-1.12.4-vanilla/Makefile.am	2018-02-11 20:19:01.875710150 -0500
+++ gst-plugins-ugly-1.12.4/Makefile.am	2018-02-11 20:19:34.402643954 -0500
@@ -1,7 +1,7 @@
 DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
 SUBDIRS = \
-	gst-libs gst ext docs m4 tests common pkgconfig
+	gst-libs gst ext docs m4 common pkgconfig
 	
 EXTRA_DIST = \
 	depcomp \
