--- a/configure.ac
+++ b/configure.ac
@@ -475,7 +475,6 @@ docs/version.entities
 tests/Makefile
 tests/check/Makefile
 m4/Makefile
-po/Makefile.in
 pkgconfig/Makefile
 pkgconfig/gstreamer-plugins-ugly-uninstalled.pc
 gst-plugins-ugly.spec
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
 DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
 SUBDIRS = \
-	gst-libs gst ext m4 common pkgconfig po
+	gst-libs gst ext m4 common pkgconfig
 
 # include before EXTRA_DIST for win32 assignment
 include $(top_srcdir)/common/win32.mak
--- a/Makefile.in
+++ b/Makefile.in
@@ -502,7 +502,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 SUBDIRS = \
-	gst-libs gst ext m4 common pkgconfig po
+	gst-libs gst ext m4 common pkgconfig
 
 
 # the MANIFEST contains all win32 related files that should be disted
