--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
 DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
 SUBDIRS = \
-	gst-libs gst ext docs m4 common pkgconfig
+	gst-libs gst ext m4 common pkgconfig
 	
 EXTRA_DIST = \
 	depcomp \
--- a/configure.ac
+++ b/configure.ac
@@ -420,9 +420,6 @@ ext/dvdread/Makefile
 ext/mpeg2dec/Makefile
 ext/sidplay/Makefile
 ext/x264/Makefile
-docs/Makefile
-docs/plugins/Makefile
-docs/version.entities
 m4/Makefile
 pkgconfig/Makefile
 pkgconfig/gstreamer-plugins-ugly-uninstalled.pc
