--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -44,6 +44,7 @@
 sasl_version = 2:23:0
 
 INCLUDES=-I$(top_srcdir)/include -I$(top_srcdir)/plugins -I$(top_builddir)/include -I$(top_srcdir)/sasldb
+SASL_STATIC_SRC = $(patsubst %.o, %.c, $(SASL_STATIC_OBJS))
 
 EXTRA_DIST = windlopen.c staticopen.h NTMakefile
 EXTRA_LIBRARIES = libsasl2.a
@@ -93,7 +94,7 @@ libsasl2.a: libsasl2.la $(SASL_STATIC_OB
 	rm -f $@
 	ln -s .libs/$@ $@
 
-$(SASL_STATIC_OBJS): linksrcs
+$(SASL_STATIC_SRC): linksrcs
 
 linksrcs:
 	-ln -s $(SASL_STATIC_SRCS) .
--- a/lib/Makefile.in
+++ b/lib/Makefile.in
@@ -176,6 +176,7 @@ SASL_KRB_LIB = @SASL_KRB_LIB@
 SASL_MECHS = @SASL_MECHS@
 SASL_STATIC_LIBS = @SASL_STATIC_LIBS@
 SASL_STATIC_OBJS = @SASL_STATIC_OBJS@
+SASL_STATIC_SRC = $(patsubst %.o, %.c, $(SASL_STATIC_OBJS)) 
 SASL_STATIC_SRCS = @SASL_STATIC_SRCS@
 SASL_UTIL_HEADERS_EXTRA = @SASL_UTIL_HEADERS_EXTRA@
 SASL_UTIL_LIBS_EXTRA = @SASL_UTIL_LIBS_EXTRA@
@@ -646,7 +647,7 @@ libsasl2.a: libsasl2.la $(SASL_STATIC_OB
 	rm -f $@
 	ln -s .libs/$@ $@
 
-$(SASL_STATIC_OBJS): linksrcs
+$(SASL_STATIC_SRC): linksrcs
 
 linksrcs:
 	-ln -s $(SASL_STATIC_SRCS) .
--- a/plugins/Makefile.am
+++ b/plugins/Makefile.am
@@ -82,12 +82,12 @@ EXTRA_LTLIBRARIES = libplain.la libanony
 	libntlm.la libpassdss.la libsasldb.la libsql.la libldapdb.la
 
 libplain_la_SOURCES = plain.c plain_init.c $(common_sources)
-libplain_la_LDFLAGS = -version-info $(plain_version)
+libplain_la_LDFLAGS = -version-info $(plain_version) $(AM_LDFLAGS)
 libplain_la_DEPENDENCIES = $(COMPAT_OBJS)
 libplain_la_LIBADD = $(PLAIN_LIBS) $(COMPAT_OBJS)
 
 libanonymous_la_SOURCES = anonymous.c anonymous_init.c $(common_sources)
-libanonymous_la_LDFLAGS = -version-info $(anonymous_version)
+libanonymous_la_LDFLAGS = -version-info $(anonymous_version) $(AM_LDFLAGS)
 libanonymous_la_DEPENDENCIES = $(COMPAT_OBJS)
 libanonymous_la_LIBADD = $(COMPAT_OBJS)
 
@@ -97,58 +97,58 @@ libkerberos4_la_DEPENDENCIES = $(COMPAT_
 libkerberos4_la_LIBADD = $(SASL_KRB_LIB) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 libgssapiv2_la_SOURCES = gssapi.c gssapiv2_init.c $(common_sources)
-libgssapiv2_la_LDFLAGS = -version-info $(gssapiv2_version)
+libgssapiv2_la_LDFLAGS = -version-info $(gssapiv2_version) $(AM_LDFLAGS)
 libgssapiv2_la_DEPENDENCIES = $(COMPAT_OBJS)
 libgssapiv2_la_LIBADD = $(GSSAPIBASE_LIBS) $(GSSAPI_LIBS) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 libcrammd5_la_SOURCES = cram.c crammd5_init.c $(common_sources)
-libcrammd5_la_LDFLAGS = -version-info $(crammd5_version)
+libcrammd5_la_LDFLAGS = -version-info $(crammd5_version) $(AM_LDFLAGS)
 libcrammd5_la_DEPENDENCIES = $(COMPAT_OBJS)
 libcrammd5_la_LIBADD = $(COMPAT_OBJS)
 
 libdigestmd5_la_SOURCES = digestmd5.c digestmd5_init.c $(common_sources)
-libdigestmd5_la_LDFLAGS = -version-info $(digestmd5_version)
+libdigestmd5_la_LDFLAGS = -version-info $(digestmd5_version) $(AM_LDFLAGS)
 libdigestmd5_la_DEPENDENCIES = $(COMPAT_OBJS)
 libdigestmd5_la_LIBADD = $(LIB_DES) $(LIB_SOCKET) $(COMPAT_OBJS)
 
 liblogin_la_SOURCES = login.c login_init.c $(common_sources)
-liblogin_la_LDFLAGS = -version-info $(login_version)
+liblogin_la_LDFLAGS = -version-info $(login_version) $(AM_LDFLAGS)
 liblogin_la_DEPENDENCIES = $(COMPAT_OBJS)
 liblogin_la_LIBADD = $(PLAIN_LIBS) $(COMPAT_OBJS)
 
 libsrp_la_SOURCES = srp.c srp_init.c $(common_sources)
-libsrp_la_LDFLAGS = -version-info $(srp_version)
+libsrp_la_LDFLAGS = -version-info $(srp_version) $(AM_LDFLAGS)
 libsrp_la_DEPENDENCIES = $(COMPAT_OBJS)
 libsrp_la_LIBADD = $(SRP_LIBS) $(COMPAT_OBJS)
 
 libotp_la_SOURCES = otp.c otp_init.c otp.h $(common_sources)
-libotp_la_LDFLAGS = -version-info $(otp_version)
+libotp_la_LDFLAGS = -version-info $(otp_version) $(AM_LDFLAGS)
 libotp_la_DEPENDENCIES = $(COMPAT_OBJS)
 libotp_la_LIBADD = $(OTP_LIBS) $(COMPAT_OBJS)
 
 libntlm_la_SOURCES = ntlm.c ntlm_init.c $(common_sources)
-libntlm_la_LDFLAGS = -version-info $(ntlm_version)
+libntlm_la_LDFLAGS = -version-info $(ntlm_version) $(AM_LDFLAGS)
 libntlm_la_DEPENDENCIES = $(COMPAT_OBJS)
 libntlm_la_LIBADD = $(NTLM_LIBS) $(COMPAT_OBJS)
 
 libpassdss_la_SOURCES = passdss.c passdss_init.c $(common_sources)
-libpassdss_la_LDFLAGS = -version-info $(passdss_version)
+libpassdss_la_LDFLAGS = -version-info $(passdss_version) $(AM_LDFLAGS)
 libpassdss_la_DEPENDENCIES = $(COMPAT_OBJS)
 libpassdss_la_LIBADD = $(PASSDSS_LIBS) $(COMPAT_OBJS)
 
 # Auxprop Plugins
 libsasldb_la_SOURCES = sasldb.c sasldb_init.c $(common_sources)
-libsasldb_la_LDFLAGS = -version-info $(sasldb_version)
+libsasldb_la_LDFLAGS = -version-info $(sasldb_version) $(AM_LDFLAGS)
 libsasldb_la_DEPENDENCIES = $(COMPAT_OBJS)
 libsasldb_la_LIBADD = ../sasldb/libsasldb.la $(SASL_DB_LIB) $(COMPAT_OBJS)
 
 libldapdb_la_SOURCES = ldapdb.c ldapdb_init.c $(common_sources)
-libldapdb_la_LDFLAGS = $(LIB_LDAP) -version-info $(ldapdb_version)
+libldapdb_la_LDFLAGS = $(LIB_LDAP) -version-info $(ldapdb_version) $(AM_LDFLAGS)
 libldapdb_la_DEPENDENCIES = $(COMPAT_OBJS)
 libldapdb_la_LIBADD = $(COMPAT_OBJS)
 
 libsql_la_SOURCES = sql.c sql_init.c $(common_sources)
-libsql_la_LDFLAGS = $(LIB_MYSQL) $(LIB_PGSQL) $(LIB_SQLITE) -version-info $(sql_version)
+libsql_la_LDFLAGS = $(LIB_MYSQL) $(LIB_PGSQL) $(LIB_SQLITE) -version-info $(sql_version) $(AM_LDFLAGS)
 libsql_la_DEPENDENCIES = $(COMPAT_OBJS)
 libsql_la_LIBADD = $(COMPAT_OBJS)
 
