--- a/lib/saslutil.c
+++ b/lib/saslutil.c
@@ -59,9 +59,7 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-#ifdef HAVE_TIME_H
 #include <time.h>
-#endif
 #include "saslint.h"
 #include <saslutil.h>
 
--- a/plugins/cram.c
+++ b/plugins/cram.c
@@ -48,6 +48,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <time.h>
 #ifndef macintosh
 #include <sys/stat.h>
 #endif
--- a/plugins/digestmd5.c
+++ b/plugins/digestmd5.c
@@ -50,6 +50,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <time.h>
 #ifndef macintosh
 #include <sys/types.h>
 #include <sys/stat.h>
