diff -rupN libffi-3.2.1.orig/configure libffi-3.2.1/configure
--- libffi-3.2.1.orig/configure	2014-11-12 12:59:57.000000000 +0100
+++ libffi-3.2.1/configure	2016-05-11 17:14:58.606625260 +0200
@@ -18725,12 +18725,6 @@ if test "x$GCC" = "xyes"; then
     toolexecdir="${libdir}"/gcc-lib/'$(target_alias)'
     toolexeclibdir="${libdir}"
   fi
-  multi_os_directory=`$CC $CFLAGS -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
-
 else
   toolexeclibdir="${libdir}"
 fi
diff -rupN libffi-3.2.1.orig/configure.ac libffi-3.2.1/configure.ac
--- libffi-3.2.1.orig/configure.ac	2014-11-12 12:56:51.000000000 +0100
+++ libffi-3.2.1/configure.ac	2016-05-11 17:15:19.694626266 +0200
@@ -601,11 +601,6 @@ if test "x$GCC" = "xyes"; then
     toolexecdir="${libdir}"/gcc-lib/'$(target_alias)'
     toolexeclibdir="${libdir}"
   fi
-  multi_os_directory=`$CC $CFLAGS -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
   AC_SUBST(toolexecdir)
 else
   toolexeclibdir="${libdir}"
