diff -Naur luaposix-release-v33.0.0.orig/ext/posix/unistd.c luaposix-release-v33.0.0/ext/posix/unistd.c
--- luaposix-release-v33.0.0.orig/ext/posix/unistd.c	2014-11-04 17:49:35.000000000 +0200
+++ luaposix-release-v33.0.0/ext/posix/unistd.c	2014-12-14 00:59:14.878622722 +0200
@@ -526,6 +526,7 @@
 #endif
 
 
+#ifndef NO_GETLOGIN
 /***
 Current logged-in user.
 @treturn[1] string username, if successful
@@ -538,6 +539,7 @@
 	checknargs(L, 0);
 	return pushstringresult(getlogin());
 }
+#endif
 
 
 /***
@@ -1037,7 +1039,9 @@
 	LPOSIX_FUNC( Pgetegid		),
 	LPOSIX_FUNC( Pgeteuid		),
 	LPOSIX_FUNC( Pgetgid		),
+#ifndef NO_GETLOGIN
 	LPOSIX_FUNC( Pgetlogin		),
+#endif
 	LPOSIX_FUNC( Pgetpgrp		),
 	LPOSIX_FUNC( Pgetpid		),
 	LPOSIX_FUNC( Pgetppid		),
