--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@ OPT_IR      = -DIR
 SOURCES = \
 	main.c slimproto.c buffer.c stream.c utils.c \
 	output.c output_alsa.c output_pa.c output_stdout.c output_pack.c decode.c \
-	flac.c pcm.c mad.c vorbis.c faad.c mpg.c
+	flac.c pcm.c vorbis.c faad.c mpg.c
 
 SOURCES_DSD      = dsd.c dop.c dsd2pcm/dsd2pcm.c
 SOURCES_FF       = ffmpeg.c
@@ -25,7 +25,7 @@ SOURCES_IR       = ir.c
 
 LINK_LINUX       = -ldl
 
-LINKALL          = -lFLAC -lmad -lvorbisfile -lfaad -lmpg123
+LINKALL          = -lFLAC -lvorbisfile -lfaad -lmpg123
 LINKALL_FF       = -lavcodec -lavformat -lavutil
 LINKALL_RESAMPLE = -lsoxr
 LINKALL_IR       = -llirc_client
--- a/decode.c
+++ b/decode.c
@@ -145,8 +145,8 @@ void decode_init(log_level level, const
 	if (!strstr(exclude_codecs, "pcm")  && (!include_codecs || strstr(include_codecs, "pcm")))  codecs[i++] = register_pcm();
 
 	// try mad then mpg for mp3 unless command line option passed
-	if (!(strstr(exclude_codecs, "mp3") || strstr(exclude_codecs, "mad")) &&
-		(!include_codecs || strstr(include_codecs, "mp3") || strstr(include_codecs, "mad")))	codecs[i] = register_mad();
+//	if (!(strstr(exclude_codecs, "mp3") || strstr(exclude_codecs, "mad")) &&
+//		(!include_codecs || strstr(include_codecs, "mp3") || strstr(include_codecs, "mad")))	codecs[i] = register_mad();
 	if (!(strstr(exclude_codecs, "mp3") || strstr(exclude_codecs, "mpg")) && !codecs[i] &&
 		(!include_codecs || strstr(include_codecs, "mp3") || strstr(include_codecs, "mpg")))    codecs[i] = register_mpg();
 
--- a/main.c
+++ b/main.c
@@ -39,7 +39,8 @@
 #else
 #define CODECS_DSD  ""
 #endif
-#define CODECS_MP3  " (mad,mpg for specific mp3 codec)"
+//#define CODECS_MP3  " (mad,mpg for specific mp3 codec)"
+#define CODECS_MP3  " (mpg123 for specific mp3 codec)"
 
 #define CODECS CODECS_BASE CODECS_FF CODECS_DSD CODECS_MP3
 
--- a/squeezelite.h
+++ b/squeezelite.h
@@ -154,7 +154,7 @@
 
 #if LINUX
 #define LIBFLAC "libFLAC.so.8"
-#define LIBMAD  "libmad.so.0"
+//#define LIBMAD  "libmad.so.0"
 #define LIBMPG "libmpg123.so.0"
 #define LIBVORBIS "libvorbisfile.so.3"
 #define LIBTREMOR "libvorbisidec.so.1"
@@ -168,7 +168,7 @@
 
 #if OSX
 #define LIBFLAC "libFLAC.8.dylib"
-#define LIBMAD  "libmad.0.dylib"
+//#define LIBMAD  "libmad.0.dylib"
 #define LIBMPG "libmpg123.0.dylib"
 #define LIBVORBIS "libvorbisfile.3.dylib"
 #define LIBTREMOR "libvorbisidec.1.dylib"
@@ -181,7 +181,7 @@
 
 #if WIN
 #define LIBFLAC "libFLAC.dll"
-#define LIBMAD  "libmad-0.dll"
+//#define LIBMAD  "libmad-0.dll"
 #define LIBMPG "libmpg123-0.dll"
 #define LIBVORBIS "libvorbisfile.dll"
 #define LIBTREMOR "libvorbisidec.dll"
@@ -194,7 +194,7 @@
 
 #if FREEBSD
 #define LIBFLAC "libFLAC.so.11"
-#define LIBMAD  "libmad.so.2"
+//#define LIBMAD  "libmad.so.2"
 #define LIBMPG "libmpg123.so.0"
 #define LIBVORBIS "libvorbisfile.so.6"
 #define LIBTREMOR "libvorbisidec.so.1"
