--- a/resource/SConscript
+++ b/resource/SConscript
@@ -67,7 +67,7 @@ if target_os == 'linux':
 	SConscript('csdk/stack/samples/linux/secure/SConscript')
 
 	# Build C/C++ unit tests
-	SConscript('unit_tests.scons')
+	# SConscript('unit_tests.scons')
 
 elif target_os == 'darwin':
 	env.Command('../../out/darwin/iotivity-csdk.framework',None,src_dir + '/tools/darwin/mkfwk_osx.sh')
--- a/service/resource-encapsulation/SConscript
+++ b/service/resource-encapsulation/SConscript
@@ -101,7 +101,7 @@ SConscript('examples/SConscript')
 ######################################################################
 # Build UnitTests Resource Client , resourceCache and resourceBroker
 ################################################ ######################
-SConscript('unittests/SConscript')
-SConscript('src/resourceCache/unittests/SConscript')
-SConscript('src/resourceBroker/unittest/SConscript')
+#SConscript('unittests/SConscript')
+#SConscript('src/resourceCache/unittests/SConscript')
+#SConscript('src/resourceBroker/unittest/SConscript')
 
