--- a/service/protocol-plugin/sample-app/linux/SConscript
+++ b/service/protocol-plugin/sample-app/linux/SConscript
@@ -41,7 +41,7 @@ if target_os not in ['windows', 'winrt']
 
 sample_env.AppendUnique(LIBS = ['oc', 'oc_logger', 'octbstack',
                                 'connectivity_abstraction', 'coap',
-                                'ppm', 'pmimpl', 'dl'])
+                                'ppm', 'boost_system', 'pmimpl', 'dl'])
 
 if env.get('SECURED') == '1':
     sample_env.AppendUnique(LIBS = ['tinydtls'])
