--- a/service/things-manager/sampleapp/linux/configuration/SConscript
+++ b/service/things-manager/sampleapp/linux/configuration/SConscript
@@ -57,7 +57,7 @@ conserver = linux_sample_env.Program('co
 conclient = linux_sample_env.Program('con-client', 'con-client.cpp')
 bootstrapserver = linux_sample_env.Program('bootstrapserver', 'bootstrapserver.cpp')
 Alias("ConServerApp", conserver)
-Alias("ConCleintApp", conclient)
+Alias("ConClientApp", conclient)
 Alias("BootstrapServerApp", bootstrapserver)
 env.AppendTarget('ConServerApp')
 env.AppendTarget('ConClientApp')
