diff --git a/compat/b64_ntop.c b/compat/b64_ntop.c
index 2b4dc2d..a1314ca 100644
--- a/compat/b64_ntop.c
+++ b/compat/b64_ntop.c
@@ -122,7 +122,7 @@ static const char Pad64 = '=';
    */
 
 int
-b64_ntop(uint8_t const *src, size_t srclength, char *target, size_t targsize) {
+local_b64_ntop(uint8_t const *src, size_t srclength, char *target, size_t targsize) {
 	size_t datalength = 0;
 	uint8_t input[3];
 	uint8_t output[4];
