diff -Nur a/setup.py b/setup.py
--- a/setup.py	2015-11-29 17:31:14.000000000 +0800
+++ b/setup.py	2015-12-07 18:39:00.863378684 +0800
@@ -13,15 +13,6 @@
        version='1.1.1',
        url='https://pypi.python.org/pypi/enum34',
        packages=['enum'],
-       package_data={
-           'enum' : [
-               'LICENSE',
-               'README',
-               'doc/enum.rst',
-               'doc/enum.pdf',
-               'test_enum.py',
-               ]
-           },
        license='BSD License',
        description='Python 3.4 Enum backported to 3.3, 3.2, 3.1, 2.7, 2.6, 2.5, and 2.4',
        long_description=long_desc,
