From 8f62d432745a65671332281363409229c238cd1e Mon Sep 17 00:00:00 2001
From: Martin Schneider <martschneider@google.com>
Date: Mon, 25 Oct 2021 10:53:01 +0800
Subject: [PATCH] Add install target to CMakeLists.txt

Signed-off-by: Martin Schneider <martschneider@google.com>
---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -649,3 +649,5 @@ add_custom_target(
     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
     DEPENDS all_tests bssl_shim handshaker fips_specific_tests_if_any
     USES_TERMINAL)
+
+install(TARGETS ssl crypto DESTINATION bin)
