--- a/src/libsodium/Makefile.am
+++ b/src/libsodium/Makefile.am
@@ -12,6 +12,7 @@ libsodium_la_SOURCES = \
 	crypto_box/crypto_box_easy.c \
 	crypto_box/crypto_box_seal.c \
 	crypto_box/curve25519xsalsa20poly1305/box_curve25519xsalsa20poly1305.c \
+	crypto_core/ed25519/core_ed25519.c \
 	crypto_core/ed25519/ref10/ed25519_ref10.c \
 	crypto_core/hchacha20/core_hchacha20.c \
 	crypto_core/hsalsa20/ref2/core_hsalsa20_ref2.c \
@@ -152,7 +153,6 @@ if !MINIMAL
 libsodium_la_SOURCES += \
 	crypto_box/curve25519xchacha20poly1305/box_curve25519xchacha20poly1305.c \
 	crypto_box/curve25519xchacha20poly1305/box_seal_curve25519xchacha20poly1305.c \
-	crypto_core/ed25519/core_ed25519.c \
 	crypto_core/ed25519/core_ristretto255.c \
 	crypto_pwhash/scryptsalsa208sha256/crypto_scrypt-common.c \
 	crypto_pwhash/scryptsalsa208sha256/crypto_scrypt.h \
--- a/src/libsodium/include/sodium.h
+++ b/src/libsodium/include/sodium.h
@@ -14,6 +14,7 @@
 #include "sodium/crypto_auth_hmacsha512256.h"
 #include "sodium/crypto_box.h"
 #include "sodium/crypto_box_curve25519xsalsa20poly1305.h"
+#include "sodium/crypto_core_ed25519.h"
 #include "sodium/crypto_core_hsalsa20.h"
 #include "sodium/crypto_core_hchacha20.h"
 #include "sodium/crypto_core_salsa20.h"
@@ -56,7 +57,6 @@
 
 #ifndef SODIUM_LIBRARY_MINIMAL
 # include "sodium/crypto_box_curve25519xchacha20poly1305.h"
-# include "sodium/crypto_core_ed25519.h"
 # include "sodium/crypto_core_ristretto255.h"
 # include "sodium/crypto_scalarmult_ristretto255.h"
 # include "sodium/crypto_secretbox_xchacha20poly1305.h"
