--- a/configure.ac
+++ b/configure.ac
@@ -3334,7 +3334,6 @@ ext/timidity/Makefile
 ext/webp/Makefile
 ext/xvid/Makefile
 ext/zbar/Makefile
-po/Makefile.in
 docs/Makefile
 docs/plugins/Makefile
 docs/libs/Makefile
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,11 +2,11 @@ DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-d
 
 SUBDIRS = \
 	gst-libs gst sys ext pkgconfig \
-	m4 common docs tests po tools
+	m4 common docs tests tools
 
 DIST_SUBDIRS = \
 	gst gst-libs sys ext pkgconfig \
-	m4 common docs tests po tools
+	m4 common docs tests tools
 
 # include before EXTRA_DIST for win32 assignment
 include $(top_srcdir)/common/win32.mak
--- a/Makefile.in
+++ b/Makefile.in
@@ -739,11 +739,11 @@ wayland_scanner = @wayland_scanner@
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 SUBDIRS = \
 	gst-libs gst sys ext pkgconfig \
-	m4 common docs tests po tools
+	m4 common docs tests tools
 
 DIST_SUBDIRS = \
 	gst gst-libs sys ext pkgconfig \
-	m4 common docs tests po tools
+	m4 common docs tests tools
 
 
 # the MANIFEST contains all win32 related files that should be disted
