--- a/Makefile.am
+++ b/Makefile.am
@@ -2,11 +2,11 @@ DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
 SUBDIRS = \
 	gst-libs gst sys ext pkgconfig \
-	m4 common docs tests po tools
+	m4 common docs tests tools
 
 DIST_SUBDIRS = \
 	gst gst-libs sys ext pkgconfig \
-	m4 common docs tests po tools
+	m4 common docs tests tools
 
 EXTRA_DIST = \
 	depcomp \
--- a/configure.ac
+++ b/configure.ac
@@ -2686,7 +2686,6 @@ ext/dtls/Makefile
 ext/webrtc/Makefile
 ext/webrtcdsp/Makefile
 ext/ttml/Makefile
-po/Makefile.in
 docs/Makefile
 docs/plugins/Makefile
 docs/libs/Makefile
