--- a/Makefile.am
+++ b/Makefile.am
@@ -2,11 +2,11 @@ DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
 SUBDIRS = \
 	gst-libs gst sys ext pkgconfig \
-	m4 common docs tests tools
+	m4 common docs tools
 
 DIST_SUBDIRS = \
 	gst gst-libs sys ext pkgconfig \
-	m4 common docs tests tools
+	m4 common docs tools
 
 EXTRA_DIST = \
 	depcomp \
--- a/configure.ac
+++ b/configure.ac
@@ -2600,24 +2600,6 @@ sys/vdpau/Makefile
 sys/wasapi/Makefile
 sys/winks/Makefile
 sys/winscreencap/Makefile
-tests/Makefile
-tests/check/Makefile
-tests/files/Makefile
-tests/examples/Makefile
-tests/examples/avsamplesink/Makefile
-tests/examples/camerabin2/Makefile
-tests/examples/codecparsers/Makefile
-tests/examples/compositor/Makefile
-tests/examples/directfb/Makefile
-tests/examples/audiomixmatrix/Makefile
-tests/examples/ipcpipeline/Makefile
-tests/examples/mpegts/Makefile
-tests/examples/mxf/Makefile
-tests/examples/opencv/Makefile
-tests/examples/uvch264/Makefile
-tests/examples/waylandsink/Makefile
-tests/examples/webrtc/Makefile
-tests/icles/Makefile
 ext/voamrwbenc/Makefile
 ext/voaacenc/Makefile
 ext/assrender/Makefile
