--- a/common/compat.c
+++ b/common/compat.c
@@ -97,8 +97,12 @@
 #include <unistd.h>
 
 #if defined (HAVE_PROGRAM_INVOCATION_SHORT_NAME) && !HAVE_DECL_PROGRAM_INVOCATION_SHORT_NAME
+#ifdef __UCLIBC__
+extern const char *program_invocation_short_name;
+#else
 extern char *program_invocation_short_name;
 #endif
+#endif
 
 #if defined (HAVE___PROGNAME) && !HAVE_DECL___PROGNAME
 extern char *__progname;
