diff --git a/main.c b/main.c
index b41a394..dd49f87 100644
--- a/main.c
+++ b/main.c
@@ -572,14 +572,6 @@ static pid_t do_cmd(char *cmd, char *machine, char *user, char **remote_argv, in
 			args[argc++] = "-l";
 			args[argc++] = user;
 		}
-#ifdef AF_INET
-		if (default_af_hint == AF_INET && strcmp(t, "ssh") == 0)
-			args[argc++] = "-4"; /* we're using ssh so we can add a -4 option */
-#endif
-#ifdef AF_INET6
-		if (default_af_hint == AF_INET6 && strcmp(t, "ssh") == 0)
-			args[argc++] = "-6"; /* we're using ssh so we can add a -6 option */
-#endif
 		args[argc++] = machine;
 #endif
 
