commit d23c0ea81e630af3cfda89aeeb52146c0c84c960
Author: Tobias Brunner <tobias@strongswan.org>
Date:   Mon May 2 09:31:49 2022 +0200

    enum: Fix compiler warning
    
    Closes strongswan/strongswan#1025

--- a/src/libstrongswan/utils/enum.c
+++ b/src/libstrongswan/utils/enum.c
@@ -97,7 +97,7 @@ char *enum_flags_to_string(enum_name_t *
 		return buf;
 	}
 
-	if (snprintf(buf, len, e->names[0]) >= len)
+	if (snprintf(buf, len, "%s", e->names[0]) >= len)
 	{
 		return NULL;
 	}
