--- a/setup.py
+++ b/setup.py
@@ -11,7 +11,7 @@ import sys
 from setuptools import find_packages, setup
 
 try:
-    from setuptools_rust import RustExtension
+    pass
 except ImportError:
     print(
         """
@@ -45,7 +45,7 @@ setuptools_rust = "setuptools-rust>=0.11
 install_requirements = ["cffi>=1.12"]
 setup_requirements = install_requirements + [setuptools_rust]
 
-if os.environ.get("CRYPTOGRAPHY_DONT_BUILD_RUST"):
+if True:
     rust_extensions = []
 else:
     rust_extensions = [
