Fixes https://rustsec.org/advisories/RUSTSEC-2023-0042.html

--- a/src/rust/Cargo.toml
+++ b/src/rust/Cargo.toml
@@ -15,7 +15,7 @@ cryptography-cffi = { path = "cryptograp
 cryptography-x509 = { path = "cryptography-x509" }
 cryptography-openssl = { path = "cryptography-openssl" }
 pem = "1.1"
-ouroboros = "0.15"
+ouroboros = "0.18"
 openssl = "0.10.54"
 openssl-sys = "0.9.88"
 foreign-types-shared = "0.1"
