--- a/m4/fd_passing.m4
+++ b/m4/fd_passing.m4
@@ -8,7 +8,7 @@ AC_DEFUN([DOVECOT_FD_PASSING], [
         CFLAGS="$CFLAGS -DBUGGY_CMSG_MACROS"
       ])
 
-      AC_RUN_IFELSE([AC_LANG_PROGRAM([[
+      AC_LINK_IFELSE([AC_LANG_PROGRAM([[
         #include <sys/types.h>
         #include <sys/socket.h>
         #include <sys/wait.h>
--- a/m4/glibc.m4
+++ b/m4/glibc.m4
@@ -2,7 +2,7 @@ AC_DEFUN([DOVECOT_GLIBC], [
   dnl * Old glibcs have broken posix_fallocate(). Make sure not to use it.
   dnl * It may also be broken in AIX.
   AC_CACHE_CHECK([whether posix_fallocate() works],i_cv_posix_fallocate_works,[
-    AC_RUN_IFELSE([AC_LANG_PROGRAM([[
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
       #define _XOPEN_SOURCE 600
       #include <stdio.h>
       #include <stdlib.h>
--- a/m4/ioloop.m4
+++ b/m4/ioloop.m4
@@ -4,7 +4,7 @@ AC_DEFUN([DOVECOT_IOLOOP], [
   
   AS_IF([test "$ioloop" = "best" || test "$ioloop" = "epoll"], [
     AC_CACHE_CHECK([whether we can use epoll],i_cv_epoll_works,[
-      AC_RUN_IFELSE([AC_LANG_PROGRAM([[
+      AC_LINK_IFELSE([AC_LANG_PROGRAM([[
         #include <sys/epoll.h>
       ]], [[
         return epoll_create(5) < 1;
--- a/m4/mmap_write.m4
+++ b/m4/mmap_write.m4
@@ -1,7 +1,7 @@
 dnl * If mmap() plays nicely with write()
 AC_DEFUN([DOVECOT_MMAP_WRITE], [
   AC_CACHE_CHECK([whether shared mmaps get updated by write()s],i_cv_mmap_plays_with_write,[
-    AC_RUN_IFELSE([AC_LANG_PROGRAM([[
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
       #include <stdio.h>
       #include <sys/types.h>
       #include <sys/stat.h>
--- a/m4/want_gssapi.m4
+++ b/m4/want_gssapi.m4
@@ -50,7 +50,7 @@ AC_DEFUN([DOVECOT_WANT_GSSAPI], [
 
           # does the kerberos library support SPNEGO?
           AC_CACHE_CHECK([whether GSSAPI supports SPNEGO],i_cv_gssapi_spnego,[
-            AC_RUN_IFELSE([AC_LANG_PROGRAM([[
+            AC_LINK_IFELSE([AC_LANG_PROGRAM([[
               #ifdef HAVE_GSSAPI_H
               #  include <gssapi.h>
               #else
