--- a/Makefile.in
+++ b/Makefile.in
@@ -2441,7 +2441,7 @@ unit_tests = $(am__append_35) unit/test-
 @CLIENT_TRUE@					monitor/uuid.h monitor/uuid.c
 
 @CLIENT_TRUE@client_bluetoothctl_LDADD = gdbus/libgdbus-internal.la @GLIB_LIBS@ @DBUS_LIBS@ \
-@CLIENT_TRUE@				-lreadline
+@CLIENT_TRUE@				-lreadline -lncurses
 
 @MONITOR_TRUE@monitor_btmon_SOURCES = monitor/main.c monitor/bt.h \
 @MONITOR_TRUE@				monitor/display.h monitor/display.c \
@@ -2691,13 +2691,13 @@ unit_tests = $(am__append_35) unit/test-
 @READLINE_TRUE@				client/display.h
 
 @READLINE_TRUE@attrib_gatttool_LDADD = lib/libbluetooth-internal.la \
-@READLINE_TRUE@			src/libshared-glib.la @GLIB_LIBS@ -lreadline
+@READLINE_TRUE@			src/libshared-glib.la @GLIB_LIBS@ -lreadline -lncurses
 
 @READLINE_TRUE@tools_obex_client_tool_SOURCES = $(gobex_sources) $(btio_sources) \
 @READLINE_TRUE@						tools/obex-client-tool.c
 
 @READLINE_TRUE@tools_obex_client_tool_LDADD = lib/libbluetooth-internal.la \
-@READLINE_TRUE@						@GLIB_LIBS@ -lreadline
+@READLINE_TRUE@						@GLIB_LIBS@ -lreadline -lncurses
 
 @READLINE_TRUE@tools_obex_server_tool_SOURCES = $(gobex_sources) $(btio_sources) \
 @READLINE_TRUE@						tools/obex-server-tool.c
@@ -2707,17 +2707,17 @@ unit_tests = $(am__append_35) unit/test-
 @READLINE_TRUE@				client/display.h client/display.c
 
 @READLINE_TRUE@tools_bluetooth_player_LDADD = gdbus/libgdbus-internal.la \
-@READLINE_TRUE@				@GLIB_LIBS@ @DBUS_LIBS@ -lreadline
+@READLINE_TRUE@				@GLIB_LIBS@ @DBUS_LIBS@ -lreadline -lncurses
 
 @READLINE_TRUE@tools_obexctl_SOURCES = tools/obexctl.c \
 @READLINE_TRUE@				client/display.h client/display.c
 
 @READLINE_TRUE@tools_obexctl_LDADD = gdbus/libgdbus-internal.la \
-@READLINE_TRUE@				@GLIB_LIBS@ @DBUS_LIBS@ -lreadline
+@READLINE_TRUE@				@GLIB_LIBS@ @DBUS_LIBS@ -lreadline -lncurses
 
 @READLINE_TRUE@tools_btmgmt_SOURCES = tools/btmgmt.c src/uuid-helper.c client/display.c
 @READLINE_TRUE@tools_btmgmt_LDADD = lib/libbluetooth-internal.la src/libshared-mainloop.la \
-@READLINE_TRUE@				-lreadline
+@READLINE_TRUE@				-lreadline -lncurses
 
 @EXPERIMENTAL_TRUE@tools_gatt_service_SOURCES = tools/gatt-service.c
 @EXPERIMENTAL_TRUE@tools_gatt_service_LDADD = @GLIB_LIBS@ @DBUS_LIBS@ gdbus/libgdbus-internal.la
