From efb04936b8ad3bf313c289a522ae5832e22c1b7e Mon Sep 17 00:00:00 2001
From: Paul Newall <quandry@ntlworld.com>
Date: Tue, 15 Oct 2013 22:20:15 +0100
Subject: detection of cups commented out in configure.in it is no longer used
 by kodakaio.

---
 ChangeLog    |  4 ++++
 configure.in | 24 ++++++++++++------------
 2 files changed, 16 insertions(+), 12 deletions(-)

--- a/configure.in
+++ b/configure.in
@@ -349,14 +349,14 @@ if test -c /dev/urandom ; then
     AC_DEFINE(HAVE_DEV_URANDOM, 1, [Is /dev/urandom available?])
 fi
 
-AC_CHECK_HEADERS([cups/cups.h], [with_cups=yes], [with_cups=no])
-if test "x${with_cups}" = "xyes"; then
-  AC_DEFINE(HAVE_CUPS,[1],[Is cups/cups.h available?])
-  AC_PATH_PROG(CUPS_CONFIG, cups-config)
-  if test "x${CUPS_CONFIG}" != "x"; then
-    LIBS="$LIBS `$CUPS_CONFIG --libs`"
-  fi
-fi
+dnl AC_CHECK_HEADERS([cups/cups.h], [with_cups=yes], [with_cups=no])
+dnl if test "x${with_cups}" = "xyes"; then
+dnl   AC_DEFINE(HAVE_CUPS,[1],[Is cups/cups.h available?])
+dnl   AC_PATH_PROG(CUPS_CONFIG, cups-config)
+dnl   if test "x${CUPS_CONFIG}" != "x"; then
+dnl     LIBS="$LIBS `$CUPS_CONFIG --libs`"
+dnl   fi
+dnl fi
 
 AC_CHECK_HEADERS([systemd/sd-daemon.h])
 PKG_CHECK_MODULES([HAVE_SYSTEMD], [libsystemd-daemon], [with_systemd=yes], [with_systemd=no])
@@ -825,7 +825,7 @@ fi
 echo "IPv6 support:  `eval eval echo ${ipv6}`"
 echo "Avahi support: `eval eval echo ${enable_avahi}`"
 echo "SNMP support:  `eval eval echo ${with_snmp}`"
-echo "CUPS support:  `eval eval echo ${with_cups}`"
+dnl echo "CUPS support:  `eval eval echo ${with_cups}`"
 echo "-> The following backends will be built:"
 for backend in ${BACKENDS} ; do
   echo $ECHO_N "${backend} " 
