From e1934720c687ed8c6125c75ac658f55b4e1513ce Mon Sep 17 00:00:00 2001
From: Luiz Angelo Daros de Luca <luizluca@gmail.com>
Date: Sun, 2 Feb 2020 21:19:15 -0300
Subject: [PATCH] genesys: fix bigendian build

Signed-off-by: Luiz Angelo Daros de Luca <luizluca@gmail.com>
---
 backend/genesys/low.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/backend/genesys/low.cpp
+++ b/backend/genesys/low.cpp
@@ -539,7 +539,7 @@ Image read_unshuffled_image_from_scanner
     }
 
 #ifdef WORDS_BIGENDIAN
-    if (depth == 16) {
+    if (session.params.depth == 16) {
         dev->pipeline.push_node<ImagePipelineNodeSwap16BitEndian>();
     }
 #endif
