https://gitlab.com/sane-project/backends/issues/237

Index: sane-backends-1.0.31/backend/genesys/image_pipeline.cpp
===================================================================
--- sane-backends-1.0.31.orig/backend/genesys/image_pipeline.cpp
+++ sane-backends-1.0.31/backend/genesys/image_pipeline.cpp
@@ -754,7 +754,11 @@ bool ImagePipelineNodeCalibrate::get_nex
 
             float value_f = static_cast<float>(value) / max_value;
             value_f = (value_f - offset_[curr_calib_i]) * multiplier_[curr_calib_i];
+#if defined(__UCLIBC__)
+            value_f = round(value_f * max_value);
+#else
             value_f = std::round(value_f * max_value);
+#endif
             value = clamp<std::int32_t>(static_cast<std::int32_t>(value_f), 0, max_value);
             set_raw_channel_to_row(out_data, x, ch, value, format);
 
