--- a/plugins/broadmobi/mm-plugin-broadmobi.c
+++ b/plugins/broadmobi/mm-plugin-broadmobi.c
@@ -19,7 +19,6 @@
 #define _LIBMM_INSIDE_MM
 #include <libmm-glib.h>
 
-#include "mm-port-enums-types.h"
 #include "mm-log-object.h"
 #include "mm-plugin-broadmobi.h"
 #include "mm-broadband-modem.h"
--- a/plugins/dlink/mm-plugin-dlink.c
+++ b/plugins/dlink/mm-plugin-dlink.c
@@ -19,7 +19,6 @@
 #define _LIBMM_INSIDE_MM
 #include <libmm-glib.h>
 
-#include "mm-port-enums-types.h"
 #include "mm-log-object.h"
 #include "mm-plugin-dlink.h"
 #include "mm-broadband-modem.h"
--- a/plugins/meson.build
+++ b/plugins/meson.build
@@ -461,7 +461,7 @@ if plugins_options['huawei']
   plugins += {'plugin-huawei': {
     'plugin': true,
     'helper': {'sources': files('huawei/mm-modem-helpers-huawei.c'), 'include_directories': plugins_incs + [huawei_inc], 'c_args': common_c_args + ['-DMM_MODULE_NAME="huawei"']},
-    'module': {'sources': sources + enums_sources, 'include_directories': plugins_incs + [huawei_inc], 'c_args': common_c_args + ['-DMM_MODULE_NAME="huawei"']},
+    'module': {'sources': sources + enums_sources + port_enums_sources, 'include_directories': plugins_incs + [huawei_inc], 'c_args': common_c_args + ['-DMM_MODULE_NAME="huawei"']},
     'test': {'sources': files('huawei/tests/test-modem-helpers-huawei.c') + enums_sources, 'include_directories': huawei_inc, 'dependencies': libhelpers_dep},
   }}
 
--- a/plugins/telit/mm-plugin-telit.c
+++ b/plugins/telit/mm-plugin-telit.c
@@ -21,7 +21,6 @@
 #define _LIBMM_INSIDE_MM
 #include <libmm-glib.h>
 
-#include "mm-port-enums-types.h"
 #include "mm-log-object.h"
 #include "mm-modem-helpers.h"
 #include "mm-plugin-telit.h"
--- a/plugins/tplink/mm-plugin-tplink.c
+++ b/plugins/tplink/mm-plugin-tplink.c
@@ -19,7 +19,6 @@
 #define _LIBMM_INSIDE_MM
 #include <libmm-glib.h>
 
-#include "mm-port-enums-types.h"
 #include "mm-log-object.h"
 #include "mm-plugin-tplink.h"
 #include "mm-broadband-modem.h"
--- a/src/meson.build
+++ b/src/meson.build
@@ -147,15 +147,15 @@ endif
 
 enums_types = 'mm-port-enums-types'
 
-enums_sources = []
-enums_sources += gnome.mkenums(
+port_enums_sources = []
+port_enums_sources += gnome.mkenums(
   enums_types + '.c',
   sources: headers,
   c_template: build_aux_dir / enums_types + '.c.template',
   fhead: '#include "mm-port-enums-types.h"',
 )
 
-enums_sources += gnome.mkenums(
+port_enums_sources += gnome.mkenums(
   enums_types + '.h',
   sources: headers,
   h_template: build_aux_dir / enums_types + '.h.template',
@@ -165,13 +165,13 @@ enums_sources += gnome.mkenums(
 
 libport = static_library(
   'port',
-  sources: sources + enums_sources,
+  sources: sources + port_enums_sources,
   include_directories: top_inc,
   dependencies: deps + private_deps,
 )
 
 libport_dep = declare_dependency(
-  sources: enums_sources[1],
+  sources: port_enums_sources[1],
   include_directories: '.',
   dependencies: deps,
   link_with: libport,
