--- a/src/crelay.c
+++ b/src/crelay.c
@@ -678,10 +678,12 @@ int main(int argc, char *argv[])
       listen(sock, 5);
       syslog(LOG_DAEMON | LOG_NOTICE, "HTTP server listening on %s:%d\n", inet_ntoa(iface), port);      
 
+#if 0
       if (daemon(0, 0) == -1) {
 	      syslog(LOG_DAEMON | LOG_ERR, "Failed to daemonize: %s", strerror(errno));
 	      exit(EXIT_FAILURE);
       }
+#endif
 
       while (1)
       {
