--- a/hdparm.h
+++ b/hdparm.h
@@ -2,6 +2,7 @@
 
 //#undef __KERNEL_STRICT_NAMES
 #include <linux/types.h>
+#include <sys/types.h>
 
 #if !defined(__GNUC__) && !defined(__attribute__)
 #define __attribute__(x)
--- a/sysfs.c
+++ b/sysfs.c
@@ -12,6 +12,7 @@
 #include <stdio.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <limits.h>
 #include <dirent.h>
 #include <sys/stat.h>
 #include <linux/types.h>
