diff --git a/lib/mosquitto.c b/lib/mosquitto.c
index bd9dbd3..7314248 100644
--- a/lib/mosquitto.c
+++ b/lib/mosquitto.c
@@ -594,7 +594,6 @@ int mosquitto_publish(struct mosquitto *mosq, int *mid, const char *topic, int p
 		pthread_mutex_lock(&mosq->out_message_mutex);
 		_mosquitto_message_queue(mosq, message, mosq_md_out);
 		if(mosq->max_inflight_messages == 0 || mosq->inflight_messages < mosq->max_inflight_messages){
-			mosq->inflight_messages++;
 			if(qos == 1){
 				message->state = mosq_ms_wait_for_puback;
 			}else if(qos == 2){
