--- a/src/twisted/python/_setup.py
+++ b/src/twisted/python/_setup.py
@@ -176,11 +176,6 @@ class ConditionalExtension(Extension, ob
 # The C extensions used for Twisted.
 _EXTENSIONS = [
     ConditionalExtension(
-        "twisted.test.raiser",
-        sources=["src/twisted/test/raiser.c"],
-        condition=lambda _: _isCPython),
-
-    ConditionalExtension(
         "twisted.internet.iocpreactor.iocpsupport",
         sources=[
             "src/twisted/internet/iocpreactor/iocpsupport/iocpsupport.c",
@@ -285,12 +280,11 @@ def getSetupArgs(extensions=_EXTENSIONS,
         "incremental >= 16.10.1",
         "Automat >= 0.3.0",
         "hyperlink >= 17.1.1",
-        "PyHamcrest >= 1.9.0",
         "attrs >= 17.4.0",
     ]
 
     arguments.update(dict(
-        packages=find_packages("src"),
+        packages=find_packages("src", exclude=["*.test", "*.test.*"]),
         use_incremental=True,
         setup_requires=["incremental >= 16.10.1"],
         install_requires=requirements,
@@ -300,7 +294,7 @@ def getSetupArgs(extensions=_EXTENSIONS,
         cmdclass=command_classes,
         include_package_data=True,
         exclude_package_data={
-            "": ["*.c", "*.h", "*.pxi", "*.pyx", "build.bat"],
+            "": ["*.c", "*.h", "*.pxi", "*.pyx", "build.bat", "test/*"],
         },
         zip_safe=False,
         extras_require=_EXTRAS_REQUIRE,
