diff --git a/share/functions/prompt_hostname.fish b/share/functions/prompt_hostname.fish
index 4348bce2..8502ce3f 100644
--- a/share/functions/prompt_hostname.fish
+++ b/share/functions/prompt_hostname.fish
@@ -2,7 +2,7 @@
 # hostname command uses. So cache the answer so including it in the prompt doesn't make fish seem
 # slow.
 if not set -q __fish_prompt_hostname
-    set -g __fish_prompt_hostname (hostname | string split '.')[1]
+    set -g __fish_prompt_hostname (uname -n | string split '.')[1]
 end
 
 function prompt_hostname
diff --git a/share/tools/web_config/sample_prompts/pythonista.fish b/share/tools/web_config/sample_prompts/pythonista.fish
index 9529035c..57ffaf86 100644
--- a/share/tools/web_config/sample_prompts/pythonista.fish
+++ b/share/tools/web_config/sample_prompts/pythonista.fish
@@ -7,7 +7,7 @@ function fish_prompt
         set -g VIRTUAL_ENV_DISABLE_PROMPT true
     end
     set_color yellow
-    printf '%s' (whoami)
+    printf '%s' (id -un)
     set_color normal
     printf ' at '
 
diff --git a/share/tools/web_config/sample_prompts/screen_savvy.fish b/share/tools/web_config/sample_prompts/screen_savvy.fish
index 411a5501..5cdcfb69 100644
--- a/share/tools/web_config/sample_prompts/screen_savvy.fish
+++ b/share/tools/web_config/sample_prompts/screen_savvy.fish
@@ -2,8 +2,8 @@
 # author: Matthias
 function fish_prompt -d "Write out the prompt"
     if test -z $WINDOW
-        printf '%s%s@%s%s%s%s%s> ' (set_color yellow) (whoami) (set_color purple) (prompt_hostname) (set_color $fish_color_cwd) (prompt_pwd) (set_color normal)
+        printf '%s%s@%s%s%s%s%s> ' (set_color yellow) (id -un) (set_color purple) (prompt_hostname) (set_color $fish_color_cwd) (prompt_pwd) (set_color normal)
     else
-        printf '%s%s@%s%s%s(%s)%s%s%s> ' (set_color yellow) (whoami) (set_color purple) (prompt_hostname) (set_color white) (echo $WINDOW) (set_color $fish_color_cwd) (prompt_pwd) (set_color normal)
+        printf '%s%s@%s%s%s(%s)%s%s%s> ' (set_color yellow) (id -un) (set_color purple) (prompt_hostname) (set_color white) (echo $WINDOW) (set_color $fish_color_cwd) (prompt_pwd) (set_color normal)
     end
 end
diff --git a/share/tools/web_config/sample_prompts/terlar.fish b/share/tools/web_config/sample_prompts/terlar.fish
index d49ef340..59b3d267 100644
--- a/share/tools/web_config/sample_prompts/terlar.fish
+++ b/share/tools/web_config/sample_prompts/terlar.fish
@@ -6,7 +6,7 @@ function fish_prompt --description 'Write out the prompt'
 
     # User
     set_color $fish_color_user
-    echo -n (whoami)
+    echo -n (id -un)
     set_color normal
 
     echo -n '@'
