--- a/sources/common/src/utils/logging/syslogloglocation.cpp
+++ b/sources/common/src/utils/logging/syslogloglocation.cpp
@@ -35,8 +35,6 @@ SyslogLogLocation::SyslogLogLocation(Variant &configuration, string identifier,
 	_priorities[_WARNING_] = LOG_WARNING;
 	_priorities[_ERROR_] = LOG_ERR;
 	_priorities[_FATAL_] = LOG_ERR;
-	_priorities[_PROD_ACCESS_] = LOG_ERR;
-	_priorities[_PROD_ERROR_] = LOG_ERR;
 	_specificLevel = specificLevel;
 	_enforceLoggerName = (_configuration[CONF_LOG_APPENDER_NAME] != "");
 	_pDefualtFormatter = NULL;
