--- a/sources/common/include/common.h
+++ b/sources/common/include/common.h
@@ -20,6 +20,10 @@
 #ifndef _COMMON_H
 #define _COMMON_H
 
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/types.h>
+
 #include "defines.h"
 #include "platform/platform.h"
 #include "utils/utils.h"
--- a/sources/common/src/utils/logging/fileloglocation.cpp
+++ b/sources/common/src/utils/logging/fileloglocation.cpp
@@ -18,6 +18,10 @@
  */
 
 
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/types.h>
+
 #include "utils/logging/fileloglocation.h"
 #include "utils/lua/luautils.h"
 
