--- a/src/racoon/ipsec_doi.c
+++ b/src/racoon/ipsec_doi.c
@@ -3582,8 +3582,8 @@ ipsecdoi_checkid1(iph1)
 	    iph1->approval->authmethod == OAKLEY_ATTR_AUTH_METHOD_PSKEY) {
 		 if (id_b->type != IPSECDOI_ID_IPV4_ADDR
 		  && id_b->type != IPSECDOI_ID_IPV6_ADDR) {
-			plog(LLV_ERROR, LOCATION, NULL,
-				"Expecting IP address type in main mode, "
+			plog(LLV_WARNING, LOCATION, NULL,
+				"Expecting IP address type in main mode (RFC2409) , "
 				"but %s.\n", s_ipsecdoi_ident(id_b->type));
 			return ISAKMP_NTYPE_INVALID_ID_INFORMATION;
 		}
