--- a/Makefile.am
+++ b/Makefile.am
@@ -31,7 +31,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = doc include m4 man src examples test build objs
+SUBDIRS = include m4 src build objs
 
 EXTRA_DIST = \
 	COPYING.FDL \
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -30,11 +30,6 @@ SUBDIRS = \
 	flac \
 	metaflac \
 	$(XMMS_DIRS) \
-	test_grabbag \
-	test_libs_common \
-	test_libFLAC \
-	test_seeking \
-	test_streams \
 	utils \
 	$(CPPLIBS_DIRS)
 
