--- a/rpi-eeprom-config
+++ b/rpi-eeprom-config
@@ -129,8 +129,8 @@ def edit_config(eeprom=None):
     """
     Implements something like 'git commit' for editing EEPROM configs.
     """
-    # Default to nano if $EDITOR is not defined.
-    editor = 'nano'
+    # Default to vi if $EDITOR is not defined.
+    editor = 'vi'
     if 'EDITOR' in os.environ:
         editor = os.environ['EDITOR']
 
@@ -317,7 +317,7 @@ Operating modes:
 
    To cancel the pending update run 'sudo rpi-eeprom-update -r'
 
-   The default text editor is nano and may be overridden by setting the 'EDITOR'
+   The default text editor is vi and may be overridden by setting the 'EDITOR'
    environment variable and passing '-E' to 'sudo' to preserve the environment.
 
 See 'rpi-eeprom-update -h' for more information about the available EEPROM
