--- a/src/util/rand_xor.c
+++ b/src/util/rand_xor.c
@@ -24,6 +24,8 @@
 
 #if defined(__linux__)
 #include <sys/file.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <unistd.h>
 #include <fcntl.h>
 #else
--- ./src/gallium/winsys/svga/drm/vmw_screen.h.orig
+++ ./src/gallium/winsys/svga/drm/vmw_screen.h
@@ -34,7 +34,7 @@
 #ifndef VMW_SCREEN_H_
 #define VMW_SCREEN_H_
 
-
+#include <sys/stat.h>
 #include "pipe/p_compiler.h"
 #include "pipe/p_state.h"
 
--- a/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_winsys.h
+++ b/src/amd/vulkan/winsys/amdgpu/radv_amdgpu_winsys.h
@@ -32,6 +32,7 @@
 #include "ac_gpu_info.h"
 #include "addrlib/addrinterface.h"
 #include <amdgpu.h>
+#include <pthread.h>
 #include "util/list.h"
 
 struct radv_amdgpu_winsys {
