--- a/m4/ax_with_curses.m4
+++ b/m4/ax_with_curses.m4
@@ -72,7 +72,7 @@ AC_DEFUN([AX_WITH_CURSES],
        AC_CACHE_CHECK([for working ncursesw], ax_cv_ncursesw,
          [LIBS="$ax_save_LIBS -lncursesw"
           AC_TRY_LINK(
-            [#include <ncurses.h>],
+            [#include <ncursesw/ncurses.h>],
             [chtype a; int b=A_STANDOUT, c=KEY_LEFT; initscr(); ],
             ax_cv_ncursesw=yes, ax_cv_ncursesw=no)])
        if test "$ax_cv_ncursesw" = yes
