--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,9 +38,9 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
   add_compile_options(/utf-8)
 elseif(CYGWIN OR MINGW)
   # See https://stackoverflow.com/questions/38139631 for details.
-  add_compile_options(-std=gnu++11)
+  add_compile_options(-std=gnu++17)
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-  add_compile_options(-std=c++11)
+  add_compile_options(-std=c++17)
 endif()
 
 if(WIN32)
--- a/re2.pc
+++ b/re2.pc
@@ -4,5 +4,5 @@ libdir=@libdir@
 Name: re2
 Description: RE2 is a fast, safe, thread-friendly regular expression engine.
 Version: 0.0.0
-Cflags: -std=c++11 -pthread -I${includedir}
+Cflags: -std=c++17 -pthread -I${includedir}
 Libs: -pthread -L${libdir} -lre2
