From 6c7a281473083f7ac47b426b2127a8c8e76caca5 Mon Sep 17 00:00:00 2001
From: Quentin Armitage <quentin@armitage.org.uk>
Date: Sat, 25 Aug 2018 11:36:11 +0100
Subject: [PATCH] Include stdbool.h in process.c

Signed-off-by: Quentin Armitage <quentin@armitage.org.uk>
---
 keepalived/core/process.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/keepalived/core/process.c b/keepalived/core/process.c
index f11c32a3..22bcf254 100644
--- a/keepalived/core/process.c
+++ b/keepalived/core/process.c
@@ -30,6 +30,7 @@
 #ifdef _HAVE_SCHED_RT_
 #include <sched.h>
 #endif
+#include <stdbool.h>
 
 #include "process.h"
 #include "logger.h"
-- 
2.11.0

