From 042dc49f8781cbc914c06e99f15de94bd153d4e2 Mon Sep 17 00:00:00 2001
From: Quentin Armitage <quentin@armitage.org.uk>
Date: Sun, 10 Nov 2024 11:31:40 +0000
Subject: [PATCH] vrrp: fix reading of iproute2 conf files when directories
 don't exist

Signed-off-by: Quentin Armitage <quentin@armitage.org.uk>
---
 lib/rttables.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/lib/rttables.c
+++ b/lib/rttables.c
@@ -412,8 +412,9 @@ initialise_list(list_head_t *l, const ch
 
 			read_file(path, l, max);
 		}
+
+		closedir(dir);
 	}
-	closedir(dir);
 #endif
 
 	/* Now read the entries in the IPROUTE_ETC_DIR subdirectory */
@@ -425,8 +426,9 @@ initialise_list(list_head_t *l, const ch
 
 			read_file(path, l, max);
 		}
+
+		closedir(dir);
 	}
-	closedir(dir);
 
 	FREE_PTR(path);
 #ifdef IPROUTE_USR_DIR
