--- a/src/libknot/xdp/eth.c
+++ b/src/libknot/xdp/eth.c
@@ -20,6 +20,7 @@
 #include <linux/if.h>
 #include <linux/sockios.h>
 #include <sys/ioctl.h>
+#include <string.h>
 #include <unistd.h>
 
 #include "contrib/openbsd/strlcpy.h"
--- a/src/libknot/xdp/xdp.c
+++ b/src/libknot/xdp/xdp.c
@@ -16,10 +16,6 @@
 
 #include <assert.h>
 #include <errno.h>
-#include <linux/if_ether.h>
-#include <linux/ip.h>
-#include <linux/ipv6.h>
-#include <linux/udp.h>
 #include <stddef.h>
 #include <stdlib.h>
 #include <string.h>
@@ -32,6 +28,11 @@
 #include "libknot/xdp/xdp.h"
 #include "contrib/macros.h"
 
+#include <linux/if_ether.h>
+#include <linux/ip.h>
+#include <linux/ipv6.h>
+#include <linux/udp.h>
+
 /* Don't fragment flag. */
 #define	IP_DF 0x4000
 
