--- a/src/contrib/atomic.h
+++ b/src/contrib/atomic.h
@@ -41,6 +41,7 @@
 
  #include <stdint.h>
  #include <stdbool.h>
+ #include <stddef.h>
 
  #define ATOMIC_SET(dst, val)  __atomic_store_n(&(dst), (val), __ATOMIC_RELAXED)
  #define ATOMIC_GET(src)       __atomic_load_n(&(src), __ATOMIC_RELAXED)
@@ -58,6 +59,7 @@
 
  #include <stdint.h>
  #include <stdbool.h>
+ #include <stddef.h>
 
  #define ATOMIC_SET(dst, val)  ((dst) = (val))
  #define ATOMIC_GET(src)       (src)
