From 442633ae37f8a4e1164a2db3ad6b55bc738ba0b2 Mon Sep 17 00:00:00 2001
From: Daniel Salzman <daniel.salzman@nic.cz>
Date: Fri, 22 May 2020 12:50:29 +0200
Subject: [PATCH] configure: allow to manually disable POSIX capabilities

---
 configure.ac | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 47772799e..a08ca8532 100644
--- a/configure.ac
+++ b/configure.ac
@@ -554,7 +554,13 @@ AS_IF([test "$enable_utilities" = "yes"], [
   ])
 ]) # Knot DNS utilities dependencies
 
+AC_ARG_ENABLE([cap-ng],
+    AS_HELP_STRING([--enable-cap-ng=auto|no], [enable POSIX capabilities [default=auto]]),
+    [enable_cap_ng="$enableval"], [enable_cap_ng=auto])
+
 AS_IF([test "$enable_daemon" = "yes"], [
+
+AS_IF([test "$enable_cap_ng" != "no"],[
   PKG_CHECK_MODULES([cap_ng], [cap-ng], [enable_cap_ng=yes], [
     enable_cap_ng=no
     AC_CHECK_HEADER([cap-ng.h], [
@@ -570,7 +576,7 @@ AS_IF([test "$enable_daemon" = "yes"], [
 ], [
   enable_cap_ng=no
   cap_ng_LIBS=
-])
+])])
 
 AS_IF([test "$enable_cap_ng" = yes],
   [AC_DEFINE([ENABLE_CAP_NG], [1], [POSIX capabilities available])]
-- 
2.17.1

