From 4f02c29b67790ff05f23fc6593e145be7c223c70 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20H=C3=A1k?= <jan.hak@nic.cz>
Date: Tue, 23 Sep 2025 12:52:25 +0200
Subject: [PATCH] fix: failing build of knot/common/stats.c on PowerPC and MIPS

---
 src/knot/common/stats.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/src/knot/common/stats.c
+++ b/src/knot/common/stats.c
@@ -127,8 +127,8 @@ int stats_server(stats_dump_ctr_f fcn, s
 	}
 
 	DUMP_VAL(params, "zone-count", knot_zonedb_size(ctx->server->zone_db));
-	DUMP_VAL(params, "tcp-io-timeout", ctx->server->stats.tcp_io_timeout);
-	DUMP_VAL(params, "tcp-idle-timeout", ctx->server->stats.tcp_idle_timeout);
+	DUMP_VAL(params, "tcp-io-timeout", ATOMIC_GET(ctx->server->stats.tcp_io_timeout));
+	DUMP_VAL(params, "tcp-idle-timeout", ATOMIC_GET(ctx->server->stats.tcp_idle_timeout));
 
 	return KNOT_EOK;
 }
