--- a/src/knot/modules/rrl/kru.inc.c
+++ b/src/knot/modules/rrl/kru.inc.c
@@ -462,7 +462,6 @@ static inline bool kru_limited_update(st
 		load_at = (_Atomic uint16_t *)ctx->load;
 	}
 
-	static_assert(ATOMIC_CHAR16_T_LOCK_FREE == 2, "insufficient atomics");
 	const uint16_t price = ctx->price16;
 	const uint32_t limit = ctx->limit16;  // 2^16 has to be representable
 	uint16_t load_orig = atomic_load_explicit(load_at, memory_order_relaxed);
