diff --git a/src/zscanner/tests/TESTS b/src/zscanner/tests/TESTS
index c88aef1..4e2148f 100644
--- a/src/zscanner/tests/TESTS
+++ b/src/zscanner/tests/TESTS
@@ -18,10 +18,8 @@
 05-2_TTL
 05-3_TTL
 05-4_TTL
-06-0_INCLUDE
 06-1_INCLUDE
 06-2_INCLUDE
-06-3_INCLUDE
 06-4_INCLUDE
 06-5_INCLUDE
 06-6_INCLUDE
diff --git a/src/zscanner/tests/unittests.in b/src/zscanner/tests/unittests.in
index 846f351..272856c 100644
--- a/src/zscanner/tests/unittests.in
+++ b/src/zscanner/tests/unittests.in
@@ -1,9 +1,9 @@
 #!/bin/sh
 
-SOURCE=@top_srcdir@/src/zscanner/tests
-BUILD=@top_builddir@/src/zscanner/tests
+SOURCE="."
+BUILD="."
 
-. @top_srcdir@/libtap/tap/libtap.sh
+. ../tap/libtap.sh
 
 cd "$BUILD"
 
@@ -11,7 +11,7 @@ TMPDIR=$(test_tmpdir)
 TESTS_DIR="$SOURCE"/data
 ZSCANNER_TOOL="$BUILD"/zscanner-tool
 
-plan 71
+plan 69
 
 mkdir -p "$TMPDIR"/includes/
 for a in 1 2 3 4 5 6; do
