--- a/klish/ksession/kcontext.c
+++ b/klish/ksession/kcontext.c
@@ -4,6 +4,9 @@
 #include <assert.h>
 #include <sys/types.h>
 #include <unistd.h>
+#undef stdin
+#undef stdout
+#undef stderr
 
 #include <faux/str.h>
 #include <faux/conv.h>
--- a/klish/ksession/kexec.c
+++ b/klish/ksession/kexec.c
@@ -14,6 +14,9 @@
 #include <termios.h>
 #include <signal.h>
 #include <errno.h>
+#undef stdin
+#undef stdout
+#undef stderr
 
 #include <faux/list.h>
 #include <faux/buf.h>
