--- a/configure.ac
+++ b/configure.ac
@@ -859,10 +859,10 @@
 AC_ARG_WITH(xcode-sdk, AC_HELP_STRING([--with-xcode-sdk], 
  [Set xcode SDK version. Default is autodetect]), 
  [],[with_xcode_sdk="yes"])
 if test "x_$with_xcode_sdk" != "x_no" ; then
    # check OSX deployment target, if needed
-   if echo $build_os | grep darwin > /dev/null; then
+   if echo $target_os | grep darwin > /dev/null; then
      sdk_p=`xcode-select -print-path`;
      if test "x_$with_xcode_sdk" = "x_yes" ; then
         sdk_v="$( /usr/bin/xcrun --show-sdk-version 2>/dev/null )"
