Index: bind-9.10.4-P3/configure.in
===================================================================
--- bind-9.10.4-P3.orig/configure.in
+++ bind-9.10.4-P3/configure.in
@@ -167,26 +167,11 @@ esac
 #
 AC_CONFIG_FILES([make/rules make/includes])
 
-AC_PATH_PROG(AR, ar)
-ARFLAGS="cruv"
-AC_SUBST(AR)
-AC_SUBST(ARFLAGS)
-
 # The POSIX ln(1) program.  Non-POSIX systems may substitute
 # "copy" or something.
 LN=ln
 AC_SUBST(LN)
 
-case "$AR" in
-	"")
-		AC_MSG_ERROR([
-ar program not found.  Please fix your PATH to include the directory in
-which ar resides, or set AR in the environment with the full path to ar.
-])
-
-		;;
-esac
-
 #
 # Etags.
 #
