From 261c84d91d1b4581df9f7f0ec031908299de7726 Mon Sep 17 00:00:00 2001
From: Mark Andrews <marka@isc.org>
Date: Thu, 19 Dec 2019 09:27:44 +1100
Subject: [PATCH] fix variable name in conditional block

Origin: upstream, https://gitlab.isc.org/isc-projects/bind9/commit/261c84d91d1b4581df9f7f0ec031908299de7726

---
 lib/isc/stats.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/isc/stats.c b/lib/isc/stats.c
index 5bce3c1100..a7ab97ce53 100644
--- a/lib/isc/stats.c
+++ b/lib/isc/stats.c
@@ -297,7 +297,7 @@ setcounter(isc_stats_t *stats,
 	isc_atomic_store((int32_t *)&stats->counters[counter].lo,
 			 (uint32_t)(value & 0xffffffff));
 # else
-	stats->counters[counter] = val;
+	stats->counters[counter] = value;
 # endif
 #endif
 }
-- 
2.22.2

