--- a/Makefile
+++ b/Makefile
@@ -211,7 +211,7 @@ GV_SHA=407efb5dcdb0f4445935f7360535800b6
 default: all
 
 .PHONY: all
-all: binaries docs
+all: binaries
 
 .PHONY: binaries
 ifeq ($(shell uname -s),FreeBSD)
@@ -798,7 +798,7 @@ rpm-install: package  ## Install rpm pac
 	/usr/bin/podman info  # will catch a broken conmon
 
 .PHONY: install
-install: install.bin install.remote install.man install.systemd  ## Install binaries to system locations
+install: install.bin install.remote install.systemd  ## Install binaries to system locations
 
 .PHONY: install.catatonit
 install.catatonit:
