--- a/Makefile
+++ b/Makefile
@@ -224,7 +224,7 @@ GV_VERSION=v0.7.1
 default: all
 
 .PHONY: all
-all: binaries docs
+all: binaries
 
 .PHONY: binaries
 ifeq ($(shell uname -s),FreeBSD)
@@ -803,7 +803,7 @@ rpm-install: package  ## Install rpm pac
 	/usr/bin/podman info  # will catch a broken conmon
 
 .PHONY: install
-install: install.bin install.remote install.man install.systemd  ## Install binaries to system locations
+install: install.bin install.remote install.systemd  ## Install binaries to system locations
 
 .PHONY: install.catatonit
 install.catatonit:
