--- a/Makefile
+++ b/Makefile
@@ -219,7 +219,7 @@ GV_SHA=db608827124caa71ba411cec8ea959bb9
 default: all
 
 .PHONY: all
-all: binaries docs
+all: binaries
 
 .PHONY: binaries
 ifeq ($(shell uname -s),FreeBSD)
@@ -804,7 +804,7 @@ rpm-install: package  ## Install rpm pac
 	/usr/bin/podman info  # will catch a broken conmon
 
 .PHONY: install
-install: install.bin install.remote install.man install.systemd  ## Install binaries to system locations
+install: install.bin install.remote install.systemd  ## Install binaries to system locations
 
 .PHONY: install.catatonit
 install.catatonit:
