--- ./src/devices/devfreq.h.orig
+++ ./src/devices/devfreq.h
@@ -25,6 +25,7 @@
 #ifndef _INCLUDE_GUARD_DEVFREQ_H
 #define _INCLUDE_GUARD_DEVFREQ_H
 
+#include <sys/time.h>
 #include "device.h"
 #include "../parameters/parameters.h"
 
diff --git a/src/perf/perf.h b/src/perf/perf.h
index ee072ae06d24..932588a684f9 100644
--- a/src/perf/perf.h
+++ b/src/perf/perf.h
@@ -26,6 +26,7 @@
 #define _INCLUDE_GUARD_PERF_H_
 
 #include <iostream>
+#include <stdio.h>
 
 
 extern "C" {
