--- a/testing/doallspeeds.sh
+++ b/testing/doallspeeds.sh
@@ -25,11 +25,11 @@ for f in ../*/*.pat; do 
 	printf `basename $f .pat`
 
 	if [ $userspace ]; then 
-		gtime=`./timeit.sh $f userspace   real | grep Total | cut -d\  -f 2`
+		gtime=`./timeit.sh $f userspace   real | grep Total | cut -d\  -f 2 | awk '{print $1}'`
 		printf \\t$gtime
 	fi
 	if [ $kernel ]; then 
-		htime=`./timeit.sh $f kernel real | grep Total | cut -d\  -f 2`
+		htime=`./timeit.sh $f kernel real | grep Total | cut -d\  -f 2 | awk '{print $1}'`
 		printf \\t$htime
 	fi
 	printf \\n
