Index: shorewall-5.1.4.1/configfiles/shorewall.conf
===================================================================
--- shorewall-5.1.4.1.orig/configfiles/shorewall.conf	2017-10-05 11:18:41.586275516 +0200
+++ shorewall-5.1.4.1/configfiles/shorewall.conf	2017-10-05 11:26:11.825609382 +0200
@@ -33,7 +33,7 @@
 #			       L O G G I N G
 ###############################################################################
 
-LOG_LEVEL="info"
+LOG_LEVEL="warn"
 
 BLACKLIST_LOG_LEVEL=
 
@@ -101,7 +101,7 @@
 
 SHOREWALL_SHELL=/bin/sh
 
-SUBSYSLOCK=/var/lock/subsys/shorewall
+SUBSYSLOCK=/var/lock/shorewall
 
 TC=
 
