Index: shorewall-5.1.9/configfiles/shorewall.conf
===================================================================
--- shorewall-5.1.9.orig/configfiles/shorewall.conf	2017-12-18 08:09:12.043625770 +0100
+++ shorewall-5.1.9/configfiles/shorewall.conf	2017-12-18 08:09:54.353402257 +0100
@@ -95,7 +95,7 @@
 
 NFACCT=
 
-PATH="/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin:/usr/local/sbin"
+PATH="/usr/sbin:/usr/bin:/sbin:/bin"
 
 PERL=/usr/bin/perl
 
