diff -u --recursive gcc-8.4.0-vanilla/gcc/real.h gcc-8.4.0/gcc/real.h
--- gcc-8.4.0-vanilla/gcc/real.h	2020-03-04 02:30:01.000000000 -0600
+++ gcc-8.4.0/gcc/real.h	2022-05-18 17:02:22.708820427 -0500
@@ -70,8 +70,10 @@
    + (REAL_VALUE_TYPE_SIZE%HOST_BITS_PER_WIDE_INT ? 1 : 0)) /* round up */
 
 /* Verify the guess.  */
+#ifndef __LP64__
 extern char test_real_width
   [sizeof (REAL_VALUE_TYPE) <= REAL_WIDTH * sizeof (HOST_WIDE_INT) ? 1 : -1];
+#endif
 
 /* Calculate the format for CONST_DOUBLE.  We need as many slots as
    are necessary to overlay a REAL_VALUE_TYPE on them.  This could be
