From 0e31e37c0f6ad0b3f393f024d1f6d8ce448e4746 Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Sat, 8 Dec 2018 18:39:50 -0800
Subject: [PATCH] Platform: Fix compilation without deprecated OpenSSL APIs

---
 src/Platform.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/Platform.cc b/src/Platform.cc
index ea73b6c6..d7882a97 100644
--- a/src/Platform.cc
+++ b/src/Platform.cc
@@ -44,6 +44,7 @@
 #ifdef HAVE_OPENSSL
 #include <openssl/err.h>
 #include <openssl/ssl.h>
+#include "libssl_compat.h"
 #endif // HAVE_OPENSSL
 #ifdef HAVE_LIBGCRYPT
 #include <gcrypt.h>
@@ -111,11 +112,13 @@ bool Platform::setUp()
 #endif // ENABLE_NLS
 
 #ifdef HAVE_OPENSSL
+#if !OPENSSL_101_API
   // for SSL initialization
   SSL_load_error_strings();
   SSL_library_init();
   // Need this to "decrypt" p12 files.
   OpenSSL_add_all_algorithms();
+#endif // !OPENSSL_101_API
 #endif // HAVE_OPENSSL
 #ifdef HAVE_LIBGCRYPT
   if (!gcry_check_version("1.2.4")) {
-- 
2.17.1

