--- a/Makefile.in
+++ b/Makefile.in
@@ -288,12 +288,12 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.6 foreign
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = include src examples debian doc win32
+SUBDIRS = include src debian
 EXTRA_DIST = INSTALL m4/shout.m4 m4/acx_pthread.m4 \
 	m4/ogg.m4 m4/vorbis.m4 m4/xiph_compiler.m4 m4/xiph_net.m4 \
 	m4/xiph_types.m4
 
-doc_DATA = COPYING NEWS README examples/example.c examples/nonblocking.c
+doc_DATA = COPYING NEWS README
 m4datadir = $(datadir)/aclocal
 m4data_DATA = m4/shout.m4
 @HAVE_PKGCONFIG_TRUE@pkgconfigdir = $(libdir)/pkgconfig
