--- a/Makefile.am
+++ b/Makefile.am
@@ -3,14 +3,14 @@
 AUTOMAKE_OPTIONS = 1.6 foreign
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = include src examples doc win32
+SUBDIRS = include src
 
 EXTRA_DIST = INSTALL m4/shout.m4 m4/acx_pthread.m4 \
 	m4/ogg.m4 m4/vorbis.m4 m4/xiph_compiler.m4 m4/xiph_net.m4 \
 	m4/xiph_types.m4 libshout.ckport
 
 docdir = $(datadir)/doc/$(PACKAGE)
-doc_DATA = COPYING NEWS README examples/example.c examples/nonblocking.c
+doc_DATA = COPYING README
 
 m4datadir = $(datadir)/aclocal
 m4data_DATA = m4/shout.m4
