diff --git a/src/tls.c b/src/tls.c
index 4562c73..f946946 100644
--- a/src/tls.c
+++ b/src/tls.c
@@ -63,12 +63,16 @@ static inline int tls_setup(shout_tls_t *tls)
 {
 	SSL_METHOD *meth;
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
 	SSL_library_init();
 	SSL_load_error_strings();
 	SSLeay_add_all_algorithms();
  	SSLeay_add_ssl_algorithms();
 
 	meth = TLSv1_client_method();
+#else
+	meth = TLS_client_method();
+#endif
 	if (!meth)
 		goto error;
 
