Index: gnuplot-5.2.8/Makefile.am
===================================================================
--- gnuplot-5.2.8.orig/Makefile.am
+++ gnuplot-5.2.8/Makefile.am
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in -*-Makefile-*-
 AUTOMAKE_OPTIONS = foreign
 
-SUBDIRS = config m4 term src docs man demo tutorial share
+SUBDIRS = config m4 term src man tutorial share
 
 EXTRA_DIST = BUGS Copyright FAQ.pdf GNUmakefile INSTALL INSTALL.gnu \
 Makefile.maint PATCHLEVEL PGPKEYS README RELEASE_NOTES \
