--- a/lib/xref.h
+++ b/lib/xref.h
@@ -135,7 +135,7 @@ extern struct xref_block *xref_blocks;
 extern void xref_block_add(struct xref_block *block);
 extern void xref_gcc_workaround(const struct xref *xref);
 
-#ifndef HAVE_SECTION_SYMS
+#if !defined(HAVE_SECTION_SYMS) || defined(__mips64)
 /* we have a build system patch to use GNU ld on Solaris;  if that doesn't
  * work we end up on Solaris ld which doesn't support the section start/end
  * symbols.
