diff --git a/Makefile.am b/Makefile.am
index 22647be..7dcf0de 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -32,7 +32,7 @@ AM_DISTCHECK_CONFIGURE_FLAGS = --enable-python-libs
 COMMON_CXXFLAGS_ONLY_WARNINGS = \
     -I$(top_srcdir)/include \
     -I$(top_builddir)/include \
-    -Wall -Wformat -W -fvisibility-inlines-hidden \
+    -Wall -Wformat -W \
     $(libprotobuf_CFLAGS)
 
 COMMON_CXXFLAGS = $(COMMON_CXXFLAGS_ONLY_WARNINGS)
