Index: domoticz/CMakeLists.txt
===================================================================
--- domoticz.orig/CMakeLists.txt	2015-11-21 22:55:17.000000000 +0100
+++ domoticz/CMakeLists.txt	2015-11-21 23:02:58.435719379 +0100
@@ -474,12 +474,12 @@
 ADD_PRECOMPILED_HEADER(domoticz "main/stdafx.h")
 
 IF(CMAKE_COMPILER_IS_GNUCXX)
-  option(USE_STATIC_LIBSTDCXX "Build with static libgcc/libstdc++ libraries" YES)
+  option(USE_STATIC_LIBSTDCXX "Build with static libgcc/libstdc++ libraries" NO)
   IF(USE_STATIC_LIBSTDCXX)
     message(STATUS "Using static libgcc/libstdc++")
-    SET(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_C_FLAGS} -static-libgcc -static-libstdc++")
-    SET(CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS} -static-libgcc -static-libstdc++")
-    SET(CMAKE_CXX_LINK_FLAGS "${CMAKE_CXX_LINK_FLAGS} -static-libgcc -static-libstdc++")
+    #SET(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_C_FLAGS} -static-libgcc -static-libstdc++")
+    #SET(CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS} -static-libgcc -static-libstdc++")
+    #SET(CMAKE_CXX_LINK_FLAGS "${CMAKE_CXX_LINK_FLAGS} -static-libgcc -static-libstdc++")
   ELSE(USE_STATIC_LIBSTDCXX)
     message(STATUS "Using dynamic libgcc_s/libstdc++")
   ENDIF(USE_STATIC_LIBSTDCXX)
