From fff4bef553cfd75030d473b3296ade88b3150909 Mon Sep 17 00:00:00 2001
From: Rob Peters <Info@Domoticz.com>
Date: Thu, 10 Mar 2022 07:09:18 +0100
Subject: [PATCH] Fixed compile error when PYTHON was disabled (Fixes #5187)

---
 hardware/plugins/DelayedLink.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/hardware/plugins/DelayedLink.h
+++ b/hardware/plugins/DelayedLink.h
@@ -1,5 +1,5 @@
 #pragma once
-
+#ifdef ENABLE_PYTHON
 #ifdef WIN32
 #	define MS_NO_COREDLL 1
 #else
@@ -574,3 +574,4 @@ static inline void py3__Py_XDECREF(PyObj
 #endif
 #pragma pop_macro("_DEBUG")
 } // namespace Plugins
+#endif //#ifdef ENABLE_PYTHON
