diff --git a/src/_cffi_src/openssl/cmac.py b/src/_cffi_src/openssl/cmac.py
index f4a3686..bf52144 100644
--- a/src/_cffi_src/openssl/cmac.py
+++ b/src/_cffi_src/openssl/cmac.py
@@ -5,7 +5,7 @@
 from __future__ import absolute_import, division, print_function
 
 INCLUDES = """
-#if OPENSSL_VERSION_NUMBER >= 0x10001000L
+#if !defined(OPENSSL_NO_CMAC) && OPENSSL_VERSION_NUMBER >= 0x10001000L
 #include <openssl/cmac.h>
 #endif
 """
@@ -28,7 +28,7 @@ void CMAC_CTX_free(CMAC_CTX *);
 """
 
 CUSTOMIZATIONS = """
-#if OPENSSL_VERSION_NUMBER < 0x10001000L
+#if defined(OPENSSL_NO_CMAC) || OPENSSL_VERSION_NUMBER < 0x10001000L
 
 static const long Cryptography_HAS_CMAC = 0;
 typedef void CMAC_CTX;
