diff -u --recursive gst-plugins-base-1.15.2-vanilla/configure.ac gst-plugins-base-1.15.2/configure.ac
--- gst-plugins-base-1.15.2-vanilla/configure.ac	2019-02-26 06:39:03.000000000 -0500
+++ gst-plugins-base-1.15.2/configure.ac	2019-03-16 13:30:18.212618930 -0400
@@ -1044,7 +1044,6 @@
 docs/libs/Makefile
 docs/plugins/Makefile
 docs/version.entities
-po/Makefile.in
 common/Makefile
 common/m4/Makefile
 m4/Makefile
diff -u --recursive gst-plugins-base-1.15.2-vanilla/Makefile.am gst-plugins-base-1.15.2/Makefile.am
--- gst-plugins-base-1.15.2-vanilla/Makefile.am	2019-01-26 13:51:01.000000000 -0500
+++ gst-plugins-base-1.15.2/Makefile.am	2019-03-16 13:30:03.006572636 -0400
@@ -13,7 +13,6 @@
 	tools 			\
 	tests 			\
 	docs			\
-	po 			\
 	common 			\
 	m4
 
@@ -24,7 +23,6 @@
 	gst sys ext		\
 	tools 			\
 	tests 			\
-	po 			\
 	common 			\
 	m4
 
