--- a/Makefile.am
+++ b/Makefile.am
@@ -11,7 +11,6 @@ SUBDIRS = 			\
 	gst-libs 		\
 	gst sys $(SUBDIRS_EXT) 	\
 	tools 			\
-	tests 			\
 	po 			\
 	common 			\
 	m4
@@ -21,7 +20,6 @@ DIST_SUBDIRS = 			\
 	gst-libs		\
 	gst sys ext		\
 	tools 			\
-	tests 			\
 	po 			\
 	common 			\
 	m4
--- a/Makefile.in
+++ b/Makefile.in
@@ -538,7 +538,6 @@ SUBDIRS = \
 	gst-libs 		\
 	gst sys $(SUBDIRS_EXT) 	\
 	tools 			\
-	tests 			\
 	po 			\
 	common 			\
 	m4
@@ -548,7 +547,6 @@ DIST_SUBDIRS = \
 	gst-libs		\
 	gst sys ext		\
 	tools 			\
-	tests 			\
 	po 			\
 	common 			\
 	m4
