--- a/configure.ac
+++ b/configure.ac
@@ -918,7 +918,6 @@ docs/design/Makefile
 docs/libs/Makefile
 docs/plugins/Makefile
 docs/version.entities
-po/Makefile.in
 common/Makefile
 common/m4/Makefile
 m4/Makefile
--- a/Makefile.am
+++ b/Makefile.am
@@ -11,7 +11,6 @@ SUBDIRS = 			\
 	gst-libs 		\
 	gst sys $(SUBDIRS_EXT) 	\
 	tools 			\
-	po 			\
 	common 			\
 	m4
 
@@ -20,7 +19,6 @@ DIST_SUBDIRS = 			\
 	gst-libs		\
 	gst sys ext		\
 	tools 			\
-	po 			\
 	common 			\
 	m4
 
--- a/Makefile.in
+++ b/Makefile.in
@@ -538,7 +538,6 @@ SUBDIRS = \
 	gst-libs 		\
 	gst sys $(SUBDIRS_EXT) 	\
 	tools 			\
-	po 			\
 	common 			\
 	m4
 
@@ -547,7 +546,6 @@ DIST_SUBDIRS = \
 	gst-libs		\
 	gst sys ext		\
 	tools 			\
-	po 			\
 	common 			\
 	m4
 
