--- a/Mailman/Defaults.py.in
+++ b/Mailman/Defaults.py.in
@@ -506,7 +506,7 @@ SMTPLIB_DEBUG_LEVEL = 0
 # standard out (or send an email to the site list owner) for manual twiddling
 # of an /etc/aliases style file.  Use 'Postfix' if you are using the Postfix
 # MTA -- but then also see POSTFIX_STYLE_VIRTUAL_DOMAINS.
-MTA = 'Manual'
+MTA = 'Postfix'
 
 # If you set MTA='Postfix', then you also want to set the following variable,
 # depending on whether you're using virtual domains in Postfix, and which
