--- a/Mailman/MailList.py
+++ b/Mailman/MailList.py
@@ -30,7 +30,7 @@ import re
 import shutil
 import socket
 import urllib
-import cPickle
+import pickle as cPickle
 
 from cStringIO import StringIO
 from UserDict import UserDict
--- a/misc/paths.py.in
+++ b/misc/paths.py.in
@@ -71,14 +71,14 @@ sys.path.append(distdir)
 # In a normal interactive Python environment, the japanese.pth and korean.pth
 # files would be imported automatically.  But because we inhibit the importing
 # of the site module, we need to be explicit about importing these codecs.
-if not jaok:
-    import japanese
+#if not jaok:
+#    import japanese
 # As of KoreanCodecs 2.0.5, you had to do the second import to get the Korean
 # codecs installed, however leave the first import in there in case an upgrade
 # changes this.
-if not kook:
-    import korean
-    import korean.aliases
+#if not kook:
+#    import korean
+#    import korean.aliases
 # Arabic and Hebrew (RFC-1556) encoding aliases. (temporary solution)
 import encodings.aliases
 encodings.aliases.aliases.update({
