--- a/Makefile
+++ b/Makefile
@@ -32,9 +32,9 @@ HEADERS	= mini_snmpd.h
 SOURCES	= mini_snmpd.c protocol.c mib.c globals.c utils.c linux.c freebsd.c
 VERSION = 1.2b
 VENDOR	= .1.3.6.1.4.1
-OFLAGS	= -O2 -DDEBUG -g
+OFLAGS	= -O2
 CFLAGS	= -Wall -Werror -DVERSION="\"$(VERSION)\"" -DVENDOR="\"$(VENDOR)\"" \
-	  $(OFLAGS) -D__TRAPS__ -D__LINUX__ -D__IPV6__
+	  $(OFLAGS) -D__TRAPS__ -D__LINUX__
 LDFLAGS	= $(OFLAGS)
 TARGET	= mini_snmpd
 MAN 	= mini_snmpd.8
