diff --git a/conf.d/charts.d.conf b/conf.d/charts.d.conf
index acb2a6f..8111859 100644
--- a/conf.d/charts.d.conf
+++ b/conf.d/charts.d.conf
@@ -30,7 +30,7 @@
 
 # the default enable/disable for all charts.d collectors
 # the default is "yes"
-# enable_all_charts="yes"
+enable_all_charts="no"
 
 # BY DEFAULT ENABLED MODULES
 # ap=yes
diff --git a/conf.d/python.d.conf b/conf.d/python.d.conf
index 940bd91..78e2f0a 100644
--- a/conf.d/python.d.conf
+++ b/conf.d/python.d.conf
@@ -7,7 +7,7 @@
 #
 
 # Enable / disable the whole python.d.plugin (all its modules)
-enabled: yes
+enabled: no
 
 # Prevent log flood
 # Define how many log messages can be written to log file in one log_interval
