--- a/lua-nginx/config
+++ b/lua-nginx/config
@@ -409,35 +409,35 @@ NGX_ADDON_DEPS="$NGX_ADDON_DEPS \
 
 CFLAGS="$CFLAGS -DNDK_SET_VAR"
 
-ngx_feature="export symbols by default (-E)"
-ngx_feature_libs="-Wl,-E"
-ngx_feature_name=
-ngx_feature_run=no
-ngx_feature_incs="#include <stdio.h>"
-ngx_feature_path=
-ngx_feature_test='printf("hello");'
-
-. auto/feature
-
-if [ $ngx_found = yes ]; then
-    CORE_LIBS="-Wl,-E $CORE_LIBS"
-fi
+#ngx_feature="export symbols by default (-E)"
+#ngx_feature_libs="-Wl,-E"
+#ngx_feature_name=
+#ngx_feature_run=no
+#ngx_feature_incs="#include <stdio.h>"
+#ngx_feature_path=
+#ngx_feature_test='printf("hello");'
+#
+#. auto/feature
+#
+#if [ $ngx_found = yes ]; then
+#    CORE_LIBS="-Wl,-E $CORE_LIBS"
+#fi
 
 # for Cygwin
-ngx_feature="export symbols by default (--export-all-symbols)"
-ngx_feature_libs="-Wl,--export-all-symbols"
-ngx_feature_name=
-ngx_feature_run=no
-ngx_feature_incs="#include <stdio.h>"
-ngx_feature_path=
-ngx_feature_test='printf("hello");'
-
-. auto/feature
-
-if [ $ngx_found = yes ]; then
-    CORE_LIBS="-Wl,--export-all-symbols $CORE_LIBS"
-fi
-
+#ngx_feature="export symbols by default (--export-all-symbols)"
+#ngx_feature_libs="-Wl,--export-all-symbols"
+#ngx_feature_name=
+#ngx_feature_run=no
+#ngx_feature_incs="#include <stdio.h>"
+#ngx_feature_path=
+#ngx_feature_test='printf("hello");'
+#
+#. auto/feature
+#
+#if [ $ngx_found = yes ]; then
+#    CORE_LIBS="-Wl,--export-all-symbols $CORE_LIBS"
+#fi
+#
 NGX_DTRACE_PROVIDERS="$NGX_DTRACE_PROVIDERS $ngx_addon_dir/dtrace/ngx_lua_provider.d"
 NGX_TAPSET_SRCS="$NGX_TAPSET_SRCS $ngx_addon_dir/tapset/ngx_lua.stp"
 
