--- a/dio.c
+++ b/dio.c
@@ -442,6 +442,26 @@ PHP_FUNCTION(dio_tcsetattr)
 
 	/* assign to correct values... */
 	switch (Baud_Rate)  {
+#ifdef B460800
+		case 460800:
+			BAUD = B460800;
+			break;
+#endif
+#ifdef B230400
+		case 230400:
+			BAUD = B230400;
+			break;
+#endif
+#ifdef B115200
+		case 115200:
+			BAUD = B115200;
+			break;
+#endif
+#ifdef B57600
+		case 57600:
+			BAUD = B57600;
+			break;
+#endif
 		case 38400:
 			BAUD = B38400;
 			break;
