diff -u --recursive gstreamer-1.12.4-vanilla/configure.ac gstreamer-1.12.4/configure.ac
--- gstreamer-1.12.4-vanilla/configure.ac	2018-02-10 17:37:16.379378242 -0500
+++ gstreamer-1.12.4/configure.ac	2018-02-10 17:40:18.742109326 -0500
@@ -1083,7 +1083,6 @@
 plugins/Makefile
 plugins/elements/Makefile
 plugins/tracers/Makefile
-po/Makefile.in
 tests/Makefile
 tests/benchmarks/Makefile
 tests/check/Makefile
diff -u --recursive gstreamer-1.12.4-vanilla/Makefile.am gstreamer-1.12.4/Makefile.am
--- gstreamer-1.12.4-vanilla/Makefile.am	2018-02-10 17:37:16.390378286 -0500
+++ gstreamer-1.12.4/Makefile.am	2018-02-10 17:40:29.112150898 -0500
@@ -9,7 +9,6 @@
 SUBDIRS = pkgconfig \
 	gst libs plugins tests \
 	docs \
-	po \
 	m4 \
 	common \
 	data
@@ -22,7 +21,6 @@
 DIST_SUBDIRS = pkgconfig \
 	gst libs plugins tools tests \
 	docs \
-	po \
 	m4 \
 	common \
 	data
