diff -u --recursive gstreamer-1.0.7-vanilla/Makefile.am gstreamer-1.0.7/Makefile.am
--- gstreamer-1.0.7-vanilla/Makefile.am	2013-06-01 18:47:44.498057069 -0500
+++ gstreamer-1.0.7/Makefile.am	2013-06-01 19:01:41.634143861 -0500
@@ -8,7 +8,6 @@
 
 SUBDIRS = pkgconfig \
 	gst libs plugins tests \
-	docs \
 	po \
 	m4 \
 	common
@@ -20,7 +19,6 @@
 # These are all the possible subdirs
 DIST_SUBDIRS = pkgconfig \
 	gst libs plugins tools tests \
-	docs \
 	po \
 	m4 \
 	common
diff -u --recursive gstreamer-1.0.7-vanilla/Makefile.in gstreamer-1.0.7/Makefile.in
--- gstreamer-1.0.7-vanilla/Makefile.in	2013-06-01 18:47:44.533056863 -0500
+++ gstreamer-1.0.7/Makefile.in	2013-06-01 19:02:04.529246907 -0500
@@ -508,13 +508,12 @@
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc --enable-docbook
 aclocaldir = $(datadir)/aclocal
 aclocal_DATA = gst-element-check-@GST_API_VERSION@.m4
-SUBDIRS = pkgconfig gst libs plugins tests docs po m4 common \
+SUBDIRS = pkgconfig gst libs plugins tests po m4 common \
 	$(am__append_1)
 
 # These are all the possible subdirs
 DIST_SUBDIRS = pkgconfig \
 	gst libs plugins tools tests \
-	docs \
 	po \
 	m4 \
 	common
