diff -u --recursive gstreamer-1.12.4-vanilla/configure.ac gstreamer-1.12.4/configure.ac
--- gstreamer-1.12.4-vanilla/configure.ac	2018-02-10 17:43:59.957864211 -0500
+++ gstreamer-1.12.4/configure.ac	2018-02-10 17:44:19.772930084 -0500
@@ -1083,20 +1083,6 @@
 plugins/Makefile
 plugins/elements/Makefile
 plugins/tracers/Makefile
-tests/Makefile
-tests/benchmarks/Makefile
-tests/check/Makefile
-tests/misc/Makefile
-tests/examples/Makefile
-tests/examples/adapter/Makefile
-tests/examples/controller/Makefile
-tests/examples/stepping/Makefile
-tests/examples/helloworld/Makefile
-tests/examples/memory/Makefile
-tests/examples/netclock/Makefile
-tests/examples/ptp/Makefile
-tests/examples/streamiddemux/Makefile
-tests/examples/streams/Makefile
 tools/Makefile
 common/Makefile
 common/m4/Makefile
diff -u --recursive gstreamer-1.12.4-vanilla/Makefile.am gstreamer-1.12.4/Makefile.am
--- gstreamer-1.12.4-vanilla/Makefile.am	2018-02-10 17:43:59.959864217 -0500
+++ gstreamer-1.12.4/Makefile.am	2018-02-10 17:44:30.298965075 -0500
@@ -7,7 +7,7 @@
 aclocal_DATA = gst-element-check-@GST_API_VERSION@.m4
 
 SUBDIRS = pkgconfig \
-	gst libs plugins tests \
+	gst libs plugins \
 	docs \
 	m4 \
 	common \
@@ -19,7 +19,7 @@
 
 # These are all the possible subdirs
 DIST_SUBDIRS = pkgconfig \
-	gst libs plugins tools tests \
+	gst libs plugins tools \
 	docs \
 	m4 \
 	common \
