--- a/Makefile.am
+++ b/Makefile.am
@@ -7,7 +7,7 @@ aclocaldir = $(datadir)/aclocal
 aclocal_DATA = gst-element-check-@GST_API_VERSION@.m4
 
 SUBDIRS = pkgconfig \
-	gst libs plugins tests \
+	gst libs plugins \
 	po \
 	m4 \
 	common
@@ -18,7 +18,7 @@ endif
 
 # These are all the possible subdirs
 DIST_SUBDIRS = pkgconfig \
-	gst libs plugins tools tests \
+	gst libs plugins tools \
 	po \
 	m4 \
 	common
--- a/Makefile.in
+++ b/Makefile.in
@@ -554,12 +554,12 @@ top_srcdir = @top_srcdir@
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc --enable-docbook
 aclocaldir = $(datadir)/aclocal
 aclocal_DATA = gst-element-check-@GST_API_VERSION@.m4
-SUBDIRS = pkgconfig gst libs plugins tests po m4 common \
+SUBDIRS = pkgconfig gst libs plugins po m4 common \
 	$(am__append_1)
 
 # These are all the possible subdirs
 DIST_SUBDIRS = pkgconfig \
-	gst libs plugins tools tests \
+	gst libs plugins tools \
 	po \
 	m4 \
 	common
