Index: shorewall-lite-5.1.7.2/shorewall-lite.conf
===================================================================
--- shorewall-lite-5.1.7.2.orig/shorewall-lite.conf	2017-10-11 08:19:35.565847520 +0200
+++ shorewall-lite-5.1.7.2/shorewall-lite.conf	2017-10-11 08:19:35.561847733 +0200
@@ -28,7 +28,7 @@
 #			       L O G G I N G
 ###############################################################################
 
-LOGFILE=
+LOGFILE=/dev/null
 
 ###############################################################################
 #	L O C A T I O N	  O F	F I L E S   A N D   D I R E C T O R I E S
