--- a/plugins/virtium/virtium-nvme.c
+++ b/plugins/virtium/virtium-nvme.c
@@ -127,7 +127,7 @@ static void vt_convert_smart_data_to_hum
 
 	capacity = le64_to_cpu(smart->raw_ns.nsze) * lba;
 
-	snprintf(tempbuff, sizeof(tempbuff), "log;%s;%llu;%s;%s;%-.*s;", smart->raw_ctrl.sn,
+	snprintf(tempbuff, sizeof(tempbuff), "log;%s;%" PRId64 ";%s;%s;%-.*s;", smart->raw_ctrl.sn,
 		 (unsigned long long)smart->time_stamp, smart->path,
 		 smart->raw_ctrl.mn, (int)sizeof(smart->raw_ctrl.fr),
 		 smart->raw_ctrl.fr);
