--- a/memcached.c
+++ b/memcached.c
@@ -13,6 +13,10 @@
  *      Anatoly Vorobey <mellon@pobox.com>
  *      Brad Fitzpatrick <brad@danga.com>
  */
+#ifndef __need_IOV_MAX
+#define __need_IOV_MAX
+#endif
+
 #include "memcached.h"
 #include <sys/stat.h>
 #include <sys/socket.h>
@@ -29,9 +33,6 @@
 #define _P1003_1B_VISIBLE
 #endif
 /* need this to get IOV_MAX on some platforms. */
-#ifndef __need_IOV_MAX
-#define __need_IOV_MAX
-#endif
 #include <pwd.h>
 #include <sys/mman.h>
 #include <fcntl.h>
