diff --git a/setuptools/dist.py b/setuptools/dist.py
index 6b9d350..1350e8a 100644
--- a/setuptools/dist.py
+++ b/setuptools/dist.py
@@ -15,7 +15,6 @@ from distutils.errors import (DistutilsOptionError, DistutilsPlatformError,
 
 from setuptools.depends import Require
 from setuptools.compat import basestring, PY2
-from setuptools import windows_support
 import pkg_resources
 
 def _get_unpatched(cls):
@@ -310,7 +309,6 @@ class Distribution(_Distribution):
         egg_cache_dir = os.path.join(os.curdir, '.eggs')
         if not os.path.exists(egg_cache_dir):
             os.mkdir(egg_cache_dir)
-            windows_support.hide_file(egg_cache_dir)
             readme_txt_filename = os.path.join(egg_cache_dir, 'README.txt')
             with open(readme_txt_filename, 'w') as f:
                 f.write('This directory contains eggs that were downloaded '
