From 333bbabc6dfad0553fb63d560ab6442a50cc9e52 Mon Sep 17 00:00:00 2001
From: Moritz Warning <moritzwarning@web.de>
Date: Fri, 22 Jul 2016 23:27:31 +0200
Subject: [PATCH 2/2] fix build

---
 make-linux.mk | 30 +++++++++++++++---------------
 1 file changed, 15 insertions(+), 15 deletions(-)

--- a/make-linux.mk
+++ b/make-linux.mk
@@ -39,24 +39,24 @@ include objects.mk
 
 # On Linux we auto-detect the presence of some libraries and if present we
 # link against the system version. This works with our package build images.
-ifeq ($(wildcard /usr/include/lz4.h),)
+#ifeq ($(wildcard $(STAGING_DIR)/usr/include/lz4.h),)
 	OBJS+=ext/lz4/lz4.o
-else
-	LDLIBS+=-llz4
-	DEFS+=-DZT_USE_SYSTEM_LZ4
-endif
-ifeq ($(wildcard /usr/include/http_parser.h),)
+#else
+#	LDLIBS+=-llz4
+#	DEFS+=-DZT_USE_SYSTEM_LZ4
+#endif
+#ifeq ($(wildcard $(STAGING_DIR)/usr/include/http_parser.h),)
 	OBJS+=ext/http-parser/http_parser.o
-else
-	LDLIBS+=-lhttp_parser
-	DEFS+=-DZT_USE_SYSTEM_HTTP_PARSER
-endif
-ifeq ($(wildcard /usr/include/json-parser/json.h),)
+#else
+#	LDLIBS+=-lhttp_parser
+#	DEFS+=-DZT_USE_SYSTEM_HTTP_PARSER
+#endif
+#ifeq ($(wildcard $(STAGING_DIR)/usr/include/json-parser/json.h),)
 	OBJS+=ext/json-parser/json.o
-else
-	LDLIBS+=-ljsonparser
-	DEFS+=-DZT_USE_SYSTEM_JSON_PARSER
-endif
+#else
+#	LDLIBS+=-ljsonparser
+#	DEFS+=-DZT_USE_SYSTEM_JSON_PARSER
+#endif
 
 ifeq ($(ZT_USE_MINIUPNPC),1)
 	OBJS+=osdep/PortMapper.o
