From bfb1a652dbf897dc065d2a1414296eb145a2224b Mon Sep 17 00:00:00 2001
From: Moritz Warning <moritzwarning@web.de>
Date: Mon, 23 Apr 2018 22:31:03 +0200
Subject: [PATCH 3/4] remove -march=armv5

---
 make-linux.mk | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/make-linux.mk b/make-linux.mk
index add1d3ae..49e14f70 100644
--- a/make-linux.mk
+++ b/make-linux.mk
@@ -229,12 +229,12 @@ endif
 # ARM32 hell -- use conservative CFLAGS
 ifeq ($(ZT_ARCHITECTURE),3)
 	ifeq ($(shell if [ -e /usr/bin/dpkg ]; then dpkg --print-architecture; fi),armel)
-		override CFLAGS+=-march=armv5 -mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
-		override CXXFLAGS+=-march=armv5 -mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
+		override CFLAGS+=-mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
+		override CXXFLAGS+=-mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
 		ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 	else
-		override CFLAGS+=-march=armv5 -mno-unaligned-access -marm
-		override CXXFLAGS+=-march=armv5 -mno-unaligned-access -marm
+		override CFLAGS+=-mno-unaligned-access -marm
+		override CXXFLAGS+=-mno-unaligned-access -marm
 	endif
 endif
 
-- 
2.17.0

