From f8b4c4a045a9711c316a5c48b238c24cc0948da1 Mon Sep 17 00:00:00 2001
From: Moritz Warning <moritzwarning@web.de>
Date: Mon, 6 May 2024 22:35:41 +0200
Subject: [PATCH 4/5] add missing libatomic

Signed-off-by: Moritz Warning <moritzwarning@web.de>
---
 make-linux.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/make-linux.mk
+++ b/make-linux.mk
@@ -11,7 +11,7 @@ endif
 
 INCLUDES?=-Irustybits/target -isystem ext -Iext/prometheus-cpp-lite-1.0/core/include -Iext-prometheus-cpp-lite-1.0/3rdparty/http-client-lite/include -Iext/prometheus-cpp-lite-1.0/simpleapi/include
 DEFS?=
-LDLIBS?=
+LDLIBS?=-latomic
 DESTDIR?=
 EXTRA_DEPS?=
 
