From 8e89af98ac00b1c9c019865faca7479fa0de6084 Mon Sep 17 00:00:00 2001
From: Moritz Warning <moritzwarning@web.de>
Date: Tue, 17 Sep 2024 21:26:08 +0200
Subject: [PATCH 5/5] remove noexecstack

---
 make-linux.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/make-linux.mk
+++ b/make-linux.mk
@@ -364,7 +364,7 @@ endif
 #override CXXFLAGS+=-fPIC -fPIE
 
 # Non-executable stack
-override LDFLAGS+=-Wl,-z,noexecstack
+#override LDFLAGS+=-Wl,-z,noexecstack
 
 .PHONY: all
 all:	one
