From 74e9711e4dbab4f8323a7081915d7d95aa04967b Mon Sep 17 00:00:00 2001
From: Moritz Warning <moritzwarning@web.de>
Date: Fri, 12 Sep 2025 11:17:47 +0200
Subject: [PATCH 5/6] remove noexecstack

---
 make-linux.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/make-linux.mk
+++ b/make-linux.mk
@@ -394,7 +394,7 @@ endif
 #override CXXFLAGS+=-fPIC -fPIE
 
 # Non-executable stack
-override LDFLAGS+=-Wl,-z,noexecstack
+#override LDFLAGS+=-Wl,-z,noexecstack
 
 .PHONY: all
 all:	one
