--- a/setuptools/command/easy_install.py
+++ b/setuptools/command/easy_install.py
@@ -1324,7 +1324,10 @@ class easy_install(Command):
             return  # already did it, or don't need to
 
         sitepy = os.path.join(self.install_dir, "site.py")
-        source = resource_string("setuptools", "site-patch.py")
+        try:
+            source = resource_string("setuptools", "site-patch.py")
+        except FileNotFoundError:
+            source = resource_string("setuptools", "site-patch.py.txt")
         source = source.decode('utf-8')
         current = ""
 
