--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -122,9 +122,8 @@ endif(WIN32)
 
 # Install pkg-config file
 
-set(prefix ${CMAKE_INSTALL_PREFIX})
 set(VERSION ${AVRO_VERSION})
-configure_file(avro-c.pc.in avro-c.pc)
+configure_file(avro-c.pc.in avro-c.pc @ONLY)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/avro-c.pc
         DESTINATION lib/pkgconfig)
 
--- a/src/avro-c.pc.in
+++ b/src/avro-c.pc.in
@@ -1,7 +1,12 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
 Name: avro-c
 Description: C library for parsing Avro data
 Version: @VERSION@
 URL: https://avro.apache.org/
-Libs: -L@prefix@/lib -lavro
-Cflags: -I@prefix@/include
+Libs: -L${libdir} -lavro
+Cflags: -I${includedir}
 Requires: @CODEC_PKG@
