diff --git a/cpp/build/Makefile b/cpp/build/Makefile
index 666472d..41dc58c 100644
--- a/cpp/build/Makefile
+++ b/cpp/build/Makefile
@@ -139,8 +139,8 @@ $(top_builddir)/libopenzwave.pc: $(top_srcdir)/cpp/build/libopenzwave.pc.in $(PK
 	@$(SED) \
 		-e 's|[@]prefix@|$(PREFIX)|g' \
 		-e 's|[@]exec_prefix@|$(PREFIX)/bin|g' \
-		-e 's|[@]libdir@|$(instlibdir)|g' \
-		-e 's|[@]includedir@|$(PREFIX)/include/openzwave/|g' \
+		-e 's|[@]libdir@|$${prefix}/lib|g' \
+		-e 's|[@]includedir@|$${prefix}/include/openzwave/|g' \
                 -e 's|[@]sysconfdir@|/etc/openzwave/|g' \
                 -e 's|[@]gitversion@|$(GITVERSION)|g' \
                 -e 's|[@]docdir@|$(docdir)/|g' \
