Index: trunk/src/cmd/Makefile.am
===================================================================
--- trunk/src/cmd/Makefile.am	(revision 98)
+++ trunk/src/cmd/Makefile.am	(revision 99)
@@ -17,5 +17,5 @@
 INCLUDES = -I$(top_srcdir)/lib
 LDADD = ../lib/libusbip.la @PACKAGE_LIBS@
 EXTRA_CFLAGS = @EXTRA_CFLAGS@
-AM_CFLAGS = -Wall -W -Wstrict-prototypes -std=gnu99 $(EXTRA_CFLAGS) @PACKAGE_CFLAGS@
+AM_CFLAGS = -Wall -W -Wstrict-prototypes -std=gnu99 $(EXTRA_CFLAGS) @PACKAGE_CFLAGS@ -DUSBIDS_FILE='"@USBIDS_DIR@/usb.ids"'
 
