diff -rupN Imaging-1.1.7.orig/_imagingft.c Imaging-1.1.7/_imagingft.c
--- Imaging-1.1.7.orig/_imagingft.c	2009-11-01 01:44:12.000000000 +0100
+++ Imaging-1.1.7/_imagingft.c	2014-12-11 01:05:21.290135484 +0100
@@ -32,7 +32,7 @@
 #include FT_FREETYPE_H
 #else
 /* freetype 2.0 */
-#include <freetype/freetype.h>
+#include <freetype2/freetype.h>
 #endif
 
 #if PY_VERSION_HEX < 0x01060000
@@ -70,7 +70,7 @@ struct {
     const char* message;
 } ft_errors[] =
 
-#include <freetype/fterrors.h>
+#include <freetype2/fterrors.h>
 
 /* -------------------------------------------------------------------- */
 /* font objects */
