From 7e1cc2d933976c20bec932613172b56c5172b7f0 Mon Sep 17 00:00:00 2001
From: lifei76 <lifei76@meituan.com>
Date: Thu, 20 Jul 2023 18:24:37 +0800
Subject: [PATCH 09/11] 7

---
 src/configuration.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/src/configuration.c
+++ b/src/configuration.c
@@ -51,7 +51,10 @@
 # error Problem configuring default log message destination.
 #endif
 /// Name of the default logging function determined at compile-time.
-#define MPTCPD_SET_LOG_FUNCTION MPTCPD_CONCAT(l_log_set_, MPTCPD_LOGGER)
+ // #define MPTCPD_SET_LOG_FUNCTION MPTCPD_CONCAT(l_log_set_, MPTCPD_LOGGER)
+
+#define MPTCPD_SET_LOG_FUNCTION l_log_set_stderr
+
 
 /**
  * @brief Get the function that sets the log message destination.
