--- a/Makefile.am
+++ b/Makefile.am
@@ -17,7 +17,7 @@
 
 ALL_RECURSIVE_TARGETS =
 
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src po
 
 changelog_etc =				\
   ChangeLog-2005			\
--- a/Makefile.in
+++ b/Makefile.in
@@ -1639,7 +1639,7 @@ top_srcdir = @top_srcdir@
 
 # Some tests always need root privileges, others need them only sometimes.
 ALL_RECURSIVE_TARGETS = install-root check-root distcheck-hook
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src po
 changelog_etc = \
   ChangeLog-2005			\
   ChangeLog-2006			\
