--- a/data/meson.build
+++ b/data/meson.build
@@ -15,15 +15,3 @@ install_data(
   [ 'its/shared-mime-info.loc', 'its/shared-mime-info.its', ],
   install_dir : get_option('datadir') / 'gettext/its'
 )
-
-custom_target('shared-mime-info-spec-html',
-    input : 'shared-mime-info-spec.xml',
-    output: 'shared-mime-info-spec-html',
-    command: [
-        xmlto,
-        '-o', '@OUTPUT@',
-        'html-nochunks',
-        '@INPUT@',
-    ],
-    build_by_default: true,
-)
--- a/meson.build
+++ b/meson.build
@@ -18,12 +18,6 @@ config.set_quoted('PACKAGE', meson.proje
 config.set_quoted('VERSION', meson.project_version())
 
 ###############################################################################
-# Find tools
-
-xmllint = find_program('xmllint')
-xmlto   = find_program('xmlto')
-
-###############################################################################
 # Find xdgmime
 
 xdgmime = get_option('xdgmime-path') / 'src'
