diff -rupN seahub-3.1.7-server.orig/seahub/settings.py seahub-3.1.7-server/seahub/settings.py
--- seahub-3.1.7-server.orig/seahub/settings.py	2014-10-20 09:32:35.000000000 +0200
+++ seahub-3.1.7-server/seahub/settings.py	2014-12-10 15:47:21.625104606 +0100
@@ -46,7 +46,7 @@ SITE_ID = 1
 
 # If you set this to False, Django will make some optimizations so as not
 # to load the internationalization machinery.
-USE_I18N = True
+USE_I18N = False
 
 # If you set this to False, Django will not format dates, numbers and
 # calendars according to the current locale.
@@ -209,7 +209,7 @@ SHOW_REPO_DOWNLOAD_BUTTON = False
 REPO_PASSWORD_MIN_LENGTH = 8
 
 # mininum length for user's password
-USER_PASSWORD_MIN_LENGTH = 6
+USER_PASSWORD_MIN_LENGTH = 8
 
 # LEVEL based on four types of input:
 # num, upper letter, lower letter, other symbols
@@ -218,7 +218,7 @@ USER_PASSWORD_STRENGTH_LEVEL = 3
 
 # default False, only check USER_PASSWORD_MIN_LENGTH
 # when True, check password strength level, STRONG(or above) is allowed
-USER_STRONG_PASSWORD_REQUIRED = False
+USER_STRONG_PASSWORD_REQUIRED = True
 
 # Using server side crypto by default, otherwise, let user choose crypto method.
 FORCE_SERVER_CRYPTO = True
