--- a/seahub/settings.py
+++ b/seahub/settings.py
@@ -46,7 +46,7 @@ SITE_ID = 1
 
 # If you set this to False, Django will make some optimizations so as not
 # to load the internationalization machinery.
-USE_I18N = True
+USE_I18N = False
 
 # If you set this to False, Django will not format dates, numbers and
 # calendars according to the current locale.
@@ -339,7 +339,7 @@ SHARE_LINK_EMAIL_LANGUAGE = ''
 ENABLE_UPLOAD_LINK_VIRUS_CHECK = False
 
 # mininum length for user's password
-USER_PASSWORD_MIN_LENGTH = 6
+USER_PASSWORD_MIN_LENGTH = 8
 
 # LEVEL based on four types of input:
 # num, upper letter, lower letter, other symbols
@@ -348,7 +348,7 @@ USER_PASSWORD_STRENGTH_LEVEL = 3
 
 # default False, only check USER_PASSWORD_MIN_LENGTH
 # when True, check password strength level, STRONG(or above) is allowed
-USER_STRONG_PASSWORD_REQUIRED = False
+USER_STRONG_PASSWORD_REQUIRED = True
 
 # Force user to change password when admin add/reset a user.
 FORCE_PASSWORD_CHANGE = True
