--- a/src/socket.c
+++ b/src/socket.c
@@ -43,6 +43,7 @@
 #else
 #  include <netinet/in.h>
 #  include <arpa/inet.h>
+#  include <sys/select.h>
 #  include <sys/socket.h>
 #  include <netdb.h>
 #  include <sys/un.h>
--- a/src/sync.c
+++ b/src/sync.c
@@ -33,6 +33,9 @@
 #include <assert.h>
 #include <stdlib.h>
 #include <stdio.h>
+#ifndef WIN32
+#include <sys/select.h>
+#endif
 #include <fcntl.h>
 #include <unistd.h>
 
