From cf4dfbe15ef8ead3a7eda974af7d804d447f00db Mon Sep 17 00:00:00 2001
From: Glenn Strauss <gstrauss@gluelogic.com>
Date: Mon, 4 Oct 2021 09:51:22 -0400
Subject: [PATCH] [core] define __BEGIN_DECLS, __END_DECLS if needed

---
 src/first.h | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

--- a/src/first.h
+++ b/src/first.h
@@ -39,6 +39,22 @@
 #include <sys/types.h>
 #include <stddef.h>
 
+#ifndef __BEGIN_DECLS
+#ifdef __cplusplus
+#define __BEGIN_DECLS extern "C" {
+#else
+#define __BEGIN_DECLS
+#endif
+#endif
+
+#ifndef __END_DECLS
+#ifdef __cplusplus
+#define __END_DECLS }
+#else
+#define __END_DECLS
+#endif
+#endif
+
 #if defined HAVE_STDINT_H
 # include <stdint.h>
 #elif defined HAVE_INTTYPES_H
