diff -u --recursive grilo-plugins-0.3.5-vanilla/configure grilo-plugins-0.3.5/configure
--- grilo-plugins-0.3.5-vanilla/configure	2018-01-07 21:45:18.874540074 -0500
+++ grilo-plugins-0.3.5/configure	2018-01-07 21:46:31.145749381 -0500
@@ -638,8 +638,6 @@
 LTLIBOBJS
 LIBOBJS
 YELP_HELP_RULES
-XMLLINT
-ITSTOOL
 HELP_DIR
 YELP_LC_DIST
 YELP_LC_MEDIA_LINKS
@@ -1084,9 +1082,7 @@
 GOM_CFLAGS
 GOM_LIBS
 TRACKER_SPARQL_CFLAGS
-TRACKER_SPARQL_LIBS
-ITSTOOL
-XMLLINT'
+TRACKER_SPARQL_LIBS'
 
 
 # Initialize some variables set by options.
@@ -1855,8 +1851,6 @@
               C compiler flags for TRACKER_SPARQL, overriding pkg-config
   TRACKER_SPARQL_LIBS
               linker flags for TRACKER_SPARQL, overriding pkg-config
-  ITSTOOL     Path to the `itstool` command
-  XMLLINT     Path to the `xmllint` command
 
 Use these variables to override the choices made by `configure' or to help
 it to find libraries and programs with nonstandard names/locations.
@@ -16976,89 +16970,6 @@
 
 
 
-# Extract the first word of "itstool", so it can be a program name with args.
-set dummy itstool; ac_word=$2
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
-$as_echo_n "checking for $ac_word... " >&6; }
-if ${ac_cv_prog_ITSTOOL+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  if test -n "$ITSTOOL"; then
-  ac_cv_prog_ITSTOOL="$ITSTOOL" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-    for ac_exec_ext in '' $ac_executable_extensions; do
-  if as_fn_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_prog_ITSTOOL="itstool"
-    $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-  done
-IFS=$as_save_IFS
-
-fi
-fi
-ITSTOOL=$ac_cv_prog_ITSTOOL
-if test -n "$ITSTOOL"; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ITSTOOL" >&5
-$as_echo "$ITSTOOL" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-
-
-if test x"$ITSTOOL" = x; then
-  as_fn_error $? "itstool not found" "$LINENO" 5
-fi
-
-
-# Extract the first word of "xmllint", so it can be a program name with args.
-set dummy xmllint; ac_word=$2
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
-$as_echo_n "checking for $ac_word... " >&6; }
-if ${ac_cv_prog_XMLLINT+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  if test -n "$XMLLINT"; then
-  ac_cv_prog_XMLLINT="$XMLLINT" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-    for ac_exec_ext in '' $ac_executable_extensions; do
-  if as_fn_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_prog_XMLLINT="xmllint"
-    $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-  done
-IFS=$as_save_IFS
-
-fi
-fi
-XMLLINT=$ac_cv_prog_XMLLINT
-if test -n "$XMLLINT"; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $XMLLINT" >&5
-$as_echo "$XMLLINT" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-
-
-if test x"$XMLLINT" = x; then
-  as_fn_error $? "xmllint not found" "$LINENO" 5
-fi
-
 YELP_HELP_RULES='
 HELP_ID ?=
 HELP_POT ?=
