diff --git a/src/dmap/grl-daap-db.c b/src/dmap/grl-daap-db.c
index c931d91..f460113 100644
--- a/src/dmap/grl-daap-db.c
+++ b/src/dmap/grl-daap-db.c
@@ -232,6 +232,11 @@ grl_daap_db_add (DMAPDb *_db, DMAPRecord *_record)
 
   g_free (id_s);
   g_object_unref (media);
+  g_free(album);
+  g_free(artist);
+  g_free(genre);
+  g_free(title);
+  g_free(url);
 
   return --nextid;
 }
diff --git a/src/dmap/grl-dpap-db.c b/src/dmap/grl-dpap-db.c
index 02ffd3a..8be278b 100644
--- a/src/dmap/grl-dpap-db.c
+++ b/src/dmap/grl-dpap-db.c
@@ -177,6 +177,12 @@ grl_dpap_db_add (DMAPDb *_db, DMAPRecord *_record)
 
   g_free (id_s);
   g_object_unref (media);
+  g_free(filename);
+  g_free(aspectratio);
+  g_free(format);
+  g_free(comments);
+  g_free(url);
+  g_byte_array_unref(thumbnail);
 
   return --nextid;
 }
