--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,12 +29,6 @@ project(SUMO)
 set(PACKAGE_VERSION "1.7.0")
 cmake_minimum_required(VERSION 3.1)
 
-find_program(CCACHE_FOUND "ccache")
-if (CCACHE_FOUND AND CCACHE_SUPPORT)
-  message(STATUS "Enabling ccache")
-  set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "ccache")
-endif()
-
 set(CMAKE_COLOR_MAKEFILE ON)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/build/cmake_modules/")
 
