--- a/configure.ac
+++ b/configure.ac
@@ -117,11 +117,6 @@ dnl - - - - - - - - - - - - - - - - - -
 case "$host" in
    x86-*-linux* | ia64-*-linux* | i586-*-linux* | i686-*-linux* | x86_64-*-linux*)
 dnl Make sure we are on architecture that supports SIMD
-   if test x$CXX = xclang++; then
-       CXXFLAGS="-msse2 $CXXFLAGS"
-   else
-       CXXFLAGS="-msse2 -mfpmath=sse $CXXFLAGS"
-   fi
    ;;
    *-cygwin*)
    AC_DEFINE([HAVE_CYGWIN], [1], [Define if compiling under cygwin])
