diff --git a/Makefile b/Makefile
index 5ce6746..c25c0c5 100644
--- a/Makefile
+++ b/Makefile
@@ -66,12 +66,10 @@ BUILDFOR	:= $(shell file /usr/bin/make | sed -e 's!.*ELF \(32\|64\)-bit.*!\1!')-
 
 ifeq ($(origin CFLAGS),undefined)
 ifeq ($(BUILDFOR),32-bit)
-CFLAGS		+= -m32
 LIBDIR		:= /lib
 USRLIBDIR	:= /usr/lib
 else
 ifeq ($(BUILDFOR),64-bit)
-CFLAGS		+= -m64
 LIBDIR		:= /lib64
 USRLIBDIR	:= /usr/lib64
 endif
