From 6ada939adf5c4f7e43d0fbe6ea45492b9564cb8d Mon Sep 17 00:00:00 2001
From: badaix <johannes.pohl@badaix.de>
Date: Thu, 19 Dec 2024 22:53:49 +0100
Subject: [PATCH] Fix compilation with boost v1.87.0

---
 server/control_server.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/server/control_server.cpp
+++ b/server/control_server.cpp
@@ -161,7 +161,7 @@ void ControlServer::start()
             {
                 LOG(INFO, LOG_TAG) << "Creating TCP acceptor for address: " << address << ", port: " << tcp_settings_.port << "\n";
                 acceptor_tcp_.emplace_back(make_unique<tcp::acceptor>(boost::asio::make_strand(io_context_.get_executor()),
-                                                                      tcp::endpoint(boost::asio::ip::address::from_string(address), tcp_settings_.port)));
+                                                                      tcp::endpoint(boost::asio::ip::make_address(address), settings_.tcp.port)));
             }
             catch (const boost::system::system_error& e)
             {
@@ -177,7 +177,7 @@ void ControlServer::start()
             {
                 LOG(INFO, LOG_TAG) << "Creating HTTP acceptor for address: " << address << ", port: " << http_settings_.port << "\n";
                 acceptor_http_.emplace_back(make_unique<tcp::acceptor>(boost::asio::make_strand(io_context_.get_executor()),
-                                                                       tcp::endpoint(boost::asio::ip::address::from_string(address), http_settings_.port)));
+                                                                       tcp::endpoint(boost::asio::ip::make_address(address), settings_.http.port)));
             }
             catch (const boost::system::system_error& e)
             {
