From d2e1635ebd96ac76cc779cb3ab73bae10fd88fd1 Mon Sep 17 00:00:00 2001
From: badaix <johannes.pohl@badaix.de>
Date: Thu, 14 Aug 2025 20:25:01 +0200
Subject: [PATCH] Fix building with boost v1.89.0

---
 client/player/alsa_player.cpp | 1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

--- a/client/player/alsa_player.cpp
+++ b/client/player/alsa_player.cpp
@@ -30,6 +30,7 @@
 
 // standard headers
 #include <chrono>
+#include <cmath>
 
 
 using namespace std::chrono_literals;
