--- a/Makefile.linux
+++ b/Makefile.linux
@@ -28,16 +28,16 @@ endif
 
 # Make sucks, prove me wrong
 , := ,
-ifeq ($(filter -D_FORTIFY_SOURCE=% -Wp$(,)-D_FORTIFY_SOURCE=%,$(CPPFLAGS) $(CFLAGS)),)
-CPPFLAGS += -D_FORTIFY_SOURCE=2
-endif
+#ifeq ($(filter -D_FORTIFY_SOURCE=% -Wp$(,)-D_FORTIFY_SOURCE=%,$(CPPFLAGS) $(CFLAGS)),)
+#CPPFLAGS += -D_FORTIFY_SOURCE=2
+#endif
 CPPFLAGS += -D_GNU_SOURCE
 
 #CFLAGS = -O -g -DDEBUG
 CFLAGS ?= -Os
-CFLAGS += -fno-strict-aliasing
-CFLAGS += -fno-common
-CFLAGS += -fstack-protector -fPIE
+#CFLAGS += -fno-strict-aliasing
+#CFLAGS += -fno-common
+#CFLAGS += -fstack-protector -fPIE
 CFLAGS += -Wall
 CFLAGS += -Wextra -Wstrict-prototypes -Wdeclaration-after-statement
 #CFLAGS += -Wno-missing-field-initializers
