From f8e83c4c7866ae830abe9bec15cfe805dd825b6e Mon Sep 17 00:00:00 2001
From: xentec <xentec@aix0.eu>
Date: Tue, 12 Sep 2017 07:18:38 +0200
Subject: [PATCH] Disable dynamic lib tests for musl's dlclose() is noop

---
 unittests/Support/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/unittests/Support/CMakeLists.txt b/unittests/Support/CMakeLists.txt
index 641163e..4f508d9 100644
--- a/unittests/Support/CMakeLists.txt
+++ b/unittests/Support/CMakeLists.txt
@@ -75,4 +75,4 @@ set_source_files_properties(AlignOfTest.cpp PROPERTIES COMPILE_FLAGS -w)
 # ManagedStatic.cpp uses <pthread>.
 target_link_libraries(SupportTests LLVMTestingSupport ${LLVM_PTHREAD_LIB})
 
-add_subdirectory(DynamicLibrary)
+# add_subdirectory(DynamicLibrary)
-- 
2.14.1

