--- a/qtbase/mkspecs/linux-openwrt-g++/qmake.conf
+++ b/qtbase/mkspecs/linux-openwrt-g++/qmake.conf
@@ -0,0 +1,32 @@
+
+#
+# qmake configuration for building with linux-openwrt-g++
+#
+
+MAKEFILE_GENERATOR      = UNIX
+CONFIG                 += incremental
+QMAKE_INCREMENTAL_STYLE = sublib
+
+include(../common/linux.conf)
+include(../common/gcc-base-unix.conf)
+include(../common/g++-unix.conf)
+
+# modifications to g++.conf
+QMAKE_CC                = $(TARGET_CROSS)gcc
+QMAKE_CXX               = $(TARGET_CROSS)g++
+
+QMAKE_CFLAGS           += $(TARGET_CFLAGS)
+QMAKE_CXXFLAGS         += $(TARGET_CXXFLAGS)
+QMAKE_LFLAGS           += $(TARGET_LDFLAGS)
+
+QMAKE_LINK_C            = $(TARGET_CROSS)gcc
+QMAKE_LINK_C_SHLIB      = $(TARGET_CROSS)gcc
+QMAKE_LINK              = $(TARGET_CROSS)g++
+QMAKE_LINK_SHLIB        = $(TARGET_CROSS)g++
+
+# modifications to linux.conf
+QMAKE_AR                = $(TARGET_CROSS)ar cqs
+QMAKE_OBJCOPY           = $(TARGET_CROSS)objcopy
+QMAKE_NM                = $(TARGET_CROSS)nm -P
+QMAKE_STRIP             = $(TARGET_CROSS)strip
+load(qt_config)
--- a/qtbase/mkspecs/linux-openwrt-g++/qplatformdefs.h
+++ b/qtbase/mkspecs/linux-openwrt-g++/qplatformdefs.h
@@ -0,0 +1,9 @@
+#include "../linux-g++/qplatformdefs.h"
+
+#undef QT_SOCKLEN_T
+
+#if defined(__GLIBC__) && (__GLIBC__ < 2)
+#define QT_SOCKLEN_T            int
+#else
+#define QT_SOCKLEN_T            socklen_t
+#endif
