--- a/t/op/tie_fetch_count.t
+++ b/t/op/tie_fetch_count.t
@@ -250,12 +250,17 @@ for ([chdir=>''],[chmod=>'0,'],[chown=>'
     check_count "$op $args\\\$tied_glob$postargs";
 }
 
-$dummy  =   crypt $var,0; check_count 'crypt $tied, ...';
-$dummy  =   crypt 0,$var; check_count 'crypt ..., $tied';
-$var = substr(chr 256,0,0);
-$dummy  =   crypt $var,0; check_count 'crypt $tied_utf8, ...';
-$var = substr(chr 256,0,0);
-$dummy  =   crypt 0,$var; check_count 'crypt ..., $tied_utf8';
+SKIP: {
+    use Config;
+    skip "crypt() is unavailable", 4, unless defined $Config{'d_crypt'};
+
+    $dummy  =   crypt $var,0; check_count 'crypt $tied, ...';
+    $dummy  =   crypt 0,$var; check_count 'crypt ..., $tied';
+    $var = substr(chr 256,0,0);
+    $dummy  =   crypt $var,0; check_count 'crypt $tied_utf8, ...';
+    $var = substr(chr 256,0,0);
+    $dummy  =   crypt 0,$var; check_count 'crypt ..., $tied_utf8';
+}
 
 SKIP:
 {
