diff --git a/setup.py b/setup.py
index 7868b7b..9ae0ef2 100644
--- a/setup.py
+++ b/setup.py
@@ -444,7 +444,6 @@ class PyBuildExt(build_ext):
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
         if cross_compiling:
             self.add_gcc_paths()
-        self.add_multiarch_paths()
 
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
