diff --git a/setup.py b/setup.py
index 2779658..b6d3d61 100644
--- a/setup.py
+++ b/setup.py
@@ -693,8 +693,8 @@ class PyBuildExt(build_ext):
         # use the same library for the readline and curses modules.
         if 'curses' in readline_termcap_library:
             curses_library = readline_termcap_library
-        elif self.compiler.find_library_file(lib_dirs, 'ncursesw'):
-            curses_library = 'ncursesw'
+        #elif self.compiler.find_library_file(lib_dirs, 'ncursesw'):
+        #    curses_library = 'ncursesw'
         elif self.compiler.find_library_file(lib_dirs, 'ncurses'):
             curses_library = 'ncurses'
         elif self.compiler.find_library_file(lib_dirs, 'curses'):
