diff --git a/sysdep/config.h b/sysdep/config.h
index 03399bd..bc5eb77 100644
--- a/sysdep/config.h
+++ b/sysdep/config.h
@@ -46,11 +46,11 @@ typedef u16 word;
 #  endif
 #else
 #  ifdef DEBUGGING
-#    define PATH_CONFIG "bird.conf"
-#    define PATH_CONTROL_SOCKET "bird.ctl"
+#    define PATH_CONFIG "bird4.conf"
+#    define PATH_CONTROL_SOCKET "bird4.ctl"
 #  else
-#    define PATH_CONFIG PATH_CONFIG_DIR "/bird.conf"
-#    define PATH_CONTROL_SOCKET PATH_CONTROL_SOCKET_DIR "/bird.ctl"
+#    define PATH_CONFIG PATH_CONFIG_DIR "/bird4.conf"
+#    define PATH_CONTROL_SOCKET PATH_CONTROL_SOCKET_DIR "/bird4.ctl"
 #  endif
 #endif
 
