--- a/lib/ipv6.c
+++ b/lib/ipv6.c
@@ -73,6 +73,8 @@ ipv6_classify(ip_addr *a)
     return IADDR_HOST | SCOPE_LINK;
   if ((x & 0xffc00000) == 0xfec00000)		/* Site-Local Address */
     return IADDR_HOST | SCOPE_SITE;
+  if ((x & 0xfe000000) == 0xfc000000)		/* Unique Local Unicast Address (RFC 4193) */
+    return IADDR_HOST | SCOPE_SITE;
   if ((x & 0xff000000) == 0xff000000)		/* Multicast Address */
     {
       unsigned int scope = (x >> 16) & 0x0f;
