--- a/src/regparse.c	2020-04-26 09:20:56.000000000 +0200
+++ b/src/regparse.c	2020-06-14 21:22:18.396966276 +0200
@@ -1294,7 +1294,9 @@
 i_free_callout_name_entry(st_callout_name_key* key, CalloutNameEntry* e,
                           void* arg ARG_UNUSED)
 {
-  xfree(e->name);
+  if (IS_NOT_NULL(e)) {
+    xfree(e->name);
+  }
   /*xfree(key->s); */ /* is same as e->name */
   xfree(key);
   xfree(e);
