--- a/src/libserver/usblowlevel.cpp     2020-06-03 16:41:02.128182612 +0200
+++ b/src/libserver/usblowlevel.cpp     2020-06-03 16:41:31.660704440 +0200
@@ -155,6 +155,7 @@
 {
   t->setAuxName("usbL");
   send_timeout = cfg->value("send-timeout", 1000);
+  loop = nullptr;
   read_trigger.set<USBLowLevelDriver,&USBLowLevelDriver::read_trigger_cb>(this);
   write_trigger.set<USBLowLevelDriver,&USBLowLevelDriver::write_trigger_cb>(this);
   read_trigger.start();
