--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [build-system]
-build-backend = "mesonpy"
-backend-path = ['./vendored-meson/meson-python']
+#build-backend = "mesonpy"
+#backend-path = ['./vendored-meson/meson-python']
 requires = [
     "Cython>=0.29.34,<3.1",
     # All dependencies of the vendored meson-python (except for meson, because
@@ -194,8 +194,8 @@ repair-wheel-command = ""
 [tool.spin]
 package = 'numpy'
 
-[tool.spin.meson]
-cli = 'vendored-meson/meson/meson.py'
+#[tool.spin.meson]
+#cli = 'vendored-meson/meson/meson.py'
 
 [tool.spin.commands]
 "Build" = [
