From 0343813e0410b469d591bc61b9a546ee2c2c15f6 Mon Sep 17 00:00:00 2001
From: rofl0r <rofl0r@users.noreply.github.com>
Date: Fri, 17 May 2024 14:40:11 +0000
Subject: [PATCH] mute warning about shadow declaration of bind_addr

---
 sockssrv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/sockssrv.c
+++ b/sockssrv.c
@@ -112,8 +112,8 @@ struct thread {
 static void dolog(const char* fmt, ...) { }
 #endif
 
-static struct addrinfo* addr_choose(struct addrinfo* list, union sockaddr_union* bind_addr) {
-	int af = SOCKADDR_UNION_AF(bind_addr);
+static struct addrinfo* addr_choose(struct addrinfo* list, union sockaddr_union* bindaddr) {
+	int af = SOCKADDR_UNION_AF(bindaddr);
 	if(af == AF_UNSPEC) return list;
 	struct addrinfo* p;
 	for(p=list; p; p=p->ai_next)
