From ee3bf7f5079bcad51cc89197b6a9e34da746fc7e Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Sat, 7 Aug 2021 20:05:04 -0700
Subject: [PATCH] fix compilation without OpenSSL engines

Signed-off-by: Rosen Penev <rosenp@gmail.com>
---
 src/Mayaqua/Encrypt.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/src/Mayaqua/Encrypt.c
+++ b/src/Mayaqua/Encrypt.c
@@ -3128,6 +3128,7 @@ bool IsEncryptedK(BUF *b, bool private_k
 
 K *OpensslEngineToK(char *key_file_name, char *engine_name)
 {
+#ifndef OPENSSL_NO_ENGINE
     K *k;
 #if OPENSSL_API_COMPAT < 0x10100000L
     ENGINE_load_dynamic();
@@ -3140,6 +3141,9 @@ K *OpensslEngineToK(char *key_file_name,
     k->pkey = pkey;
     k->private_key = true;
     return k;
+#else
+    return NULL;
+#endif
 }
 
 // Convert the BUF to a K
