Index: bash-4.4/configure
===================================================================
--- bash-4.4.orig/configure
+++ bash-4.4/configure
@@ -5420,8 +5420,7 @@ if test $opt_readline = yes; then
 		# static version specified as -llibname to override the
 		# dynamic version
 		case "${host_os}" in
-		darwin[89]*|darwin10*) READLINE_LIB='${READLINE_LIBRARY}' ;;
-		*)		READLINE_LIB=-lreadline ;;
+		*) READLINE_LIB='${READLINE_LIBRARY}' ;;
 		esac
 	fi
 else
Index: bash-4.4/configure.ac
===================================================================
--- bash-4.4.orig/configure.ac
+++ bash-4.4/configure.ac
@@ -573,8 +573,7 @@ if test $opt_readline = yes; then
 		# static version specified as -llibname to override the
 		# dynamic version
 		case "${host_os}" in
-		darwin[[89]]*|darwin10*) READLINE_LIB='${READLINE_LIBRARY}' ;;
-		*)		READLINE_LIB=-lreadline ;;
+		*) READLINE_LIB='${READLINE_LIBRARY}' ;;
 		esac
 	fi
 else
