--- a/libmariadb/libmariadb/secure/openssl.c
+++ b/libmariadb/libmariadb/secure/openssl.c
@@ -419,8 +419,10 @@ void ma_tls_end()
     if (mariadb_deinitialize_ssl)
     {
 #ifndef HAVE_OPENSSL_1_1_API
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10000000L
       ERR_remove_state(0);
+#else
+      ERR_remove_thread_state(NULL);
 #endif
       EVP_cleanup();
       CRYPTO_cleanup_all_ex_data();
