--- a/support-files/CMakeLists.txt
+++ b/support-files/CMakeLists.txt
@@ -66,7 +66,7 @@ IF(UNIX AND NOT WITHOUT_SERVER)
     INSTALL(FILES magic DESTINATION ${inst_location} COMPONENT SupportFiles)
     INSTALL(DIRECTORY policy DESTINATION ${inst_location} COMPONENT SupportFiles)
     FIND_PROGRAM(CHECKMODULE checkmodule)
-    FIND_PROGRAM(SEMODULE_PACKAGE semodule_package)
+#    FIND_PROGRAM(SEMODULE_PACKAGE semodule_package)
     MARK_AS_ADVANCED(CHECKMODULE SEMODULE_PACKAGE)
 
     # Build pp files in policy/selinux
