--- a/mysys/CMakeLists.txt
+++ b/mysys/CMakeLists.txt
@@ -155,7 +155,7 @@ ENDIF()
 IF(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64|powerpc64" OR CMAKE_SYSTEM_NAME MATCHES AIX)
   SET(MYSYS_SOURCES ${MYSYS_SOURCES} crc32/crc32_ppc64.c crc32/crc32c_ppc.c)
   SET_SOURCE_FILES_PROPERTIES(crc32/crc32_ppc64.c crc32/crc32c_ppc.c PROPERTIES
-        COMPILE_FLAGS "${COMPILE_FLAGS} -maltivec -mvsx -mpower8-vector -mcrypto -mpower8-vector")
+        COMPILE_FLAGS "${COMPILE_FLAGS} -mvsx -mcrypto")
   ADD_DEFINITIONS(-DHAVE_POWER8 -DHAS_ALTIVEC)
 ELSE()
   SET (MYSYS_SOURCES ${MYSYS_SOURCES} crc32ieee.cc)
