From 139407b6b649217d4d53de118e265198a2d67e2b Mon Sep 17 00:00:00 2001
From: Sergei Golubchik <serg@mariadb.org>
Date: Wed, 12 Jul 2017 22:39:58 +0200
Subject: [PATCH 5/5] fix compilation with OpenSSL 1.1

---
 mysys_ssl/openssl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mysys_ssl/openssl.c b/mysys_ssl/openssl.c
index 31d29fb06d9..34f4d629ba6 100644
--- a/mysys_ssl/openssl.c
+++ b/mysys_ssl/openssl.c
@@ -34,7 +34,8 @@ int check_openssl_compatibility()
 
 static uint testing, alloc_size, alloc_count;
 
-static void *coc_malloc(size_t size, const char *, int)
+static void *coc_malloc(size_t size, const char *f __attribute__((unused)),
+                                             int l __attribute__((unused)))
 {
   if (unlikely(testing))
   {
-- 
2.13.3

