--- a/configure.ac
+++ b/configure.ac
@@ -765,7 +765,7 @@ if test x_$ub_test_python != x_no; then
    fi
 fi
 
-if test "`uname`" = "NetBSD"; then
+if test "${UNAME:-`uname`}" = "NetBSD"; then
 	NETBSD_LINTFLAGS='"-D__RENAME(x)=" -D_NETINET_IN_H_'
 	AC_SUBST(NETBSD_LINTFLAGS)
 fi
@@ -1210,7 +1210,7 @@ esac
 AC_ARG_ENABLE(tfo-client, AS_HELP_STRING([--enable-tfo-client],[Enable TCP Fast Open for client mode]))
 case "$enable_tfo_client" in
 	yes)
-		case `uname` in
+		case ${UNAME:-`uname`} in
 			Linux) AC_CHECK_DECL([MSG_FASTOPEN], [AC_MSG_WARN([Check the platform specific TFO kernel parameters are correctly configured to support client mode TFO])],
 			                     [AC_MSG_ERROR([TCP Fast Open is not available for client mode: please rerun without --enable-tfo-client])], 
 			                     [AC_INCLUDES_DEFAULT 
