Index: daemon/remote.c
===================================================================
--- a/daemon/remote.c	(revision 5105)
+++ b/daemon/remote.c	(working copy)
@@ -1987,7 +1987,7 @@
 				return NULL;
 			}
 		} else {
-#ifndef HAVE_SSL_SET1_HOST
+#if ! defined(HAVE_SSL_SET1_HOST) && ! defined(HAVE_X509_VERIFY_PARAM_SET1_HOST)
 			if(auth_name)
 			  log_err("no name verification functionality in "
 				"ssl library, ignored name for %s", todo);
Index: iterator/iter_fwd.c
===================================================================
--- a/iterator/iter_fwd.c	(revision 5105)
+++ b/iterator/iter_fwd.c	(working copy)
@@ -239,7 +239,7 @@
 				s->name, p->str);
 			return 0;
 		}
-#ifndef HAVE_SSL_SET1_HOST
+#if ! defined(HAVE_SSL_SET1_HOST) && ! defined(HAVE_X509_VERIFY_PARAM_SET1_HOST)
 		if(tls_auth_name)
 			log_err("no name verification functionality in "
 				"ssl library, ignored name for %s", p->str);
Index: iterator/iter_hints.c
===================================================================
--- a/iterator/iter_hints.c	(revision 5105)
+++ b/iterator/iter_hints.c	(working copy)
@@ -252,7 +252,7 @@
 				s->name, p->str);
 			return 0;
 		}
-#ifndef HAVE_SSL_SET1_HOST
+#if ! defined(HAVE_SSL_SET1_HOST) && ! defined(HAVE_X509_VERIFY_PARAM_SET1_HOST)
 		if(auth_name)
 			log_err("no name verification functionality in "
 				"ssl library, ignored name for %s", p->str);
