--- a/src/lzmainfo/Makefile.am
+++ b/src/lzmainfo/Makefile.am
@@ -20,7 +20,8 @@ lzmainfo_CPPFLAGS = \
 	-DLOCALEDIR=\"$(localedir)\" \
 	-I$(top_srcdir)/src/common \
 	-I$(top_srcdir)/src/liblzma/api \
-	-I$(top_builddir)/lib
+	-I$(top_builddir)/lib \
+	-std=gnu99
 
 lzmainfo_LDADD = $(top_builddir)/src/liblzma/liblzma.la
 
--- a/src/liblzma/common/common.h
+++ b/src/liblzma/common/common.h
@@ -89,9 +89,9 @@ typedef lzma_ret (*lzma_init_function)(
 /// function prototype.
 typedef lzma_ret (*lzma_code_function)(
 		lzma_coder *coder, lzma_allocator *allocator,
-		const uint8_t *restrict in, size_t *restrict in_pos,
-		size_t in_size, uint8_t *restrict out,
-		size_t *restrict out_pos, size_t out_size,
+		const uint8_t *in, size_t *in_pos,
+		size_t in_size, uint8_t *out,
+		size_t *out_pos, size_t out_size,
 		lzma_action action);
 
 /// Type of a function to free the memory allocated for the coder
@@ -235,9 +235,9 @@ extern void lzma_next_end(lzma_next_code
 
 /// Copy as much data as possible from in[] to out[] and update *in_pos
 /// and *out_pos accordingly. Returns the number of bytes copied.
-extern size_t lzma_bufcpy(const uint8_t *restrict in, size_t *restrict in_pos,
-		size_t in_size, uint8_t *restrict out,
-		size_t *restrict out_pos, size_t out_size);
+extern size_t lzma_bufcpy(const uint8_t *in, size_t *in_pos,
+		size_t in_size, uint8_t *out,
+		size_t *out_pos, size_t out_size);
 
 
 /// \brief      Return if expression doesn't evaluate to LZMA_OK
--- a/src/liblzma/common/common.c
+++ b/src/liblzma/common/common.c
@@ -70,9 +70,9 @@ lzma_free(void *ptr, lzma_allocator *all
 //////////
 
 extern size_t
-lzma_bufcpy(const uint8_t *restrict in, size_t *restrict in_pos,
-		size_t in_size, uint8_t *restrict out,
-		size_t *restrict out_pos, size_t out_size)
+lzma_bufcpy(const uint8_t *in, size_t *in_pos,
+		size_t in_size, uint8_t *out,
+		size_t *out_pos, size_t out_size)
 {
 	const size_t in_avail = in_size - *in_pos;
 	const size_t out_avail = out_size - *out_pos;
--- a/src/liblzma/Makefile.am
+++ b/src/liblzma/Makefile.am
@@ -23,7 +23,8 @@ liblzma_la_CPPFLAGS = \
 	-I$(top_srcdir)/src/liblzma/delta \
 	-I$(top_srcdir)/src/liblzma/simple \
 	-I$(top_srcdir)/src/common \
-	-DTUKLIB_SYMBOL_PREFIX=lzma_
+	-DTUKLIB_SYMBOL_PREFIX=lzma_ \
+	-std=gnu99
 liblzma_la_LDFLAGS = -no-undefined -version-info 5:4:0
 
 include $(srcdir)/common/Makefile.inc
--- a/src/xz/Makefile.am
+++ b/src/xz/Makefile.am
@@ -46,7 +46,8 @@ xz_CPPFLAGS = \
 	-DLOCALEDIR=\"$(localedir)\" \
 	-I$(top_srcdir)/src/common \
 	-I$(top_srcdir)/src/liblzma/api \
-	-I$(top_builddir)/lib
+	-I$(top_builddir)/lib \
+	-std=gnu99
 
 xz_LDADD = $(top_builddir)/src/liblzma/liblzma.la
 
