diff --git a/configure b/configure
index 4c2546e..e516137 100755
--- a/configure
+++ b/configure
@@ -2439,11 +2439,6 @@ if test -n "$x_libraries" && test "x$x_libraries" != xNONE ; then
 	LDFLAGS="$LDFLAGS -L$x_libraries"
 fi
 
-# May be equired for BSD
-LDFLAGS="$LDFLAGS -L/usr/local/lib"
-
-CPPFLAGS="$CPPFLAGS -I/usr/local/include"
-
 #Check for Darwin sw directory
 echo "$as_me:2448: checking for /sw/lib" >&5
 echo $ECHO_N "checking for /sw/lib... $ECHO_C" >&6
@@ -2505,9 +2500,6 @@ fi
 fi
 echo "$as_me:2506: result: $ac_cv_file__usr_pkg_lib" >&5
 echo "${ECHO_T}$ac_cv_file__usr_pkg_lib" >&6
-if test $ac_cv_file__usr_pkg_lib = yes; then
-  LDFLAGS="$LDFLAGS -L/usr/pkg/lib"
-fi
 
 echo "$as_me:2512: checking for /usr/pkg/include" >&5
 echo $ECHO_N "checking for /usr/pkg/include... $ECHO_C" >&6
@@ -3092,9 +3084,6 @@ fi
 fi
 echo "$as_me:3093: result: $ac_cv_file__usr_local_pgsql_lib" >&5
 echo "${ECHO_T}$ac_cv_file__usr_local_pgsql_lib" >&6
-if test $ac_cv_file__usr_local_pgsql_lib = yes; then
-  LDFLAGS="$LDFLAGS -L/usr/local/pgsql/lib"
-fi
 
 echo "$as_me:3099: checking for /usr/local/pgsql/include" >&5
 echo $ECHO_N "checking for /usr/local/pgsql/include... $ECHO_C" >&6
