From ca3b8191510c1006d031d02c50edcf6b4f6a6e8f Mon Sep 17 00:00:00 2001
From: R4SAS <r4sas@i2pmail.org>
Date: Thu, 10 Dec 2020 18:32:41 +0300
Subject: [PATCH] [avx] check ig c++ target supports AVX

Signed-off-by: R4SAS <r4sas@i2pmail.org>
---
 libi2pd/Crypto.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/libi2pd/Crypto.cpp
+++ b/libi2pd/Crypto.cpp
@@ -638,7 +638,7 @@ namespace crypto
 	{
 		uint64_t buf[256];
 		uint64_t hash[12]; // 96 bytes
-#if defined(__x86_64__) || defined(__i386__)
+#if (defined(__x86_64__) || defined(__i386__)) && defined(__AVX__) // not all X86 targets supports AVX (like old Pentium, see #1600)
 		if(i2p::cpu::avx)
 		{
 			__asm__
--- a/libi2pd/Identity.cpp
+++ b/libi2pd/Identity.cpp
@@ -828,7 +828,7 @@ namespace data
 	XORMetric operator^(const IdentHash& key1, const IdentHash& key2)
 	{
 		XORMetric m;
-#if defined(__x86_64__) || defined(__i386__)
+#if (defined(__x86_64__) || defined(__i386__)) && defined(__AVX__) // not all X86 targets supports AVX (like old Pentium, see #1600)
 		if(i2p::cpu::avx)
 		{
 			__asm__
