--- a/sh.proc.c
+++ b/sh.proc.c
@@ -48,7 +48,7 @@ RCSID("$tcsh: sh.proc.c,v 3.127 2015/02/
 #endif /* aiws */
 
 #if defined(_BSD) || (defined(IRIS4D) && __STDC__) || defined(__lucid) || defined(__linux__) || defined(__GNU__) || defined(__GLIBC__)
-# if !defined(__ANDROID__)
+# if !(defined(__ANDROID__) || (defined(__linux__) && !defined(__GLIBC__)))
 #  define BSDWAIT
 # endif
 #endif /* _BSD || (IRIS4D && __STDC__) || __lucid || glibc */
