--- a/etc/afpd/volume.c
+++ b/etc/afpd/volume.c
@@ -380,7 +380,11 @@ static int getvolparams(const AFPObj *ob
                         ashort |= VOLPBIT_ATTR_UNIXPRIV;
                     if (vol->v_flags & AFPVOL_TM)
                         ashort |= VOLPBIT_ATTR_TM;
-                    if (!ldap_config_valid || vol->v_flags & AFPVOL_NONETIDS)
+                    if (
+#ifdef HAVE_LDAP
+                        !ldap_config_valid ||
+#endif
+                        vol->v_flags & AFPVOL_NONETIDS)
                         ashort |= VOLPBIT_ATTR_NONETIDS;
                     if (obj->afp_version >= 32) {
                         if (vol->v_vfs_ea)
