--- a/pdns/Makefile.am
+++ b/pdns/Makefile.am
@@ -68,11 +68,9 @@ CLEANFILES = \
 	backends/gsql/gsqlbackend.gcda \
 	backends/gsql/gsqlbackend.gcno \
 	backends/gsql/gsqlbackend.gcov \
-	dnsmessage.pb.cc dnsmessage.pb.h \
-	pdns.conf-dist
+	dnsmessage.pb.cc dnsmessage.pb.h
 
 noinst_SCRIPTS = pdns.init
-sysconf_DATA = pdns.conf-dist
 
 sbin_PROGRAMS = pdns_server
 bin_PROGRAMS = \
@@ -112,7 +110,7 @@ if IXFRDIST
 bin_PROGRAMS += \
 	ixfrdist
 
-sysconf_DATA += \
+sysconf_DATA = \
 	ixfrdist.example.yml
 endif
 
@@ -1248,9 +1246,6 @@ dnspcap2protobuf_LDADD = \
 endif
 endif
 
-pdns.conf-dist: pdns_server
-	$(AM_V_GEN)./pdns_server --no-config --config=default 2>/dev/null > $@
-
 testrunner_SOURCES = \
 	arguments.cc \
 	auth-caches.cc auth-caches.hh \
