--- a/pdns/ixfrdist-web.cc
+++ b/pdns/ixfrdist-web.cc
@@ -25,6 +25,8 @@
 #include "iputils.hh"
 #include "ixfrdist-stats.hh"
 
+using namespace boost::placeholders;
+
 string doGetStats();
 
 IXFRDistWebServer::IXFRDistWebServer(const ComboAddress &listenAddress, const NetmaskGroup &acl, const string &loglevel) :
--- a/pdns/webserver.cc
+++ b/pdns/webserver.cc
@@ -36,6 +36,8 @@
 #include "uuid-utils.hh"
 #include <yahttp/router.hpp>
 
+using namespace boost::placeholders;
+
 json11::Json HttpRequest::json()
 {
   string err;
--- a/pdns/ws-auth.cc
+++ b/pdns/ws-auth.cc
@@ -49,6 +49,7 @@
 #include "tsigutils.hh"
 
 using json11::Json;
+using namespace boost::placeholders;
 
 extern StatBag S;
 
