--- a/cfg/vnstat.conf
+++ b/cfg/vnstat.conf
@@ -78,13 +78,13 @@ DaemonUser ""
 DaemonGroup ""
 
 # how often (in seconds) interface data is updated
-UpdateInterval 30
+UpdateInterval 60
 
 # how often (in seconds) interface status changes are checked
-PollInterval 5
+PollInterval 30
 
 # how often (in minutes) data is saved to file
-SaveInterval 5
+SaveInterval 30
 
 # how often (in minutes) data is saved when all interface are offline
 OfflineSaveInterval 30
