--- a/Makefile
+++ b/Makefile
@@ -43,7 +43,8 @@ install:
 # install default config if such doesn't exist
 	@if [ ! -f "$(DESTDIR)/etc/vnstat.conf" ]; \
 	then echo "Installing config to $(DESTDIR)/etc/vnstat.conf"; \
-	install -D -m 644 cfg/vnstat.conf $(DESTDIR)/etc/vnstat.conf; \
+	install -d -m 755 $(DESTDIR)/etc; \
+	install -m 644 cfg/vnstat.conf $(DESTDIR)/etc/vnstat.conf; \
 	fi
 
 # install everything else
@@ -63,7 +64,7 @@ install:
 	then echo install -m 644 man/vnstati.1 $(MAN)/man1; \
 	install -m 644 man/vnstati.1 $(MAN)/man1; \
 	fi
-	
+
 	@if [ -f $(MAN)/man1/vnstat.1.gz ]; \
 	then gzip -f9 $(MAN)/man1/vnstat.1; \
 	gzip -f9 $(MAN)/man1/vnstatd.1; \
