--- a/cmake/CheckFunctions.cmake
+++ b/cmake/CheckFunctions.cmake
@@ -29,24 +29,6 @@ function(check_leptonica_tiff_support)
   "  return ret_val;}\n")
   if(${CMAKE_VERSION} VERSION_LESS "3.25")
     message(STATUS "Testing TIFF support in Leptonica is available with CMake >= 3.25 (you have ${CMAKE_VERSION}))")
-  else()
-    set(CMAKE_TRY_COMPILE_CONFIGURATION ${CMAKE_BUILD_TYPE})
-    try_run(
-      LEPT_TIFF_RESULT
-      LEPT_TIFF_COMPILE_SUCCESS
-      SOURCE_FROM_CONTENT tiff_test.cpp "${TIFF_TEST}"
-      CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${Leptonica_INCLUDE_DIRS}"
-      LINK_LIBRARIES ${Leptonica_LIBRARIES}
-      COMPILE_OUTPUT_VARIABLE
-      COMPILE_OUTPUT)
-    if(NOT LEPT_TIFF_COMPILE_SUCCESS)
-      message(STATUS "COMPILE_OUTPUT: ${COMPILE_OUTPUT}")
-      message(STATUS "Leptonica_INCLUDE_DIRS: ${Leptonica_INCLUDE_DIRS}")
-      message(STATUS "Leptonica_LIBRARIES: ${Leptonica_LIBRARIES}")
-      message(STATUS "LEPT_TIFF_RESULT: ${LEPT_TIFF_RESULT}")
-      message(STATUS "LEPT_TIFF_COMPILE: ${LEPT_TIFF_COMPILE}")
-      message(WARNING "Failed to compile test")
-    endif()
   endif()
 endfunction(check_leptonica_tiff_support)
 
