--- a/setup.py
+++ b/setup.py
@@ -55,16 +55,16 @@ HERE = os.path.abspath(os.path.dirname(_
 # ...so we can import _common.py
 sys.path.insert(0, os.path.join(HERE, "psutil"))
 
-from _common import AIX  # noqa: E402
-from _common import BSD  # noqa: E402
-from _common import FREEBSD  # noqa: E402
-from _common import LINUX  # noqa: E402
-from _common import MACOS  # noqa: E402
-from _common import NETBSD  # noqa: E402
-from _common import OPENBSD  # noqa: E402
-from _common import POSIX  # noqa: E402
-from _common import SUNOS  # noqa: E402
-from _common import WINDOWS  # noqa: E402
+AIX = False
+BSD = False
+FREEBSD = False
+LINUX = True
+MACOS = False
+NETBSD = False
+OPENBSD = False
+POSIX = True
+SUNOS = False
+WINDOWS = False
 from _common import hilite  # noqa: E402
 
 PYPY = '__pypy__' in sys.builtin_module_names
