--- a/setup.py
+++ b/setup.py
@@ -43,16 +43,16 @@ HERE = os.path.abspath(os.path.dirname(_
 # ...so we can import _common.py and _compat.py
 sys.path.insert(0, os.path.join(HERE, "psutil"))
 
-from _common import AIX  # NOQA
-from _common import BSD  # NOQA
-from _common import FREEBSD  # NOQA
-from _common import LINUX  # NOQA
-from _common import MACOS  # NOQA
-from _common import NETBSD  # NOQA
-from _common import OPENBSD  # NOQA
-from _common import POSIX  # NOQA
-from _common import SUNOS  # NOQA
-from _common import WINDOWS  # NOQA
+AIX = False
+BSD = False
+FREEBSD = False
+LINUX = True
+MACOS = False
+NETBSD = False
+OPENBSD = False
+POSIX = True
+SUNOS = False
+WINDOWS = False
 from _common import hilite  # NOQA
 from _compat import PY3  # NOQA
 from _compat import which  # NOQA
