--- a/configure.ac
+++ b/configure.ac
@@ -147,8 +147,10 @@ case "${host_os}" in
     *linux*)
         # Define on Linux to enable all library features. Define if using a gnu compiler
         if test "x$GXX" = "xyes"; then
-            CPPFLAGS="-D_GNU_SOURCE $CPPFLAGS"
+            CPPFLAGS="-D_GNU_SOURCE $CPPFLAGS -Wno-long-long"
         fi
+        libzmq_pedantic="no"
+        libzmq_werror="no"
         AC_DEFINE(ZMQ_HAVE_LINUX, 1, [Have Linux OS])
         libzmq_on_linux="yes"
 
