--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,8 @@
 cmake_minimum_required(VERSION 2.8.11)
 project(ZeroMQ)
 
+include(FindPkgConfig)
+
 option(WITH_OPENPGM "Build with support for OpenPGM" OFF)
 
 if(APPLE)
@@ -21,7 +23,7 @@ if (NOT ENABLE_CURVE)
     message (STATUS "CURVE security is disabled")
 
 elseif (WITH_LIBSODIUM)
-    find_package (Sodium)
+    pkg_search_module (SODIUM REQUIRED libsodium)
     if (SODIUM_FOUND)
         message (STATUS "Using libsodium for CURVE security")
         include_directories (${SODIUM_INCLUDE_DIRS})
