--- a/configure.ac	2014-06-04 10:19:35.140061796 +0800
+++ b/configure.ac	2014-06-04 11:08:17.231997926 +0800
@@ -127,8 +127,10 @@
     *linux*)
         # Define on Linux to enable all library features. Define if using a gnu compiler
         if test "x$GXX" = "xyes"; then
-            CPPFLAGS="-D_GNU_SOURCE $CPPFLAGS"
+            CPPFLAGS="-D_GNU_SOURCE $CPPFLAGS -Wno-long-long"
         fi
+        libzmq_pedantic="no"
+        libzmq_werror="no"
         AC_DEFINE(ZMQ_HAVE_LINUX, 1, [Have Linux OS])
         libzmq_on_linux="yes"
 
