--- a/src/metadata.hpp	2015-06-17 21:39:33.629934688 +0800
+++ b/src/metadata.hpp	2014-10-14 22:07:36.000000000 +0800
@@ -31,7 +31,7 @@
     {
         public:
 
-            typedef std::map <std::string, const std::string> dict_t;
+            typedef std::map <std::string, std::string> dict_t;
 
             metadata_t (const dict_t &dict);
             virtual ~metadata_t ();
--- a/src/socket_base.cpp	2015-06-17 21:42:26.289413233 +0800
+++ b/src/socket_base.cpp	2015-06-02 18:31:00.000000000 +0800
@@ -20,6 +20,7 @@
 #include <new>
 #include <string>
 #include <algorithm>
+#include <ctype.h>
 
 #include "platform.hpp"
 
--- a/src/stream_engine.cpp	2015-06-17 21:57:29.599443517 +0800
+++ b/src/stream_engine.cpp	2015-06-02 18:31:00.000000000 +0800
@@ -198,7 +198,7 @@
             //  Compile metadata.
             typedef metadata_t::dict_t properties_t;
             properties_t properties;
-            properties.insert(std::make_pair("Peer-Address", peer_address));
+            properties.insert(std::make_pair<std::string, std::string>("Peer-Address", peer_address));
             zmq_assert (metadata == NULL);
             metadata = new (std::nothrow) metadata_t (properties);
         }
@@ -805,7 +805,7 @@
 
     //  If we have a peer_address, add it to metadata
     if (!peer_address.empty()) {
-        properties.insert(std::make_pair("Peer-Address", peer_address));
+        properties.insert(std::make_pair<std::string, std::string>("Peer-Address", peer_address));
     }
 
     //  Add ZAP properties.
