Index: serdisplib-2.01/src/Makefile.in
===================================================================
--- serdisplib-2.01.orig/src/Makefile.in
+++ serdisplib-2.01/src/Makefile.in
@@ -28,7 +28,7 @@ includedir = @includedir@
 datarootdir = @datarootdir@
 
 CC=@CC@
-AR=@AR@ -r
+AR=@AR@
 LN_S=@LN_S@
 INSTALL=@INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
@@ -180,7 +180,7 @@ programs: $(PROGRAMS)
 
 $(LIB_DIR)/$(LIB_STATIC): $(LIB_OBJECTS)
 	$(top_srcdir)/mkinstalldirs $(top_srcdir)/lib
-	$(AR) $(LIB_DIR)/$(LIB_STATIC) $(LIB_OBJECTS)
+	$(AR) -r -- $(LIB_DIR)/$(LIB_STATIC) $(LIB_OBJECTS)
 
 $(LIB_DIR)/$(LIB_SHARED): $(LIB_OBJECTS)
 	$(top_srcdir)/mkinstalldirs $(top_srcdir)/lib
