Derived from uptream commit d5b09f2b: http://octo.cx/d5b09f2

--- a/src/types.db
+++ b/src/types.db
@@ -152,7 +152,7 @@ tcp_connections		value:GAUGE:0:429496729
 temperature		value:GAUGE:-273.15:U
 threads			value:GAUGE:0:U
 time_dispersion		seconds:GAUGE:-1000000:1000000
-timeleft		timeleft:GAUGE:0:3600
+timeleft		timeleft:GAUGE:0:U
 time_offset		seconds:GAUGE:-1000000:1000000
 total_bytes		value:DERIVE:0:U
 total_requests		value:DERIVE:0:U

