--- a/configure.ac
+++ b/configure.ac
@@ -713,6 +713,11 @@ AC_CACHE_CHECK([whether clock_boottime a
   ]
 )
 
+# For the iwinfo plugin
+AC_CHECK_LIB([iwinfo], [iwinfo_backend],
+    [with_iwinfo="yes"],
+    [with_iwinfo="no (libiwinfo not found)"]
+)
 
 #
 # Checks for typedefs, structures, and compiler characteristics.
@@ -6619,6 +6624,7 @@ plugin_ipmi="no"
 plugin_ipstats="no"
 plugin_ipvs="no"
 plugin_irq="no"
+plugin_iwinfo="no"
 plugin_load="no"
 plugin_log_logstash="no"
 plugin_mcelog="no"
@@ -7086,6 +7092,7 @@ AC_PLUGIN([iptables],            [$with_
 AC_PLUGIN([ipstats],             [$plugin_ipstats],           [IP packet statistics])
 AC_PLUGIN([ipvs],                [$plugin_ipvs],              [IPVS connection statistics])
 AC_PLUGIN([irq],                 [$plugin_irq],               [IRQ statistics])
+AC_PLUGIN([iwinfo],              [$with_iwinfo],              [Common iwinfo wireless statistics])
 AC_PLUGIN([java],                [$with_java],                [Embed the Java Virtual Machine])
 AC_PLUGIN([load],                [$plugin_load],              [System load])
 AC_PLUGIN([log_logstash],        [$plugin_log_logstash],      [Logstash json_event compatible logging])
@@ -7465,6 +7472,7 @@ AC_MSG_RESULT([    libyajl . . . . . . .
 AC_MSG_RESULT([    oracle  . . . . . . . $with_oracle])
 AC_MSG_RESULT([    protobuf-c  . . . . . $have_protoc_c])
 AC_MSG_RESULT([    protoc 3  . . . . . . $have_protoc3])
+AC_MSG_RESULT([    iwinfo  . . . . . . . $with_iwinfo])
 AC_MSG_RESULT()
 AC_MSG_RESULT([  Features:])
 AC_MSG_RESULT([    daemon mode . . . . . $enable_daemon])
@@ -7533,6 +7541,7 @@ AC_MSG_RESULT([    iptables  . . . . . .
 AC_MSG_RESULT([    ipstats . . . . . . . $enable_ipstats])
 AC_MSG_RESULT([    ipvs  . . . . . . . . $enable_ipvs])
 AC_MSG_RESULT([    irq . . . . . . . . . $enable_irq])
+AC_MSG_RESULT([    iwinfo  . . . . . . . $enable_iwinfo])
 AC_MSG_RESULT([    java  . . . . . . . . $enable_java])
 AC_MSG_RESULT([    load  . . . . . . . . $enable_load])
 AC_MSG_RESULT([    logfile . . . . . . . $enable_logfile])
--- a/src/collectd.conf.in
+++ b/src/collectd.conf.in
@@ -147,6 +147,7 @@
 #@BUILD_PLUGIN_IPTABLES_TRUE@LoadPlugin iptables
 #@BUILD_PLUGIN_IPVS_TRUE@LoadPlugin ipvs
 #@BUILD_PLUGIN_IRQ_TRUE@LoadPlugin irq
+#@BUILD_PLUGIN_IWINFO_TRUE@LoadPlugin iwinfo
 #@BUILD_PLUGIN_JAVA_TRUE@LoadPlugin java
 @BUILD_PLUGIN_LOAD_TRUE@@BUILD_PLUGIN_LOAD_TRUE@LoadPlugin load
 #@BUILD_PLUGIN_LPAR_TRUE@LoadPlugin lpar
@@ -897,6 +898,12 @@
 #	IgnoreSelected true
 #</Plugin>
 
+#<Plugin iwinfo>
+#   Interface "ath0"
+#   Interface "ra0"
+#   Interface "wlan0"
+#</Plugin>
+
 #<Plugin java>
 #	JVMArg "-verbose:jni"
 #	JVMArg "-Djava.class.path=@prefix@/share/collectd/java/collectd-api.jar"
--- a/src/collectd.conf.pod
+++ b/src/collectd.conf.pod
@@ -4343,6 +4343,27 @@ and all other interrupts are collected.
 
 =back
 
+=head2 Plugin C<iwinfo>
+
+=over 4
+
+=item B<Interface> I<Interface>
+
+Select this interface. By default all detected wireless interfaces will be
+collected. For a more detailed description see B<IgnoreSelected> below.
+
+=item B<IgnoreSelected> I<true>|I<false>
+
+If no configuration if given, the B<iwinfo>-plugin will collect data from all
+detected wireless interfaces. You can use the B<Interface>-option to pick the
+interfaces you're interested in. Sometimes, however, it's easier/preferred to
+collect all interfaces I<except> a few ones. This option enables you to do
+that: By setting B<IgnoreSelected> to I<true> the effect of B<Interface> is
+inverted: All selected interfaces are ignored and all other interfaces are
+collected.
+
+=back
+
 =head2 Plugin C<java>
 
 The I<Java> plugin makes it possible to write extensions for collectd in Java.
--- a/src/types.db
+++ b/src/types.db
@@ -308,6 +308,7 @@ snr                     value:GAUGE:0:U
 spam_check              value:GAUGE:0:U
 spam_score              value:GAUGE:U:U
 spl                     value:GAUGE:U:U
+stations                value:GAUGE:0:256
 swap                    value:GAUGE:0:1099511627776
 swap_io                 value:DERIVE:0:U
 sysevent                value:GAUGE:0:1
--- a/Makefile.am
+++ b/Makefile.am
@@ -1246,6 +1246,14 @@ irq_la_LDFLAGS = $(PLUGIN_LDFLAGS)
 irq_la_LIBADD = libignorelist.la
 endif
 
+if BUILD_PLUGIN_IWINFO
+pkglib_LTLIBRARIES += iwinfo.la
+iwinfo_la_SOURCES = src/iwinfo.c
+#iwinfo_la_LDFLAGS = -module -avoid-version
+iwinfo_la_LDFLAGS = $(PLUGIN_LDFLAGS)
+iwinfo_la_LIBADD = -liwinfo libignorelist.la
+endif
+
 if BUILD_PLUGIN_JAVA
 pkglib_LTLIBRARIES += java.la
 java_la_SOURCES = src/java.c
